<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Generalized Linear Models | New statistics for the design researcher</title>
  <meta name="description" content="A statistics book for designers, human factors specialists, UX researchers, applied psychologists and everyone else who works hard to make this world a better place." />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Generalized Linear Models | New statistics for the design researcher" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A statistics book for designers, human factors specialists, UX researchers, applied psychologists and everyone else who works hard to make this world a better place." />
  <meta name="github-repo" content="schmettow/New_Stats" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Generalized Linear Models | New statistics for the design researcher" />
  
  <meta name="twitter:description" content="A statistics book for designers, human factors specialists, UX researchers, applied psychologists and everyone else who works hard to make this world a better place." />
  

<meta name="author" content="Martin Schmettow" />


<meta name="date" content="2020-09-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="mlm.html"/>
<link rel="next" href="wwm.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="part"><span><b>I Preparations</b></span></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#whom-for"><i class="fa fa-check"></i><b>1.1</b> Whom this book is for</a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#design-researcher"><i class="fa fa-check"></i><b>1.1.1</b> The empirical design researcher</a></li>
<li class="chapter" data-level="1.1.2" data-path="index.html"><a href="index.html#experimentalist"><i class="fa fa-check"></i><b>1.1.2</b> The experimentalist</a></li>
<li class="chapter" data-level="1.1.3" data-path="index.html"><a href="index.html#applied-researcher"><i class="fa fa-check"></i><b>1.1.3</b> The applied researcher</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#what-new-stats"><i class="fa fa-check"></i><b>1.2</b> What is New Statistics?</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#how-to-read"><i class="fa fa-check"></i><b>1.3</b> How to read this book</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#quant-design-research"><i class="fa fa-check"></i><b>1.4</b> Quantitative design research</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#studies-in-design-research"><i class="fa fa-check"></i><b>1.5</b> Studies in Design Research</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#observations-measures"><i class="fa fa-check"></i><b>1.6</b> Observations and measures</a><ul>
<li class="chapter" data-level="1.6.1" data-path="index.html"><a href="index.html#interaction-seq"><i class="fa fa-check"></i><b>1.6.1</b> Interaction sequences</a></li>
<li class="chapter" data-level="1.6.2" data-path="index.html"><a href="index.html#perf-measures"><i class="fa fa-check"></i><b>1.6.2</b> performance measures</a></li>
<li class="chapter" data-level="1.6.3" data-path="index.html"><a href="index.html#experience-tbd"><i class="fa fa-check"></i><b>1.6.3</b> experience [TBD]</a></li>
<li class="chapter" data-level="1.6.4" data-path="index.html"><a href="index.html#design-features-tbd"><i class="fa fa-check"></i><b>1.6.4</b> design features [TBD]</a></li>
<li class="chapter" data-level="1.6.5" data-path="index.html"><a href="index.html#the-human-factor-tbd"><i class="fa fa-check"></i><b>1.6.5</b> the human factor [TBD]</a></li>
<li class="chapter" data-level="1.6.6" data-path="index.html"><a href="index.html#situations-tbd"><i class="fa fa-check"></i><b>1.6.6</b> situations [TBD]</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="gsr.html"><a href="gsr.html"><i class="fa fa-check"></i><b>2</b> Getting started with R</a><ul>
<li class="chapter" data-level="2.1" data-path="gsr.html"><a href="gsr.html#setting-up-r"><i class="fa fa-check"></i><b>2.1</b> Setting up the R environment</a><ul>
<li class="chapter" data-level="2.1.1" data-path="gsr.html"><a href="gsr.html#install-cran"><i class="fa fa-check"></i><b>2.1.1</b> Installing CRAN packages</a></li>
<li class="chapter" data-level="2.1.2" data-path="gsr.html"><a href="gsr.html#install-github"><i class="fa fa-check"></i><b>2.1.2</b> Installing packages from Github</a></li>
<li class="chapter" data-level="2.1.3" data-path="gsr.html"><a href="gsr.html#first-program"><i class="fa fa-check"></i><b>2.1.3</b> A first statistical program</a></li>
<li class="chapter" data-level="2.1.4" data-path="gsr.html"><a href="gsr.html#bibliographic-notes"><i class="fa fa-check"></i><b>2.1.4</b> Bibliographic notes</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="gsr.html"><a href="gsr.html#r-primer"><i class="fa fa-check"></i><b>2.2</b> Learning R: a primer</a><ul>
<li class="chapter" data-level="2.2.1" data-path="gsr.html"><a href="gsr.html#objects"><i class="fa fa-check"></i><b>2.2.1</b> Assigning and calling Objects</a></li>
<li class="chapter" data-level="2.2.2" data-path="gsr.html"><a href="gsr.html#vectors"><i class="fa fa-check"></i><b>2.2.2</b> Vectors</a></li>
<li class="chapter" data-level="2.2.3" data-path="gsr.html"><a href="gsr.html#object-types"><i class="fa fa-check"></i><b>2.2.3</b> Basic object types</a></li>
<li class="chapter" data-level="2.2.4" data-path="gsr.html"><a href="gsr.html#operators-functions"><i class="fa fa-check"></i><b>2.2.4</b> Operators and functions</a></li>
<li class="chapter" data-level="2.2.5" data-path="gsr.html"><a href="gsr.html#data-frames"><i class="fa fa-check"></i><b>2.2.5</b> Storing data in data frames</a></li>
<li class="chapter" data-level="2.2.6" data-path="gsr.html"><a href="gsr.html#import-export"><i class="fa fa-check"></i><b>2.2.6</b> Import, export and archiving</a></li>
<li class="chapter" data-level="2.2.7" data-path="gsr.html"><a href="gsr.html#case-env"><i class="fa fa-check"></i><b>2.2.7</b> Case environments</a></li>
<li class="chapter" data-level="2.2.8" data-path="gsr.html"><a href="gsr.html#structuring-data"><i class="fa fa-check"></i><b>2.2.8</b> Structuring data</a></li>
<li class="chapter" data-level="2.2.9" data-path="gsr.html"><a href="gsr.html#data-transformation"><i class="fa fa-check"></i><b>2.2.9</b> Data transformation</a></li>
<li class="chapter" data-level="2.2.10" data-path="gsr.html"><a href="gsr.html#plotting"><i class="fa fa-check"></i><b>2.2.10</b> Plotting data</a></li>
<li class="chapter" data-level="2.2.11" data-path="gsr.html"><a href="gsr.html#fitting"><i class="fa fa-check"></i><b>2.2.11</b> Fitting regression models</a></li>
<li class="chapter" data-level="2.2.12" data-path="gsr.html"><a href="gsr.html#knitting"><i class="fa fa-check"></i><b>2.2.12</b> Knitting statistical reports</a></li>
<li class="chapter" data-level="2.2.13" data-path="gsr.html"><a href="gsr.html#ex-gsr"><i class="fa fa-check"></i><b>2.2.13</b> Exercises</a></li>
<li class="chapter" data-level="2.2.14" data-path="gsr.html"><a href="gsr.html#lib_gsr"><i class="fa fa-check"></i><b>2.2.14</b> Bibliographic notes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="ebs.html"><a href="ebs.html"><i class="fa fa-check"></i><b>3</b> Elements of Bayesian statistics</a><ul>
<li class="chapter" data-level="3.1" data-path="ebs.html"><a href="ebs.html#decision-making"><i class="fa fa-check"></i><b>3.1</b> Rational decision making in design research</a><ul>
<li class="chapter" data-level="3.1.1" data-path="ebs.html"><a href="ebs.html#measuring-uncertainty"><i class="fa fa-check"></i><b>3.1.1</b> Measuring uncertainty</a></li>
<li class="chapter" data-level="3.1.2" data-path="ebs.html"><a href="ebs.html#benchmarking-designs"><i class="fa fa-check"></i><b>3.1.2</b> Benchmarking designs</a></li>
<li class="chapter" data-level="3.1.3" data-path="ebs.html"><a href="ebs.html#comparing-designs"><i class="fa fa-check"></i><b>3.1.3</b> Comparison of designs</a></li>
<li class="chapter" data-level="3.1.4" data-path="ebs.html"><a href="ebs.html#prior-knowledge"><i class="fa fa-check"></i><b>3.1.4</b> Prior knowledge</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="ebs.html"><a href="ebs.html#descriptive-stats"><i class="fa fa-check"></i><b>3.2</b> Descriptive statistics</a><ul>
<li class="chapter" data-level="3.2.1" data-path="ebs.html"><a href="ebs.html#frequencies"><i class="fa fa-check"></i><b>3.2.1</b> Frequencies</a></li>
<li class="chapter" data-level="3.2.2" data-path="ebs.html"><a href="ebs.html#central-tendency"><i class="fa fa-check"></i><b>3.2.2</b> Central tendency</a></li>
<li class="chapter" data-level="3.2.3" data-path="ebs.html"><a href="ebs.html#dispersion"><i class="fa fa-check"></i><b>3.2.3</b> Dispersion</a></li>
<li class="chapter" data-level="3.2.4" data-path="ebs.html"><a href="ebs.html#associations"><i class="fa fa-check"></i><b>3.2.4</b> Associations</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="ebs.html"><a href="ebs.html#bayes-prob-theory"><i class="fa fa-check"></i><b>3.3</b> Bayesian probability theory</a><ul>
<li class="chapter" data-level="3.3.1" data-path="ebs.html"><a href="ebs.html#set-theory"><i class="fa fa-check"></i><b>3.3.1</b> Some set theory</a></li>
<li class="chapter" data-level="3.3.2" data-path="ebs.html"><a href="ebs.html#probability"><i class="fa fa-check"></i><b>3.3.2</b> Probability</a></li>
<li class="chapter" data-level="3.3.3" data-path="ebs.html"><a href="ebs.html#likelihood"><i class="fa fa-check"></i><b>3.3.3</b> Likelihood</a></li>
<li class="chapter" data-level="3.3.4" data-path="ebs.html"><a href="ebs.html#bayes-freq-prob"><i class="fa fa-check"></i><b>3.3.4</b> Bayesian and frequentist probability</a></li>
<li class="chapter" data-level="3.3.5" data-path="ebs.html"><a href="ebs.html#bayes-theorem"><i class="fa fa-check"></i><b>3.3.5</b> Bayes theorem</a></li>
<li class="chapter" data-level="3.3.6" data-path="ebs.html"><a href="ebs.html#dynamics-belief"><i class="fa fa-check"></i><b>3.3.6</b> Bayesian dynamics of belief</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="ebs.html"><a href="ebs.html#statmod"><i class="fa fa-check"></i><b>3.4</b> Statistical models</a><ul>
<li class="chapter" data-level="3.4.1" data-path="ebs.html"><a href="ebs.html#structural-part"><i class="fa fa-check"></i><b>3.4.1</b> The structural part</a></li>
<li class="chapter" data-level="3.4.2" data-path="ebs.html"><a href="ebs.html#distributions"><i class="fa fa-check"></i><b>3.4.2</b> Distributions: shapes of randomness</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="ebs.html"><a href="ebs.html#bayes-estimation"><i class="fa fa-check"></i><b>3.5</b> Towards Bayesian estimation</a></li>
</ul></li>
<li class="part"><span><b>II Models</b></span></li>
<li class="chapter" data-level="4" data-path="lm.html"><a href="lm.html"><i class="fa fa-check"></i><b>4</b> Linear models</a><ul>
<li class="chapter" data-level="4.1" data-path="lm.html"><a href="lm.html#GMM"><i class="fa fa-check"></i><b>4.1</b> Quantification at work: grand mean models</a><ul>
<li class="chapter" data-level="4.1.1" data-path="lm.html"><a href="lm.html#random-walk"><i class="fa fa-check"></i><b>4.1.1</b> Do the random walk: Markov Chain Monte Carlo sampling</a></li>
<li class="chapter" data-level="4.1.2" data-path="lm.html"><a href="lm.html#likelihood-random-term"><i class="fa fa-check"></i><b>4.1.2</b> Likelihood and random term</a></li>
<li class="chapter" data-level="4.1.3" data-path="lm.html"><a href="lm.html#posterior-dist"><i class="fa fa-check"></i><b>4.1.3</b> Working with the posterior distribution</a></li>
<li class="chapter" data-level="4.1.4" data-path="lm.html"><a href="lm.html#clu"><i class="fa fa-check"></i><b>4.1.4</b> Center and interval estimates</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="lm.html"><a href="lm.html#lrm"><i class="fa fa-check"></i><b>4.2</b> Walk the line: linear regression</a><ul>
<li class="chapter" data-level="4.2.1" data-path="lm.html"><a href="lm.html#transform-measures"><i class="fa fa-check"></i><b>4.2.1</b> Transforming measures</a></li>
<li class="chapter" data-level="4.2.2" data-path="lm.html"><a href="lm.html#correlations"><i class="fa fa-check"></i><b>4.2.2</b> Correlations</a></li>
<li class="chapter" data-level="4.2.3" data-path="lm.html"><a href="lm.html#endless-linear"><i class="fa fa-check"></i><b>4.2.3</b> Endlessly linear</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="lm.html"><a href="lm.html#factorial-models"><i class="fa fa-check"></i><b>4.3</b> Factorial Models</a><ul>
<li class="chapter" data-level="4.3.1" data-path="lm.html"><a href="lm.html#cgm"><i class="fa fa-check"></i><b>4.3.1</b> A versus B: Comparison of groups</a></li>
<li class="chapter" data-level="4.3.2" data-path="lm.html"><a href="lm.html#dummy"><i class="fa fa-check"></i><b>4.3.2</b> Not stupid: dummy variables</a></li>
<li class="chapter" data-level="4.3.3" data-path="lm.html"><a href="lm.html#treatment-contrasts"><i class="fa fa-check"></i><b>4.3.3</b> Treatment contrast</a></li>
<li class="chapter" data-level="4.3.4" data-path="lm.html"><a href="lm.html#amm"><i class="fa fa-check"></i><b>4.3.4</b> Absolute Means Model</a></li>
<li class="chapter" data-level="4.3.5" data-path="lm.html"><a href="lm.html#ofm"><i class="fa fa-check"></i><b>4.3.5</b> Ordered Factors Models</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="lm.html"><a href="lm.html#mpm"><i class="fa fa-check"></i><b>4.4</b> Putting it all together: multi predictor models</a><ul>
<li class="chapter" data-level="4.4.1" data-path="lm.html"><a href="lm.html#mrm"><i class="fa fa-check"></i><b>4.4.1</b> On surface: multiple regression models</a></li>
<li class="chapter" data-level="4.4.2" data-path="lm.html"><a href="lm.html#mfm"><i class="fa fa-check"></i><b>4.4.2</b> Crossover: multifactorial models</a></li>
<li class="chapter" data-level="4.4.3" data-path="lm.html"><a href="lm.html#grm"><i class="fa fa-check"></i><b>4.4.3</b> Line-by-line: grouped regression models</a></li>
<li class="chapter" data-level="4.4.4" data-path="lm.html"><a href="lm.html#emp-stat-control"><i class="fa fa-check"></i><b>4.4.4</b> Empirical versus statistical control</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="lm.html"><a href="lm.html#cfxm"><i class="fa fa-check"></i><b>4.5</b> Conditional effects models</a><ul>
<li class="chapter" data-level="4.5.1" data-path="lm.html"><a href="lm.html#cmrm"><i class="fa fa-check"></i><b>4.5.1</b> Conditional multiple regression</a></li>
<li class="chapter" data-level="4.5.2" data-path="lm.html"><a href="lm.html#cmfm"><i class="fa fa-check"></i><b>4.5.2</b> Conditional multifactorial models</a></li>
<li class="chapter" data-level="4.5.3" data-path="lm.html"><a href="lm.html#saturation"><i class="fa fa-check"></i><b>4.5.3</b> Hitting the boundaries of saturation</a></li>
<li class="chapter" data-level="4.5.4" data-path="lm.html"><a href="lm.html#amplification"><i class="fa fa-check"></i><b>4.5.4</b> More than the sum: amplification</a></li>
<li class="chapter" data-level="4.5.5" data-path="lm.html"><a href="lm.html#cfx-theory"><i class="fa fa-check"></i><b>4.5.5</b> Conditional effects and theory</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="lm.html"><a href="lm.html#prm"><i class="fa fa-check"></i><b>4.6</b> Doing the rollercoaster: polynomial regression models</a><ul>
<li class="chapter" data-level="4.6.1" data-path="lm.html"><a href="lm.html#test-stat"><i class="fa fa-check"></i><b>4.6.1</b> Make yourself a test statistic</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="mlm.html"><a href="mlm.html"><i class="fa fa-check"></i><b>5</b> Multilevel models</a><ul>
<li class="chapter" data-level="5.1" data-path="mlm.html"><a href="mlm.html#intercept-re"><i class="fa fa-check"></i><b>5.1</b> The Human Factor: Intercept random effects</a></li>
<li class="chapter" data-level="5.2" data-path="mlm.html"><a href="mlm.html#slope-re"><i class="fa fa-check"></i><b>5.2</b> Slope random effects: variance in change</a></li>
<li class="chapter" data-level="5.3" data-path="mlm.html"><a href="mlm.html#thinking-multi-level"><i class="fa fa-check"></i><b>5.3</b> Thinking multi-level</a></li>
<li class="chapter" data-level="5.4" data-path="mlm.html"><a href="mlm.html#universality"><i class="fa fa-check"></i><b>5.4</b> Testing universality of theories</a></li>
<li class="chapter" data-level="5.5" data-path="mlm.html"><a href="mlm.html#non-human-populations"><i class="fa fa-check"></i><b>5.5</b> Non-human populations and cross-overs</a></li>
<li class="chapter" data-level="5.6" data-path="mlm.html"><a href="mlm.html#nested-re"><i class="fa fa-check"></i><b>5.6</b> Nested random effects</a></li>
<li class="chapter" data-level="5.7" data-path="mlm.html"><a href="mlm.html#pool-shrink"><i class="fa fa-check"></i><b>5.7</b> What are random effects? On pooling and shrinkage</a></li>
<li class="chapter" data-level="5.8" data-path="mlm.html"><a href="mlm.html#psychometrics"><i class="fa fa-check"></i><b>5.8</b> Psychometrics and designometric models</a><ul>
<li class="chapter" data-level="5.8.1" data-path="mlm.html"><a href="mlm.html#coverage"><i class="fa fa-check"></i><b>5.8.1</b> Coverage</a></li>
<li class="chapter" data-level="5.8.2" data-path="mlm.html"><a href="mlm.html#reliability"><i class="fa fa-check"></i><b>5.8.2</b> Reliability</a></li>
<li class="chapter" data-level="5.8.3" data-path="mlm.html"><a href="mlm.html#validity"><i class="fa fa-check"></i><b>5.8.3</b> Validity</a></li>
<li class="chapter" data-level="5.8.4" data-path="mlm.html"><a href="mlm.html#designometrix"><i class="fa fa-check"></i><b>5.8.4</b> Design-o-metrix</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="glm.html"><a href="glm.html"><i class="fa fa-check"></i><b>6</b> Generalized Linear Models</a><ul>
<li class="chapter" data-level="6.1" data-path="glm.html"><a href="glm.html#elements-glm"><i class="fa fa-check"></i><b>6.1</b> Elements of Generalized Linear Models</a><ul>
<li class="chapter" data-level="6.1.1" data-path="glm.html"><a href="glm.html#relinking-linearity"><i class="fa fa-check"></i><b>6.1.1</b> Re-linking linearity</a></li>
<li class="chapter" data-level="6.1.2" data-path="glm.html"><a href="glm.html#choosing-randomness"><i class="fa fa-check"></i><b>6.1.2</b> Choosing patterns of randomness</a></li>
<li class="chapter" data-level="6.1.3" data-path="glm.html"><a href="glm.html#mean-var-rel"><i class="fa fa-check"></i><b>6.1.3</b> Mean-variance relationship</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="glm.html"><a href="glm.html#count-data"><i class="fa fa-check"></i><b>6.2</b> Count data</a><ul>
<li class="chapter" data-level="6.2.1" data-path="glm.html"><a href="glm.html#pois-reg"><i class="fa fa-check"></i><b>6.2.1</b> Poisson regression</a></li>
<li class="chapter" data-level="6.2.2" data-path="glm.html"><a href="glm.html#logistic-aka-binomial-regression-logistic-reg"><i class="fa fa-check"></i><b>6.2.2</b> Logistic (aka Binomial) regression (#logistic-reg)</a></li>
<li class="chapter" data-level="6.2.3" data-path="glm.html"><a href="glm.html#overdispersion"><i class="fa fa-check"></i><b>6.2.3</b> Modelling overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="glm.html"><a href="glm.html#duration-measures"><i class="fa fa-check"></i><b>6.3</b> Duration measures</a><ul>
<li class="chapter" data-level="6.3.1" data-path="glm.html"><a href="glm.html#exp-gam-reg"><i class="fa fa-check"></i><b>6.3.1</b> Exponential and Gamma regression</a></li>
<li class="chapter" data-level="6.3.2" data-path="glm.html"><a href="glm.html#exgauss-reg"><i class="fa fa-check"></i><b>6.3.2</b> ExGaussian regression</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="glm.html"><a href="glm.html#rating-scales"><i class="fa fa-check"></i><b>6.4</b> Rating scales</a><ul>
<li class="chapter" data-level="6.4.1" data-path="glm.html"><a href="glm.html#ord-logist-reg"><i class="fa fa-check"></i><b>6.4.1</b> Ordered logistic regression</a></li>
<li class="chapter" data-level="6.4.2" data-path="glm.html"><a href="glm.html#beta-reg"><i class="fa fa-check"></i><b>6.4.2</b> Beta regression</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="glm.html"><a href="glm.html#distributional-models"><i class="fa fa-check"></i><b>6.5</b> Beyond mean: distributional models</a><ul>
<li class="chapter" data-level="6.5.1" data-path="glm.html"><a href="glm.html#item-anchoring"><i class="fa fa-check"></i><b>6.5.1</b> Item-level anchoring in rating scales</a></li>
<li class="chapter" data-level="6.5.2" data-path="glm.html"><a href="glm.html#part-employment"><i class="fa fa-check"></i><b>6.5.2</b> Participant-level employment of scale</a></li>
<li class="chapter" data-level="6.5.3" data-path="glm.html"><a href="glm.html#part-skew"><i class="fa fa-check"></i><b>6.5.3</b> Participant-level skew in reaction times</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="wwm.html"><a href="wwm.html"><i class="fa fa-check"></i><b>7</b> Working with models</a><ul>
<li class="chapter" data-level="7.1" data-path="wwm.html"><a href="wwm.html#model-criticism"><i class="fa fa-check"></i><b>7.1</b> Model criticism</a><ul>
<li class="chapter" data-level="7.1.1" data-path="wwm.html"><a href="wwm.html#residual-analysis"><i class="fa fa-check"></i><b>7.1.1</b> Residual analysis</a></li>
<li class="chapter" data-level="7.1.2" data-path="wwm.html"><a href="wwm.html#fitted-responses"><i class="fa fa-check"></i><b>7.1.2</b> Fitted responses analysis</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="wwm.html"><a href="wwm.html#model-selection"><i class="fa fa-check"></i><b>7.2</b> Model selection</a><ul>
<li class="chapter" data-level="7.2.1" data-path="wwm.html"><a href="wwm.html#overfitting"><i class="fa fa-check"></i><b>7.2.1</b> The problem of over-fitting</a></li>
<li class="chapter" data-level="7.2.2" data-path="wwm.html"><a href="wwm.html#cross-validation"><i class="fa fa-check"></i><b>7.2.2</b> Cross validation and LOO</a></li>
<li class="chapter" data-level="7.2.3" data-path="wwm.html"><a href="wwm.html#ic"><i class="fa fa-check"></i><b>7.2.3</b> Information Criteria</a></li>
<li class="chapter" data-level="7.2.4" data-path="wwm.html"><a href="wwm.html#choose-dist"><i class="fa fa-check"></i><b>7.2.4</b> Choosing response distributions</a></li>
<li class="chapter" data-level="7.2.5" data-path="wwm.html"><a href="wwm.html#testing-theories"><i class="fa fa-check"></i><b>7.2.5</b> Testing theories</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="reflections.html"><a href="reflections.html"><i class="fa fa-check"></i><b>8</b> Reflections</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="cases.html"><a href="cases.html"><i class="fa fa-check"></i><b>A</b> Cases</a><ul>
<li class="chapter" data-level="A.1" data-path="cases.html"><a href="cases.html#real-cases"><i class="fa fa-check"></i><b>A.1</b> Real cases</a><ul>
<li class="chapter" data-level="A.1.1" data-path="cases.html"><a href="cases.html#cue8"><i class="fa fa-check"></i><b>A.1.1</b> CUE8</a></li>
<li class="chapter" data-level="A.1.2" data-path="cases.html"><a href="cases.html#uncanny"><i class="fa fa-check"></i><b>A.1.2</b> Uncanny Valley</a></li>
<li class="chapter" data-level="A.1.3" data-path="cases.html"><a href="cases.html#ipump"><i class="fa fa-check"></i><b>A.1.3</b> IPump</a></li>
<li class="chapter" data-level="A.1.4" data-path="cases.html"><a href="cases.html#sleepstudy"><i class="fa fa-check"></i><b>A.1.4</b> Case Sleepstudy</a></li>
<li class="chapter" data-level="A.1.5" data-path="cases.html"><a href="cases.html#egan"><i class="fa fa-check"></i><b>A.1.5</b> Egan</a></li>
<li class="chapter" data-level="A.1.6" data-path="cases.html"><a href="cases.html#mmn"><i class="fa fa-check"></i><b>A.1.6</b> Case: Millers Magic Number</a></li>
<li class="chapter" data-level="A.1.7" data-path="cases.html"><a href="cases.html#aup"><i class="fa fa-check"></i><b>A.1.7</b> AUP</a></li>
<li class="chapter" data-level="A.1.8" data-path="cases.html"><a href="cases.html#hugme"><i class="fa fa-check"></i><b>A.1.8</b> Hugme</a></li>
</ul></li>
<li class="chapter" data-level="A.2" data-path="cases.html"><a href="cases.html#synthetic-data-sets"><i class="fa fa-check"></i><b>A.2</b> Synthetic data sets</a><ul>
<li class="chapter" data-level="A.2.1" data-path="cases.html"><a href="cases.html#rainfall"><i class="fa fa-check"></i><b>A.2.1</b> Rainfall</a></li>
<li class="chapter" data-level="A.2.2" data-path="cases.html"><a href="cases.html#sec99"><i class="fa fa-check"></i><b>A.2.2</b> 99 seconds</a></li>
<li class="chapter" data-level="A.2.3" data-path="cases.html"><a href="cases.html#rational"><i class="fa fa-check"></i><b>A.2.3</b> Rational</a></li>
<li class="chapter" data-level="A.2.4" data-path="cases.html"><a href="cases.html#browsingab"><i class="fa fa-check"></i><b>A.2.4</b> BrowsingAB</a></li>
<li class="chapter" data-level="A.2.5" data-path="cases.html"><a href="cases.html#headache"><i class="fa fa-check"></i><b>A.2.5</b> Headache</a></li>
<li class="chapter" data-level="A.2.6" data-path="cases.html"><a href="cases.html#reading"><i class="fa fa-check"></i><b>A.2.6</b> Reading time</a></li>
<li class="chapter" data-level="A.2.7" data-path="cases.html"><a href="cases.html#argame"><i class="fa fa-check"></i><b>A.2.7</b> AR_game</a></li>
<li class="chapter" data-level="A.2.8" data-path="cases.html"><a href="cases.html#sleep"><i class="fa fa-check"></i><b>A.2.8</b> Sleep</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">New statistics for the design researcher</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="glm" class="section level1">
<h1><span class="header-section-number">6</span> Generalized Linear Models</h1>
<p>In the preceding chapters we got acquainted with the linear model as an extremely flexible tool to represent dependencies between predictors and outcome variables. We saw how factors and covariates gracefully work together and how complex research designs can be captured by multiple random effects. It was all about specifying an appropriate (and often sophisticated) right-hand side of the regression formula, the predictor term. Little space has been dedicated to the outcome variables, except that sometimes we used log-transformed outcomes to accomodate the Gaussian error term. That is now going to change, and we will start by examining the assumptions that are associated with the outcome variable.</p>
<p>Have you wondered about why I have been using so many simulated data sets up to this point? The reason for using simulated data is: the linear model, as introduced so far, makes assumptions that are <em>never</em> truly met by real data. The simulated data sets so far were meant to demonstrate some features found in real data sets, but generously wiped over some other frequent peculiarities.</p>
<p>Another question is probably lurking in the minds of readers with some classic statistics training: what has happened to the assumptions of ANOVA and the like, and where are all the neat tests that check for Normality, constant variance and such? The Gaussian linear model, which we used throughout <a href="#LM"><strong>??</strong></a> and <a href="#MLM"><strong>??</strong></a>, makes many assumptions, but in my view, the three crucial assumptions are:</p>
<ol style="list-style-type: decimal">
<li><em>Linearity</em> of the association between predictors and outcome variable.</li>
<li><em>Gaussian distribution</em> of responses</li>
<li><em>constant variance</em> of response distribution</li>
</ol>
<p>In the next section we will review these assumptions and lead them ad absurdum. Simply put, in the real world there is no such thing as Gaussian distribution and true linearity. Checking assumptions on a model that you know is inappropriate, seems a futile exercise, unless better alternatives are available, and that is the case: with <em>Generalized Linear Models</em> (GLMs) we extend our regression modelling framework once again, this time focussing on the responses and their shape of randomness.</p>
<p>As we will see, GLMs solves some common problems with linearity and gives us more choices on the shape of randomness. To say that once and for all: What GLMs do not do is relax the assumptions of linear models. And because I have met at least one seasoned researcher who divided the world of data into two categories, “parametric data”, that meets ANOVA assumptions, and “non-parametric data” that does not, let me get this perfectly straight: <em>data is neither parametric nor non-parametric</em>. Instead, data is the result of a process that distributes measures in some form and a good model aligns to this form. Second, a <em>model is parametric</em>, when the statistics it produces have a useful interpretations, like the intercept is the group mean of the reference group and the intercept random effect represents the variation between individuals. All models presented in this chapter (and this book) fulfill this requirement and <em>all are parametric</em>. There may be just one counter-example, which is polynomial regression <a href="#rollercoaster"><strong>??</strong></a>, which we used for its ability to render non-monotonic curves. The polynomial coefficients have no interpretation in terms of the cognitive processes leading to the Uncanny Valley. However, as we have seen in <a href="#test-statistic"><strong>??</strong></a>, they can easily be used to derive meaningful parameters, such as teh positions of shoulder and trough. A clear example of a non-parametric method is teh Mann-Withney U-test, which operates on sums of ranks, which typically has no useful interpretation.</p>
<p>The GLM framework rests on two extensions that bring us a huge step closer to our valuable data. The first one is a minor mathematical trick to establish linearity, the <em>link function</em>. The second is the expected <em>shape of randomness</em>. After we established the GLM framework <a href="glm.html#elements-glm">6.1</a>, I will introduce a good dozen of model classes, that leaves little reason to crudely approximate with Gaussian distributions, data transformations, or unintelligible non-parametric procedures. There almost always is a clear choice right at the beginning that largely depends on the properties of the response variable, for example:</p>
<ul>
<li><em>Poisson LM</em> is the first choice for outcome variables that are counted (with no upper limit), like number of errors.<br />
</li>
<li><em>Binomial (aka logistic) LM</em> covers the case of successful task completion, where counts have an upper boundary.</li>
</ul>
<p>These two GLM families have been around for more than half a century. The quest for a good model for reaction time and time-on-task was more difficult, as there does not seem to be a generally accepted default. Luckily, with recent developments in Bayesian regression engines the choice of random distributions has become much broader. For RT and ToT, I will suggest exponentially-modified Gaussian <em>(ExGauss)</em> models or, to some extent, <em>Gamma</em> models. For rating scales, where responses fall into a few ordered categories, <em>ordinal logistic regression</em> is a generally accepted approach. For (quasi)continuous rating scales will I make a novel suggestion, the <em>Beta LM</em>.</p>
<p>Too many choices can be a burden, but as we will see, most of the time the appropriate model family is obvious. For the impatient readers, here is the recipe: Answer the following three questions about the outcome variable. And, to make it even easier, it is always safe to answer Yes to the third question.</p>
<ol style="list-style-type: decimal">
<li>Is the outcome variable discrete or continuous?</li>
<li>What are the lower and upper boundaries of outcome measures?</li>
<li>Can we expect over-dispersion?</li>
</ol>
<p>Then use the graph below to identify the correct distribution family and jump to the respective section.</p>
<div class="figure">
<img src="Illustrations/GLM_decision_chart.png" alt="" />
<p class="caption">GLM decision chart</p>
</div>
<div id="elements-glm" class="section level2">
<h2><span class="header-section-number">6.1</span> Elements of Generalized Linear Models</h2>
<p>GLM is a <em>framework for modelling</em> that produces a <em>family of models</em>. Every member of this family uses specific <em>link functions</em> to establish linearity and chooses a particular <em>random distribution</em>, that has an adequate shape and mean-variance relationship.</p>
<p>Sometimes GLMs are mistaken as a way to relax assumptions of linear models, (or even called non-parametric). They are definitely not! Every member makes precise assumptions on the level of measurement and the shape of randomness (see Table A)<!-- #108 -->. One can even argue that Poisson, Binomial and exponential regression are stricter than Gaussian, as they use only one parameter, with the consequence of a tight association between variance and mean. A few members of GLM are classic: Poisson, Binomial (aka logistic) and exponential regression have routinely been used before they were united under the hood of GLM. These and a few others are called <em>canonical</em> GLMs, as they possess some convenient mathematical properties, that made efficient estimation possible, back in the days of expensive computer time.</p>
<p>For a first understanding of Generalized Linear Models, you should know that linear models are one family of Generalized Linear Models, which we call a Gaussian linear model. The three crucial assumptions of Gaussian linear models are encoded in the model formula:</p>
<p><span class="math display">\[
\begin{aligned}
\mu_i &amp;=\beta_0+ \beta_1 x_{1i}+ \dots +\beta_k x_{ki}\\ 
y_i &amp;\sim \textrm{Gaus}(\mu_i,\sigma)
\end{aligned}
\]</span></p>
<p>The first term, we call the likelihood and it represents the systematic quantitative relations we expect to find in the data. When it is a sum of products, like above, we call it linear. <em>Linearity</em> is a frequently under-regarded assumption of linear models and it is doomed to fail in <a href="glm.html#relinking-linearity">6.1.1</a>. The second term defines the pattern of randomness and it hosts two further assumptions: <em>Gaussian distribution</em> and <em>constant error variance</em> of the random component.</p>
<p>In classic statistics education, the tendency is still to present these assumptions as preconditions for a successful ANOVA or linear regression. The very term <em>pre</em>condition suggest, that they need to be checked upfront and the classic statisticians are used to deploy a zoo of null hypothesis tests for this purpose, although it is widely held among statisticians that this practice is illogical. If an assumptions seems to be violated, let’s say Normality, researchers then often turn to non-parametric tests. Many also just continue with ANOVA and add some shameful statements to the discussion of results or humbly cite one research paper that claims ANOVAs robustness to violations.</p>
<p>The parameters of a polynomial model usually don’t have a direct interpretation. However, we saw that useful parameters, such as the minimum of the curve, can be derived. Therefore, polynomial models can be called <em>semiparametric</em>, at least. As an example for a <em>non-parametric</em> test, the Mann-Whitney <em>U</em> statistic is composed of the number of times observations in group A are larger than in group B. The resulting sum <em>U</em> usually bears little relation to any real world process or question. Strictly speaking, the label non-parametric has nothing to do with ANOVA assumptions. It refers to the usefulness of parameters. A research problem, where <em>U</em> as the sum of wins has a useful interpretation could be that in some dueling disciplines, such as Fencing, team competitions are constructed by letting every athlete from a team duel every member of the opponent team. We could call the <em>U</em>-test parametric, and perhaps, the group means turn out to be meaningless.</p>
<div id="relinking-linearity" class="section level3">
<h3><span class="header-section-number">6.1.1</span> Re-linking linearity</h3>
<p>In the chapter on Linear Models, we encountered several situations where linearity itself was violated:</p>
<ul>
<li>in <a href="lm.html#ofm">4.3.5</a> we used ordered factors to estimate a learning curve</li>
<li>in <a href="lm.html#saturation">4.5.3</a>, we used conditional effects when two or more interventions improve the same process, e.g. visual recognition of letters</li>
<li>and in <a href="#plm"><strong>??</strong></a> we used polynomials to estimate wildly curved relationships</li>
</ul>
<p>The case of Polynomial regression is special in two ways: first, the curvature itself is of theoretical interest (e.g. finding the “trough” of the Uncanny Valley efect). Second, a polynomial curve (of second degree or more) is no longer monotonously increasing (or decreasing). In contrast, learning curves and saturation effects have in common that in both situations performance steadily increases when we add more to the predictor side. There is a limit to performance, which is reached asymptotically, but there always is a monotonous trend.</p>
<div class="sourceCode" id="cb541"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb541-1"><a href="glm.html#cb541-1"></a><span class="kw">attach</span>(IPump)</span>
<span id="cb541-2"><a href="glm.html#cb541-2"></a></span>
<span id="cb541-3"><a href="glm.html#cb541-3"></a>D_Novel <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb541-4"><a href="glm.html#cb541-4"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span> session, <span class="dt">y =</span> deviations)) <span class="op">+</span></span>
<span id="cb541-5"><a href="glm.html#cb541-5"></a><span class="co">#  geom_jitter() +</span></span>
<span id="cb541-6"><a href="glm.html#cb541-6"></a><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="dt">se =</span> F)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-2-1.png" width="90%" /></p>
<p>We used an OFM with stairways coding to account for this non-linearity, but that has one disadvantage. From a practical perspective it would interesting to know, how performance improves when practice continues. What would be performance in (hypothetical) sessions 4, 5 and 10. Because the OFM just makes up one estimate for every level, there is no way to get predictions beyond the observed range.</p>
<p>With an LRM, the slope parameter applies to all steps, which gives us the possibility of deriving predictions beyond the observed range. To demonstrate this on the deviations from optimal path, the following code estimates a plain LRM and then injects some new (virtual) data to get predictions beyond the observed range:</p>
<div class="sourceCode" id="cb542"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb542-1"><a href="glm.html#cb542-1"></a>M_LRM_<span class="dv">1</span> &lt;-<span class="st"> </span><span class="kw">stan_glm</span>(deviations <span class="op">~</span><span class="st"> </span><span class="dv">1</span> <span class="op">+</span><span class="st"> </span>session, <span class="dt">data =</span> D_Novel)</span></code></pre></div>
<div class="sourceCode" id="cb543"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb543-1"><a href="glm.html#cb543-1"></a>D_new &lt;-<span class="st"> </span></span>
<span id="cb543-2"><a href="glm.html#cb543-2"></a><span class="st">  </span><span class="kw">tibble</span>(<span class="dt">session =</span> <span class="kw">as.integer</span>(<span class="kw">c</span>(<span class="dv">0</span><span class="op">:</span><span class="dv">9</span>)),</span>
<span id="cb543-3"><a href="glm.html#cb543-3"></a>         <span class="dt">range =</span> <span class="kw">if_else</span>(session <span class="op">&lt;</span><span class="st"> </span><span class="dv">3</span>, <span class="st">&quot;observed&quot;</span>, <span class="st">&quot;beyond obs.&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb543-4"><a href="glm.html#cb543-4"></a><span class="st">  </span><span class="kw">as_tbl_obs</span>()</span>
<span id="cb543-5"><a href="glm.html#cb543-5"></a></span>
<span id="cb543-6"><a href="glm.html#cb543-6"></a><span class="kw">predict</span>(M_LRM_<span class="dv">1</span>, <span class="dt">newdata =</span> D_new) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb543-7"><a href="glm.html#cb543-7"></a><span class="st">  </span><span class="kw">print</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb543-8"><a href="glm.html#cb543-8"></a><span class="st">  </span><span class="kw">left_join</span>(D_new) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb543-9"><a href="glm.html#cb543-9"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> session, <span class="dt">y =</span> center, <span class="dt">ymin =</span> lower, <span class="dt">ymax =</span> upper)) <span class="op">+</span></span>
<span id="cb543-10"><a href="glm.html#cb543-10"></a><span class="st">  </span><span class="kw">geom_ribbon</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> range), <span class="dt">alpha =</span> <span class="fl">.2</span>) <span class="op">+</span></span>
<span id="cb543-11"><a href="glm.html#cb543-11"></a><span class="st">  </span><span class="kw">geom_step</span>() <span class="op">+</span></span>
<span id="cb543-12"><a href="glm.html#cb543-12"></a><span class="st">  </span><span class="kw">geom_hline</span>(<span class="kw">aes</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">linetype =</span> <span class="dv">2</span>) <span class="op">+</span></span>
<span id="cb543-13"><a href="glm.html#cb543-13"></a><span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> <span class="dv">0</span><span class="op">:</span><span class="dv">10</span>)</span></code></pre></div>
<pre><code>## **  10 predictions (scale: resp) with 95% credibility limits (five shown below) 
##   Obs center lower upper
## 1   1  0.782 -1.37  2.94
## 2   2  0.627 -1.54  2.74
## 3   5  0.137 -2.04  2.33
## 4   8 -0.316 -2.56  1.91
## 5  10 -0.680 -3.04  1.68</code></pre>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-5-1.png" width="90%" /></p>
<div class="sourceCode" id="cb545"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb545-1"><a href="glm.html#cb545-1"></a><span class="kw">detach</span>(IPump)</span></code></pre></div>
<p>When we use a linear model when there truly is an asymptotic curve, negative values are produced, which are impossible. As the graph shows, this is most pronounced the further we move away from the observed range. But, it also effects the observed levels, as the lower credibility limits are negative, too.</p>
<p>Such a non-linearity happens to all outcome variables that have natural lower or upper boundaries, and that includes all outcome variables in the universe, except its very own spatial extension, perhaps. Speed of light can only be approached asymptotically (if you have some mass to carry around) and even the darkest intergalactic spaces are lit by cosmic background radiation and therefore have a temperature just slightly above absolute zero. It is an unescapable truth, that all our ephemeral measures in design research have boundaries and therefore suffer from wrong predictions:</p>
<ul>
<li>Errors and other countable incidences are bounded at zero</li>
<li>Rating scales are bounded at the lower and upper extreme item</li>
<li>Task completion has a lower bound of zero and upper bound is the number of tasks.</li>
<li>Temporal measures formally have lower bound of zero, but psychologically, the lower bound always is a positive number.</li>
</ul>
<p>The bare linear term is always inadequate, which is unfortunate, because its strength is the endless versatility in specifying relations between predictor variables and outcome. How can we escape the problem of boundaries, without sacrificing such a useful tool?
Generalized linear models use a simple mathematical trick that keeps linear terms, but confines the fitted responses to the natural boundaries of the measures. In linear models, the linear term <span class="math inline">\(\mu\)</span> is mapped directly to fitted responses: <span class="math inline">\(\mu_i = \beta_0 + x_1\beta_1\)</span>. In GLMs, a layer is drawn between the fitted response <span class="math inline">\(\mu\)</span> and the linear term, <em>linear predictor <span class="math inline">\(\theta\)</span></em>. The <em>link function</em> <span class="math inline">\(\phi\)</span> transforms between <span class="math inline">\(\mu\)</span> and <span class="math inline">\(\theta\)</span>. In order to transform back to the scale of measurement, the inverse of the link function, the <em>mean function</em> is used.</p>
<p><span class="math display">\[
\begin{aligned}
\theta_i &amp;= \beta_0 + x_1 \beta_1\\ 
\mu_i &amp;= \phi( \theta_i)
\end{aligned}
\]</span></p>
<p>In arithmetics an abundance of functions exists for every possible purpose. However, a link function <span class="math inline">\(\phi\)</span> must fulfill two criteria:</p>
<ol style="list-style-type: decimal">
<li>mapping from the (linear) range <span class="math inline">\([-\infty; \infty]\)</span> to the range of the response, e.g. <span class="math inline">\([0; \infty]\)</span>.</li>
<li>be monotonically increasing</li>
</ol>
<p>Intuitively speaking, a monotonically increasing function always preserves the order, such that the following holds for a link function.</p>
<p><span class="math display">\[
\theta_i &gt; \theta_j \rightarrow \phi(\theta_i) &gt; \phi(\theta_j) \rightarrow \mu_i &gt; \mu_j
\]</span></p>
<p>One reason for monotonicity is that for a link function <span class="math inline">\(\phi\)</span> there must exist the inverse, which is called the mean function (<span class="math inline">\(\phi^{-1}\)</span>). For example, <span class="math inline">\(x^2\)</span> is not a proper link function, because its inverse, <span class="math inline">\(\sqrt{x}\)</span> can take <em>two</em> values (e.g. <span class="math inline">\(\sqrt{x} = [2, -2]\)</span>) and therefore is not a function on the desired range, strictly speaking.</p>
<p>The most typical cases are that there is a lower boundary of zero, or there are two boundaries. An adequate link function for count variables would map the range of natural numbers (only lower bound) to the <em>linear range</em> of <span class="math inline">\(\eta\)</span> that is <span class="math inline">\([-\infty; \infty]\)</span>. The <em>logarithm</em> is such a function and its inverse is the <em>exponential</em> function, which bends the linear range back into the boundary. Other variables, like success rates or rating scales, have lower and upper boundaries. A suitable pair of functions is the <em>logit</em> link function and the <em>logistic</em> mean function.</p>
<div class="sourceCode" id="cb546"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb546-1"><a href="glm.html#cb546-1"></a>plot_glmfun &lt;-<span class="st"> </span><span class="cf">function</span>(<span class="dt">f =</span> log,</span>
<span id="cb546-2"><a href="glm.html#cb546-2"></a>                        <span class="dt">title =</span> <span class="st">&quot;log link function&quot;</span>,</span>
<span id="cb546-3"><a href="glm.html#cb546-3"></a>                        <span class="dt">lower =</span> <span class="fl">.01</span>, <span class="dt">upper =</span> <span class="dv">3</span>,</span>
<span id="cb546-4"><a href="glm.html#cb546-4"></a>                        <span class="dt">dir =</span> <span class="st">&quot;link&quot;</span>){</span>
<span id="cb546-5"><a href="glm.html#cb546-5"></a>  out &lt;-<span class="st">  </span></span>
<span id="cb546-6"><a href="glm.html#cb546-6"></a><span class="st">    </span><span class="kw">tibble</span>(<span class="dt">x =</span> <span class="kw">seq</span>(lower, upper, (upper <span class="op">-</span><span class="st"> </span>lower)<span class="op">/</span><span class="dv">100</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb546-7"><a href="glm.html#cb546-7"></a><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(x)) <span class="op">+</span></span>
<span id="cb546-8"><a href="glm.html#cb546-8"></a><span class="st">    </span><span class="kw">stat_function</span>(<span class="dt">fun =</span> f) <span class="op">+</span></span>
<span id="cb546-9"><a href="glm.html#cb546-9"></a><span class="st">    </span><span class="kw">labs</span>(<span class="dt">title =</span> title) <span class="op">+</span></span>
<span id="cb546-10"><a href="glm.html#cb546-10"></a><span class="st">    </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="kw">expression</span>(mu), <span class="dt">y =</span> <span class="kw">expression</span>(eta))</span>
<span id="cb546-11"><a href="glm.html#cb546-11"></a>  <span class="cf">if</span>(dir <span class="op">==</span><span class="st"> &quot;mean&quot;</span>) out &lt;-<span class="st"> </span>out <span class="op">+</span><span class="st"> </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="kw">expression</span>(eta), </span>
<span id="cb546-12"><a href="glm.html#cb546-12"></a>                                      <span class="dt">y =</span> <span class="kw">expression</span>(mu))</span>
<span id="cb546-13"><a href="glm.html#cb546-13"></a>  out</span>
<span id="cb546-14"><a href="glm.html#cb546-14"></a>}</span>
<span id="cb546-15"><a href="glm.html#cb546-15"></a></span>
<span id="cb546-16"><a href="glm.html#cb546-16"></a>gridExtra<span class="op">::</span><span class="kw">grid.arrange</span>(</span>
<span id="cb546-17"><a href="glm.html#cb546-17"></a>  <span class="kw">plot_glmfun</span>(),</span>
<span id="cb546-18"><a href="glm.html#cb546-18"></a>  <span class="kw">plot_glmfun</span>(<span class="dt">f =</span> exp, <span class="st">&quot;exponential mean function&quot;</span>,  <span class="fl">-3.5</span>, <span class="dv">1</span>, <span class="dt">dir =</span> <span class="st">&quot;mean&quot;</span>),</span>
<span id="cb546-19"><a href="glm.html#cb546-19"></a>  <span class="kw">plot_glmfun</span>(<span class="dt">f =</span> logit, <span class="st">&quot;logit link function&quot;</span>, <span class="fl">0.01</span>, <span class="fl">.99</span>),</span>
<span id="cb546-20"><a href="glm.html#cb546-20"></a>  <span class="kw">plot_glmfun</span>(<span class="dt">f =</span> inv_logit, <span class="st">&quot;logistic mean function&quot;</span>,  <span class="dv">-5</span>, <span class="dv">5</span>, <span class="dt">dir =</span> <span class="st">&quot;mean&quot;</span>))</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-7-1.png" width="90%" /></p>
<p>Using the link function comes at a cost: the linear coefficients <span class="math inline">\(\beta_i\)</span> is losing its interpretation as increment-per-unit and no longer has a natural interpretation. Later we will see that logarithmic and logit scales gain an intuitive interpretation when parameters are exponentiated, <span class="math inline">\(\textrm{exp}(\beta_i)\)</span> (<a href="#poisson-reg"><strong>??</strong></a> and <a href="#logistic-reg"><strong>??</strong></a></p>
<!--
Who needs a well-defined link between observations and fitted responses? Applied design researchers do when predictions are their business. Reconsider the Sec99 case study, where the population mean of ToT was predicted from a sample. This is a statement on the status quo, and therefore it is not a prediction, in a strict sense. Comparing designs A and B is status quo, too, as both designs do exist. Generally, we  classify all research questions as *status quo* that stay confined in the range of $y_i$. In the IPump study, however, it is compelling to ask: "how will the nurse perform in session 4?". "When will she reach error-free operation?". These are predictive questions, where the question reaches into the unknown and link functions become utterly important.
-->
</div>
<div id="choosing-randomness" class="section level3">
<h3><span class="header-section-number">6.1.2</span> Choosing patterns of randomness</h3>
<p>The second term of a linear model, <span class="math inline">\(y_i \sim Norm(\mu_i, \sigma)\)</span> states that the observed values are drawn from Gaussian distributions(see @ref(resid_normality)). But Gaussian distributions have the same problem as the linearity assumption: the range is <span class="math inline">\([-\infty; \infty]\)</span>. in fact, a Gaussian distribution can only be a reasonable approximation when the measures are far off the boundaries of measures and the error is much smaller than the predicted values (@ref(normal_distributions)).</p>
<p>The problem can be demonstrated by simulating observations, using a Gaussian pattern of randomness, and see how this fails to produce realistic data. Imagine a study comparing a novel and a legacy interface design for medical infusion pumps. The researchers let trained nurses perform a single task on both devices and count the errors. Assuming, the average number of errors per tasks is <span class="math inline">\(\mu_L = 3\)</span> for the legacy device and <span class="math inline">\(\mu_N = 1.2\)</span> for the novel device, with standard deviation of <span class="math inline">\(\sigma = .8\)</span>. We can simulate a basic data set as:</p>
<div class="sourceCode" id="cb547"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb547-1"><a href="glm.html#cb547-1"></a>N =<span class="st"> </span><span class="dv">80</span></span>
<span id="cb547-2"><a href="glm.html#cb547-2"></a>D_pumps_sim &lt;-<span class="st"> </span></span>
<span id="cb547-3"><a href="glm.html#cb547-3"></a><span class="st">  </span><span class="kw">tibble</span>(<span class="dt">Design =</span> <span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;L&quot;</span>, <span class="st">&quot;N&quot;</span>), N<span class="op">/</span><span class="dv">2</span>),</span>
<span id="cb547-4"><a href="glm.html#cb547-4"></a>             <span class="dt">mu =</span> <span class="kw">if_else</span>(Design <span class="op">==</span><span class="st"> &quot;L&quot;</span>, <span class="dv">3</span>, <span class="fl">1.2</span>),</span>
<span id="cb547-5"><a href="glm.html#cb547-5"></a>             <span class="dt">errors =</span> <span class="kw">rnorm</span>(N, mu, <span class="dt">sd =</span> <span class="dv">1</span>))</span></code></pre></div>
<p>We illustrate the data set using histograms:</p>
<div class="sourceCode" id="cb548"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb548-1"><a href="glm.html#cb548-1"></a>D_pumps_sim <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb548-2"><a href="glm.html#cb548-2"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> errors)) <span class="op">+</span></span>
<span id="cb548-3"><a href="glm.html#cb548-3"></a><span class="st">  </span><span class="kw">facet_grid</span>(<span class="op">~</span>Design) <span class="op">+</span></span>
<span id="cb548-4"><a href="glm.html#cb548-4"></a><span class="st">  </span><span class="kw">geom_histogram</span>(<span class="dt">bins =</span> <span class="dv">20</span>) <span class="op">+</span></span>
<span id="cb548-5"><a href="glm.html#cb548-5"></a><span class="st">  </span><span class="kw">geom_vline</span>(<span class="dt">col =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">xintercept =</span> <span class="dv">0</span>) <span class="op">+</span></span>
<span id="cb548-6"><a href="glm.html#cb548-6"></a><span class="st">  </span><span class="kw">coord_flip</span>()</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-9-1.png" width="90%" /></p>
<p>We immediatly see, that simulation with Gaussian distributions is rather inappropriate: a substantial number of simulated observations is <em>negative</em>, which strictly makes no sense for error counts. The pragmatic and impatient reader may suggest to adjust the standard deviation (or move the averages up) to make negative values less unlikely. That would be a poor solution as Gaussian distributions support the full range of real numbers, no matter how small the variance is (but not zero). There is always a chance of negative simulations, as tiny as it may be. Repeatedly running the simulation until <code>pumps</code> contains exclusively positive numbers (and zero), would compromise the idea of random numbers itself. The second reason is that the simulations very purpose was to express and explore expectations from the linear model (CG). We can simply conclude that any model that assumes normally distributed errors must be wrong when the outcome is bounded below or above, which means: always.</p>
<p>Recall how linearity is gradually bent when a magnitude approaches its natural limit. A similar effect occurs for distributions. Distributions that respect a lower or upper limit get squeezed like chewing gum into a corner when approaching the boundaries. Review the sections on Binomial <a href="ebs.html#binomial-dist">3.4.2.3</a> and Poisson distributions <a href="ebs.html#poisson-dist">3.4.2.4</a> for illustrations. As a matter of fact, a lot of real data in design research is skewed that way, making Gaussian distributions a poor fit.</p>
<p>A common misconception is that random distributions approach the Gaussian distribution with larger sample sizes. But, what really happens only, is that increasing the number of observations renders the true distribution more clearly.</p>
<p>In chapter <a href="ebs.html#distributions">3.4.2</a> a number of random distributions were introduced, together with conditions of when they arise. The major criteria were related to properties of the outcome measure: how it is bounded and whether it is discrete (countable) or continuous.
Generalized Linear Models give the researcher a larger choice for modelling the random component and the following table lists some common candidates.</p>
<table>
<thead>
<tr class="header">
<th align="left">boundaries</th>
<th align="left">discrete</th>
<th align="left">continuous</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">unbounded</td>
<td align="left">NA</td>
<td align="left">Normal</td>
</tr>
<tr class="even">
<td align="left">lower</td>
<td align="left">Poisson</td>
<td align="left">Exponential</td>
</tr>
<tr class="odd">
<td align="left">lower and upper</td>
<td align="left">Binomial</td>
<td align="left">Beta</td>
</tr>
</tbody>
</table>
<p>That is not to say that these five are the only possible choices. Many dozens of statistical distributions are known and these five are just making the least assumptions on the shape of randomness in their class (mathematicians call this <em>maximum entropy distributions</em>). In fact, we will soon discover that real data frequently violates principles of these distributions. For example, count measures in behavioural research typically show a variance that exceeds the mean, which speaks against the Poisson distributions. As we will see in <a href="glm.html#overdispersion">6.2.3</a> and <a href="#zero-inflation"><strong>??</strong></a>, Poisson distribution can still be used in such cases with some additional tweaks borrowed from multi-level modelling (observation-level random effects).</p>
<p>As we will see, response times in design research are particularly misbehaved, as they do not have their lower boundary at zero, but at the lowest human possible time to solve the task. In contrast, most continuous distributions assume that measures near zero are at least possible. In case of response times, we will take advantage of the fact, that modern Bayesian estimation engines support a larger range of distributions than ever seen before (i.e., in classic statistics). The <code>stan_glm</code> regression engine has been designed with downwards compatibility in mind, which is why it includes fewer distributions. Luckily, there is a sibling engine in the package <code>brms</code>, which is more progressive and gives many more choices, such as the Exponential-Gaussian distribution.</p>
</div>
<div id="mean-var-rel" class="section level3">
<h3><span class="header-section-number">6.1.3</span> Mean-variance relationship</h3>
<p>The third assumption of linear models is rooted in the random component term as well. Recall, that there is just one parameter <span class="math inline">\(\sigma\)</span> for the dispersion of randomness and that any Gaussian distribution’s dispersion is exclusively determined by <span class="math inline">\(\sigma\)</span>. That is more harmful as it may sound. In most real data, the dispersion of randomness depends on the location, as can be illustrated by the following simulation.</p>
<p>Imagine a survey on commuter behaviour that asks the following questions:</p>
<ol style="list-style-type: decimal">
<li>How long is the route?</li>
<li>How long does it <em>typically</em> take?</li>
<li>What are the maximum and minimum travel times you remember?</li>
</ol>
<p>If we simulate such data from a linear model, the relationship between length of route and travel time would look like a evenly wide band, which is due to the constant variance:</p>
<div class="sourceCode" id="cb549"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb549-1"><a href="glm.html#cb549-1"></a>N =<span class="st"> </span><span class="dv">100</span></span>
<span id="cb549-2"><a href="glm.html#cb549-2"></a><span class="kw">tibble</span>(<span class="dt">Obs =</span> <span class="kw">as.factor</span>(<span class="dv">1</span><span class="op">:</span>N),</span>
<span id="cb549-3"><a href="glm.html#cb549-3"></a>           <span class="dt">km  =</span> <span class="kw">runif</span>(N, <span class="dv">2</span>, <span class="dv">40</span>),</span>
<span id="cb549-4"><a href="glm.html#cb549-4"></a>           <span class="dt">min =</span> <span class="kw">rnorm</span>(N, km <span class="op">*</span><span class="st"> </span><span class="dv">2</span>, <span class="dv">10</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb549-5"><a href="glm.html#cb549-5"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> km, <span class="dt">y =</span> min)) <span class="op">+</span></span>
<span id="cb549-6"><a href="glm.html#cb549-6"></a><span class="st">  </span><span class="kw">geom_point</span>() <span class="op">+</span></span>
<span id="cb549-7"><a href="glm.html#cb549-7"></a><span class="st">  </span><span class="kw">geom_quantile</span>(<span class="dt">quantiles =</span> <span class="kw">c</span>(.<span class="dv">25</span>, <span class="fl">.5</span>, <span class="fl">.75</span>))</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-11-1.png" width="90%" /></p>
<p>Again, we get some impossible negative data points, but what is also unrealistic is that persons who live right around the corner experience the same range of possible travel times than people who drive dozens of kilometers. Most of the time, we intuit the dispersion of randomness to increase with the magnitude.</p>
<p>Most other distributions do not have constant variance. For example, a Gamma distribution takes two parameters, shape <span class="math inline">\(\alpha\)</span> and scale <span class="math inline">\(\tau\)</span> and both of them influence mean and variance of the distribution, such that the variance increases by the mean by square:</p>
<p><span class="math display">\[
\begin{aligned}
Y &amp;\sim \textrm{Gamma}(\alpha, \theta)\\ 
E(Y) &amp;= \alpha \theta\\
\textrm{Var}(Y) &amp;= \alpha \theta^2\\
\textrm{Var}(Y) &amp;= E(Y) \theta
\end{aligned}
\]</span></p>
<div class="sourceCode" id="cb550"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb550-1"><a href="glm.html#cb550-1"></a><span class="kw">tibble</span>(<span class="dt">Obs =</span> <span class="kw">as.factor</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>),</span>
<span id="cb550-2"><a href="glm.html#cb550-2"></a>           <span class="dt">km  =</span> <span class="kw">runif</span>(<span class="dv">100</span>, <span class="dv">2</span>, <span class="dv">40</span>),</span>
<span id="cb550-3"><a href="glm.html#cb550-3"></a>           <span class="dt">min =</span> <span class="kw">rgamma</span>(<span class="dv">100</span>, <span class="dt">shape =</span> km <span class="op">*</span><span class="st"> </span><span class="fl">.5</span>, <span class="dt">scale =</span> <span class="dv">4</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb550-4"><a href="glm.html#cb550-4"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> km, <span class="dt">y =</span> min)) <span class="op">+</span></span>
<span id="cb550-5"><a href="glm.html#cb550-5"></a><span class="st">  </span><span class="kw">geom_point</span>() <span class="op">+</span></span>
<span id="cb550-6"><a href="glm.html#cb550-6"></a><span class="st">  </span><span class="kw">geom_quantile</span>(<span class="dt">quantiles =</span> <span class="kw">c</span>(.<span class="dv">25</span>, <span class="fl">.5</span>, <span class="fl">.75</span>))</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-12-1.png" width="90%" /></p>
<p>A similar situation arises for count data. When counting user errors, we would expect a larger variance for complex tasks and interfaces, e.g. writing an article in a word processor, as compared to the rather simple situation like operating a medical infusion pump. For count data, the Poisson distribution is often a starting point and for Poisson distributed variables, mean and variance are both exactly determined by the Poisson rate parameter <span class="math inline">\(\lambda\)</span>, and therefore linearly connected.</p>
<p><span class="math display">\[
\begin{aligned}
Y &amp;\sim \textrm{Poisson}(\lambda)\\
\textrm{Var}(Y) &amp;= E(Y) = \lambda
\end{aligned}
\]</span></p>
<div class="sourceCode" id="cb551"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb551-1"><a href="glm.html#cb551-1"></a><span class="kw">tibble</span>(<span class="dt">Obs =</span> <span class="kw">as.factor</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>),</span>
<span id="cb551-2"><a href="glm.html#cb551-2"></a>           <span class="dt">Task  =</span> <span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;article&quot;</span>, <span class="st">&quot;infusion&quot;</span>), <span class="dv">50</span>),</span>
<span id="cb551-3"><a href="glm.html#cb551-3"></a>           <span class="dt">errors =</span> <span class="kw">rpois</span>(<span class="dv">100</span>, <span class="dt">lambda =</span> <span class="kw">if_else</span>(Task <span class="op">==</span><span class="st"> &quot;article&quot;</span>, <span class="dv">200</span>, <span class="dv">8</span>))) <span class="op">%&gt;%</span></span>
<span id="cb551-4"><a href="glm.html#cb551-4"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Task, <span class="dt">y =</span> errors)) <span class="op">+</span></span>
<span id="cb551-5"><a href="glm.html#cb551-5"></a><span class="st">  </span><span class="kw">geom_boxplot</span>() <span class="op">+</span></span>
<span id="cb551-6"><a href="glm.html#cb551-6"></a><span class="st">  </span><span class="kw">geom_jitter</span>()</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-13-1.png" width="90%" /></p>
<p>Not by coincidence, practically all distributions with a lower boundary have variance increase with the mean. Distributions that have two boundaries, like binomial or beta distributions also have a mean-variance relationship, but a different one. For binomially distributed variables, mean and variance are determined as follows:</p>
<p><span class="math display">\[
\begin{aligned}
Y &amp;\sim \textrm{Binom}(p, k)\\
E(Y) &amp;= p k\\
\textrm{Var}(Y) &amp;= p (1-p) k\\
\textrm{Var}(Y) &amp;= E(Y)(1-p)
\end{aligned}
\]</span></p>
<p>To see this, imagine a study that examines the relationship between user expertise (for the convenience on a scale of zero to one) and success rate on ten tasks. The result is a cigar-like shape. For binomial distributions, variance gets largest, when the chance of success is centered at <span class="math inline">\(p = .5\)</span>. This is very similar for other distributions with two boundaries, such as beta and logit-Gaussian distributions.</p>
<div class="sourceCode" id="cb552"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb552-1"><a href="glm.html#cb552-1"></a><span class="kw">tibble</span>(<span class="dt">expertise  =</span> <span class="kw">runif</span>(<span class="dv">1000</span>, <span class="dv">0</span>, <span class="dv">1</span>),</span>
<span id="cb552-2"><a href="glm.html#cb552-2"></a>           <span class="dt">successes =</span> <span class="kw">rbinom</span>(<span class="dv">1000</span>, <span class="dv">25</span>, expertise)) <span class="op">%&gt;%</span></span>
<span id="cb552-3"><a href="glm.html#cb552-3"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> expertise, <span class="dt">y =</span> successes)) <span class="op">+</span></span>
<span id="cb552-4"><a href="glm.html#cb552-4"></a><span class="st">  </span><span class="kw">geom_point</span>()</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-14-1.png" width="90%" /></p>
<p>In conclusion, the Gaussian distribution assumption of constant variance is flawed in two aspects: real distributions are typically asymmetric and have mean and variance linked. Both phenomena are tightly linked to the presence of boundaries. Broadly, the deviation from symmetry gets worse when observations are close to the boundaries (e.g. low error rates), whereas differences in variance is more pronounced when the means are far apart from each other.</p>
<p>Still, using distributions that are not Gaussian sometimes carries minor complications. Gaussian distributions have the convenient property that the amount of randomness is directly expressed as the parameter <span class="math inline">\(\sigma\)</span>. That allowed us to compare the fit of two models A and B by comparing <span class="math inline">\(\sigma_A\)</span> and <span class="math inline">\(\sigma_B\)</span>. In random distributions with just one parameter, the variance of randomness is fixed by the location (e.g. Poisson <span class="math inline">\(\lambda\)</span> or Binomial <span class="math inline">\(p\)</span>). For distributions with more than one parameter, dispersion of randomness typically is a function of two or more parameters, as can be seen in teh formulas above. For example, Gamma distributions have two parameters, but these do not pull location and dispersion as neatly apart as Gaussian distributions do. Instead, mean and variance depend on both parameters for Gamma distributions.</p>
<p>Using distributions with entanglement of location and dispersion seems to be a step back, but frequently it is necessary to render a realistic association between location of fitted responses and amount of absolute randomness. Most distributions with a lower bound (e.g. Poisson, exponential and Gamma) increase variance with mean, whereas double bounded distributions (beta and binomial) typically have maximum variance when the distribution is centered and symmetric. For the researcher this all means that selecting a distribution class for a Generalized Linear Model, the choice determines the shape of randomness <em>and</em> the relation between location and variance.</p>
<p>The following sections are organized by type of typical outcome variable (counts, time intervals and rating scales). Each section (except rating scales) first introduces a one-parametric model (e.g. Poisson). A frequent problem with these models is that the location-variance relation is too strict. When errors are more widely dispersed than is allowed, this is called over-dispersion and one can either use a trick borrowed from multi-level models, observation-level random effects @(olre) or select a two-parametric distribution class (e.g., Negative-Binomial).</p>
</div>
</div>
<div id="count-data" class="section level2">
<h2><span class="header-section-number">6.2</span> Count data</h2>
<p>Gaussian distributions assume that the random variable under investigation is continuous. For measures, such as time, it is natural and it can be a reasonable approximation for measures with fine-grained steps, such as average scores of self-report scales with a large number of items. Other frequently used measures are clearly, i.e. naturally, discrete, in particular everything that is counted. Examples are: number of errors, number of succesfully completed tasks or the number of users. Naturally, count measures have a lower bound and frequently this is zero. A distinction has to be made, though, for the upper bound. In some cases, there is no well defined upper bound, or it is very large (e.g. number of users) and Poisson regression applies. In other cases, the upper bound is given by the research design, for example the number of tasks given to a user. When there is an upper bound, Binomial distributions apply, which is called logistic regression.</p>
<div id="pois-reg" class="section level3">
<h3><span class="header-section-number">6.2.1</span> Poisson regression</h3>
<p>If data can be considered successes in a fixed number of trials, logistic regression is the model type of choice. When the outcome variable is of type count, but there is no apparent upper limit, Poisson regression applies.</p>
<p>In brief, Poisson regression has the following attributes:</p>
<ol style="list-style-type: decimal">
<li>The outcome variable is bounded at zero (and that must be a possible outcome, indeed).</li>
<li>The linear predictor is on a logarithmic scale, with the exponential function being the inverse.</li>
<li>The random component follows a Poisson distribution.</li>
<li>Variance of randomness increases linearly with the mean.</li>
</ol>
<p>The link function is the logarithm, as it transforms from the non-negative range of numbers to real numbers. For a start, we have a look at a Poisson GMM. Recall the smart smurfer game from section <a href="ebs.html#poisson-dist">3.4.2.4</a>. Imagine that in an advanced level of the game <!-- #99 -->, items are well hidden from the player and therefore extremely difficult to catch. To compensate for the decreased visibility of items, the level carries an abundance them. In fact, the goal of the designers is that visibility and abundance are so carefully balanced that, on average, a player finds three items. We simulate a data set for one player repeating the level 30 times and run our first Poisson model, which is a plain GMM.</p>
<div class="sourceCode" id="cb553"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb553-1"><a href="glm.html#cb553-1"></a><span class="kw">set.seed</span>(<span class="dv">6</span>)</span>
<span id="cb553-2"><a href="glm.html#cb553-2"></a>D_Pois &lt;-<span class="st"> </span></span>
<span id="cb553-3"><a href="glm.html#cb553-3"></a><span class="st">  </span><span class="kw">tibble</span>(<span class="dt">Obs =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">30</span>,</span>
<span id="cb553-4"><a href="glm.html#cb553-4"></a>  <span class="dt">items_found =</span> <span class="kw">rpois</span>(<span class="dv">30</span>, <span class="dt">lambda =</span> <span class="fl">3.4</span>))</span>
<span id="cb553-5"><a href="glm.html#cb553-5"></a></span>
<span id="cb553-6"><a href="glm.html#cb553-6"></a>D_Pois <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb553-7"><a href="glm.html#cb553-7"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> items_found)) <span class="op">+</span></span>
<span id="cb553-8"><a href="glm.html#cb553-8"></a><span class="st">  </span><span class="kw">geom_histogram</span>()</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-16-1.png" width="90%" /></p>
<div class="sourceCode" id="cb554"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb554-1"><a href="glm.html#cb554-1"></a>M_Pois &lt;-<span class="st"> </span></span>
<span id="cb554-2"><a href="glm.html#cb554-2"></a><span class="st">  </span><span class="kw">stan_glm</span>(items_found <span class="op">~</span><span class="st"> </span><span class="dv">1</span>,</span>
<span id="cb554-3"><a href="glm.html#cb554-3"></a>           <span class="dt">family =</span> poisson,</span>
<span id="cb554-4"><a href="glm.html#cb554-4"></a>           <span class="dt">data =</span> D_Pois)</span></code></pre></div>
<!-- bayr #1 -->
<div class="sourceCode" id="cb555"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb555-1"><a href="glm.html#cb555-1"></a><span class="kw">fixef</span>(M_Pois)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">model</th>
<th align="left">type</th>
<th align="left">fixef</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">object</td>
<td align="left">fixef</td>
<td align="left">Intercept</td>
<td align="right">1.31</td>
<td align="right">1.16</td>
<td align="right">1.48</td>
</tr>
</tbody>
</table>
<p>Poisson distributions have only one parameter <span class="math inline">\(\lambda\)</span> (lambda), has a direct interpretation as it represents the expected mean (and variance) of the distribution. On the contrary, the regression coefficient is on a logarithmic scale to ensure it has no boundaries. To reverse to the scale of measurement, we use the exponential function is the <em>mean function</em> <a href="glm.html#relinking-linearity">6.1.1</a>:</p>
<div class="sourceCode" id="cb556"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb556-1"><a href="glm.html#cb556-1"></a><span class="kw">fixef</span>(M_Pois, <span class="dt">mean.func =</span> exp)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">model</th>
<th align="left">type</th>
<th align="left">fixef</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">object</td>
<td align="left">fixef</td>
<td align="left">Intercept</td>
<td align="right">3.72</td>
<td align="right">3.18</td>
<td align="right">4.41</td>
</tr>
</tbody>
</table>
<p>The exponentiated intercept coefficient can be interpreted as the expected number of items found per session. Together with the credibility limits it would allow the conclusion that the items are slightly easier to find than three per session.</p>
<p>Finally, let’s take a look of the formalism of the Poisson GMM:</p>
<p><span class="math display">\[
\begin{aligned}
\theta_i &amp;= \beta_0\\
\mu_i &amp;= \exp(\theta_i)\\
y_i &amp;\sim \textrm{Pois}(\mu_i)
\end{aligned}
\]</span></p>
<p>In linear models, the first equation used to directly relate fitted responses <span class="math inline">\(\mu_i\)</span> to the linear term. As any linear term is allowed to have negative results, this could lead to problems in the last line, because Poisson <span class="math inline">\(\lambda\)</span> is strictly non-negative. <em>Linear predictor</em> <span class="math inline">\(\theta_i\)</span> is taking the punches from the linear term and hands it over to the fitted responses <span class="math inline">\(\mu_i\)</span> via the exponential function. This function takes any number and returns a positive number, and that makes it safe for the last term that defines the pattern of randomness.</p>
<div id="speaking-multipliers" class="section level4">
<h4><span class="header-section-number">6.2.1.1</span> Speaking multipliers</h4>
<p>To demonstrate the interpretation of coefficients other than the intercept (or absolute group means), we turn to the more complex case of the infusion pump study. In this study, the deviations from a normative path were counted as a measure for error-proneness. In the following regression analysis, we examine the reduction of deviations by training sessions as well as the differences between the two devices. As we are interested in the improvement from first to second session and second to third, successive difference contrasts apply <a href="lm.html#treatment-contrasts">4.3.3</a>.</p>
<div class="sourceCode" id="cb557"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb557-1"><a href="glm.html#cb557-1"></a><span class="kw">attach</span>(IPump)</span></code></pre></div>
<div class="sourceCode" id="cb558"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb558-1"><a href="glm.html#cb558-1"></a>M_dev &lt;-<span class="st"> </span></span>
<span id="cb558-2"><a href="glm.html#cb558-2"></a><span class="st">  </span><span class="kw">stan_glmer</span>(deviations <span class="op">~</span><span class="st"> </span>Design <span class="op">+</span><span class="st"> </span>session <span class="op">+</span><span class="st"> </span>session<span class="op">:</span>Design <span class="op">+</span><span class="st"> </span></span>
<span id="cb558-3"><a href="glm.html#cb558-3"></a><span class="st">               </span>(<span class="dv">1</span> <span class="op">+</span><span class="st"> </span>Design <span class="op">+</span><span class="st"> </span>session<span class="op">|</span>Part) <span class="op">+</span></span>
<span id="cb558-4"><a href="glm.html#cb558-4"></a><span class="st">               </span>(<span class="dv">1</span> <span class="op">+</span><span class="st"> </span>Design<span class="op">|</span>Task) <span class="op">+</span></span>
<span id="cb558-5"><a href="glm.html#cb558-5"></a><span class="st">               </span>(<span class="dv">1</span><span class="op">|</span>Obs), <span class="co">## observation-level ramdom effect</span></span>
<span id="cb558-6"><a href="glm.html#cb558-6"></a>             <span class="dt">family =</span> poisson,</span>
<span id="cb558-7"><a href="glm.html#cb558-7"></a>             <span class="dt">data =</span> D_pumps) </span></code></pre></div>
<p>Note that in order to account for over-dispersion, observation-level random effect (<code>1|Obs</code>) has been used, see <a href="glm.html#overdispersion">6.2.3</a>. For the current matter, we only need to inspect population-level coefficients:</p>
<div class="sourceCode" id="cb559"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb559-1"><a href="glm.html#cb559-1"></a><span class="kw">fixef</span>(M_dev)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">fixef</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Intercept</td>
<td align="right">0.829</td>
<td align="right">0.233</td>
<td align="right">1.405</td>
</tr>
<tr class="even">
<td align="left">DesignNovel</td>
<td align="right">-1.522</td>
<td align="right">-2.307</td>
<td align="right">-0.740</td>
</tr>
<tr class="odd">
<td align="left">session</td>
<td align="right">-0.236</td>
<td align="right">-0.337</td>
<td align="right">-0.133</td>
</tr>
<tr class="even">
<td align="left">DesignNovel:session</td>
<td align="right">-0.075</td>
<td align="right">-0.244</td>
<td align="right">0.090</td>
</tr>
</tbody>
</table>
<p>These coefficients are on a logarithmic scale and cannot be interpreted right away. By using the exponential mean function, we reverse the logarithm and obtain the following table:</p>
<div class="sourceCode" id="cb560"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb560-1"><a href="glm.html#cb560-1"></a><span class="kw">fixef</span>(M_dev, <span class="dt">mean.func =</span> exp)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">fixef</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Intercept</td>
<td align="right">2.291</td>
<td align="right">1.262</td>
<td align="right">4.076</td>
</tr>
<tr class="even">
<td align="left">DesignNovel</td>
<td align="right">0.218</td>
<td align="right">0.100</td>
<td align="right">0.477</td>
</tr>
<tr class="odd">
<td align="left">session</td>
<td align="right">0.790</td>
<td align="right">0.714</td>
<td align="right">0.875</td>
</tr>
<tr class="even">
<td align="left">DesignNovel:session</td>
<td align="right">0.927</td>
<td align="right">0.784</td>
<td align="right">1.094</td>
</tr>
</tbody>
</table>
<p>The intercept now has the interpretation as the number of deviations with the legacy design in the first session. However, all the other coefficients are no longer summative, but <em>multiplicative</em>. It would therefore be incorrect to speak of them in terms of differences.</p>
<p><span class="math display">\[
\begin{aligned}
\mu_i &amp;= \exp(\beta_0 + x_1\beta_1 + x_2\beta_2) \\
&amp;= \exp(\beta_0) \exp(x_1\beta_1) \exp(x_2\beta_2)
\end{aligned}
\]</span></p>
<p>It has alwyays been more natural to speak of these effects in multiplicative form. If we would say “With the novel interface 1.8 fewer errors are being made”, that means nothing. 1.8 fewer than what? Instead, the following statements make perfect sense:</p>
<ol style="list-style-type: decimal">
<li>In the first session, the novel design produces 2.291 <em>times</em> the deviations than with the legacy design.</li>
<li>For the legacy design, every new training session reduces the number of deviations <em>by factor</em> 0.79<br />
</li>
<li>The reduction rate per training session of the novel design is *92.73% as compared to the legacy design.</li>
</ol>
<div class="sourceCode" id="cb561"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb561-1"><a href="glm.html#cb561-1"></a><span class="kw">detach</span>(IPump)</span></code></pre></div>
<p>To summarize: reporting coefficients on the linearized scale is usually not an option As we are not used to think in logarithmic terms, any quantitative message would get lost. By applying the mean function, we get back to the original scale. As it turns out, speaking multiplicative is natural and conveys more information. As we will see next, linearized scales can even describe non-linear relationships, such as training progress.</p>
</div>
<div id="learning-curves" class="section level4">
<h4><span class="header-section-number">6.2.1.2</span> Linearizing learning curves</h4>
<p>The Achilles heel of Gaussian linear models is the linearity assumption. All measures in this universe are finite, which means that all processes eventually hit a boundary. Linearity is an approximation that works well if you stay away from the boundaries. If you can’t, saturation effects happen and that means you have to add interaction effects or ordered factors to your model. Recall the IPump case, where we saw a learning curve over three sessions of practice with design Legacy and Novel. Learning curves are non-linear and we had to use an ordered factor model. Learning curves also have a lower boundary, which is often called the asymptote. As we will see now, learning curves can be described by one log-linearized slope coefficient, if the lower boundary is Zero.</p>
<p>The idea of replacing the OFM with a *linear__ized__ regression model (LzRM)<em>, is attractive. For one, with such a model we can obtain valid </em>forecasts* of the learning process. And second, the LzRM is more <em>parsimonous</em> <a href="wwm.html#overfitting">7.2.1</a>. For any sequence length, an LzRM just needs two parameters: intercept and slope, whereas the OFM requires one coefficient per session.</p>
<p>What if I also told you, that exponential functions make pretty good learning curves? (Review the simulation in <a href="lm.html#ofm">4.3.5</a>) Even the idea of a multiplicative effect bears some good intuition for learning processes, for example, if we say that by every session, errors are reduced to, say 80%, compared to the previous. This can be demonstrated by simulation of a learning experiment. This simulation takes a constant step size of <span class="math inline">\(\log(.8) = -0.223\)</span> on the linearized scale, resulting in a reduction of 20% per session.</p>
<div class="sourceCode" id="cb562"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb562-1"><a href="glm.html#cb562-1"></a>initial_deviations &lt;-<span class="st"> </span><span class="dv">20</span></span>
<span id="cb562-2"><a href="glm.html#cb562-2"></a>learning_rate &lt;-<span class="st"> </span><span class="fl">.8</span></span>
<span id="cb562-3"><a href="glm.html#cb562-3"></a></span>
<span id="cb562-4"><a href="glm.html#cb562-4"></a>D_learn &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">session =</span> <span class="dv">0</span><span class="op">:</span><span class="dv">10</span>,</span>
<span id="cb562-5"><a href="glm.html#cb562-5"></a>                  <span class="dt">theta =</span> <span class="kw">log</span>(initial_deviations) <span class="op">+</span><span class="st">  </span>session <span class="op">*</span><span class="st"> </span><span class="kw">log</span>(learning_rate),</span>
<span id="cb562-6"><a href="glm.html#cb562-6"></a>                  <span class="dt">mu =</span> <span class="kw">exp</span>(theta),</span>
<span id="cb562-7"><a href="glm.html#cb562-7"></a>                  <span class="dt">errors =</span> <span class="kw">rpois</span>(<span class="dv">11</span>, mu))</span>
<span id="cb562-8"><a href="glm.html#cb562-8"></a></span>
<span id="cb562-9"><a href="glm.html#cb562-9"></a>D_learn <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb562-10"><a href="glm.html#cb562-10"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span>  session, <span class="dt">y =</span> errors)) <span class="op">+</span></span>
<span id="cb562-11"><a href="glm.html#cb562-11"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">col =</span> <span class="st">&quot;simul. responses&quot;</span>)) <span class="op">+</span></span>
<span id="cb562-12"><a href="glm.html#cb562-12"></a><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">y =</span> mu, <span class="dt">col =</span> <span class="st">&quot;response scale&quot;</span>)) <span class="op">+</span></span>
<span id="cb562-13"><a href="glm.html#cb562-13"></a><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">y =</span> theta, <span class="dt">col =</span> <span class="st">&quot;linearized scale&quot;</span>)) <span class="op">+</span></span>
<span id="cb562-14"><a href="glm.html#cb562-14"></a><span class="st">  </span><span class="kw">ylim</span>(<span class="dv">0</span>,<span class="dv">20</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-28-1.png" width="90%" /></p>
<p>While the linear predictior scale is a straight line, the response scale clearly is a curve-of-diminishing returns. That opens up the possibility that learning the novel pump design has a constant multiplier (or rate) on the response scale, which would result in a constant difference on the linearized scale. In the following, we estimate two Poisson models, one linearized OFM (OzFM) (with stairway dummies <a href="lm.html#ofm">4.3.5</a>) and one LzRM. Then we will assess the model fit (using fitted responses). If the learning process is linear on the log scale, we can expect to see the following:</p>
<ol style="list-style-type: decimal">
<li>The two step coefficients of the OzFM become similar (they were wide apart for ToT).</li>
<li>The slope effect of the LzRM is the same as the step sizes.</li>
<li>Both models fit similar initial performance (intercepts)</li>
</ol>
<p>We estimate both models as usual, with conditional effects for Design:</p>
<div class="sourceCode" id="cb563"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb563-1"><a href="glm.html#cb563-1"></a><span class="kw">attach</span>(IPump)</span></code></pre></div>
<div class="sourceCode" id="cb564"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb564-1"><a href="glm.html#cb564-1"></a>D_agg &lt;-<span class="st"> </span></span>
<span id="cb564-2"><a href="glm.html#cb564-2"></a><span class="st">  </span>D_agg <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb564-3"><a href="glm.html#cb564-3"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Step_1 =</span> <span class="kw">as.integer</span>(session <span class="op">&gt;=</span><span class="st"> </span><span class="dv">1</span>),</span>
<span id="cb564-4"><a href="glm.html#cb564-4"></a>         <span class="dt">Step_2 =</span> <span class="kw">as.integer</span>(session <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>))</span>
<span id="cb564-5"><a href="glm.html#cb564-5"></a></span>
<span id="cb564-6"><a href="glm.html#cb564-6"></a><span class="co">## Ordered  regression model</span></span>
<span id="cb564-7"><a href="glm.html#cb564-7"></a>M_pois_cozfm &lt;-<span class="st"> </span></span>
<span id="cb564-8"><a href="glm.html#cb564-8"></a><span class="st">  </span>D_agg <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb564-9"><a href="glm.html#cb564-9"></a><span class="st">  </span><span class="kw">brm</span>(deviations <span class="op">~</span><span class="st"> </span><span class="dv">0</span> <span class="op">+</span><span class="st"> </span>Design <span class="op">+</span><span class="st"> </span>Step_<span class="dv">1</span><span class="op">:</span>Design <span class="op">+</span><span class="st"> </span>Step_<span class="dv">2</span><span class="op">:</span>Design, </span>
<span id="cb564-10"><a href="glm.html#cb564-10"></a>      <span class="dt">family =</span> <span class="st">&quot;poisson&quot;</span>, <span class="dt">data =</span> .)</span>
<span id="cb564-11"><a href="glm.html#cb564-11"></a></span>
<span id="cb564-12"><a href="glm.html#cb564-12"></a><span class="co">## Linear regression model</span></span>
<span id="cb564-13"><a href="glm.html#cb564-13"></a>M_pois_clzrm &lt;-</span>
<span id="cb564-14"><a href="glm.html#cb564-14"></a><span class="st">  </span>D_agg <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb564-15"><a href="glm.html#cb564-15"></a><span class="st">  </span><span class="kw">brm</span>(deviations <span class="op">~</span><span class="st"> </span><span class="dv">0</span> <span class="op">+</span><span class="st"> </span>Design <span class="op">+</span><span class="st"> </span>session<span class="op">:</span>Design, <span class="dt">family =</span> <span class="st">&quot;poisson&quot;</span>, <span class="dt">data =</span> .)</span></code></pre></div>
<p>For the question of a constant rate of learning, we compare the one linear coefficient of the regression model with the two steps of teh ordered factor model:</p>
<div class="sourceCode" id="cb565"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb565-1"><a href="glm.html#cb565-1"></a>T_fixef &lt;-</span>
<span id="cb565-2"><a href="glm.html#cb565-2"></a><span class="st">  </span><span class="kw">bind_rows</span>(</span>
<span id="cb565-3"><a href="glm.html#cb565-3"></a>    <span class="kw">posterior</span>(M_pois_cozfm),</span>
<span id="cb565-4"><a href="glm.html#cb565-4"></a>    <span class="kw">posterior</span>(M_pois_clzrm)</span>
<span id="cb565-5"><a href="glm.html#cb565-5"></a>  ) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb565-6"><a href="glm.html#cb565-6"></a><span class="st">  </span><span class="kw">fixef</span>(<span class="dt">mean.func =</span> exp) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb565-7"><a href="glm.html#cb565-7"></a><span class="st">  </span><span class="kw">separate</span>(fixef, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&quot;Design&quot;</span>, <span class="st">&quot;Learning_unit&quot;</span>), <span class="dt">sep =</span> <span class="st">&quot;:&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb565-8"><a href="glm.html#cb565-8"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">model =</span> <span class="kw">if_else</span>(<span class="kw">str_detect</span>(model, <span class="st">&quot;lz&quot;</span>), </span>
<span id="cb565-9"><a href="glm.html#cb565-9"></a>                         <span class="st">&quot;Linearized regression&quot;</span>, <span class="st">&quot;Ordered factors&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb565-10"><a href="glm.html#cb565-10"></a><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(Learning_unit)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb565-11"><a href="glm.html#cb565-11"></a><span class="st">  </span><span class="kw">arrange</span>(Design, Learning_unit, model) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb565-12"><a href="glm.html#cb565-12"></a><span class="st">  </span><span class="kw">discard_redundant</span>()</span>
<span id="cb565-13"><a href="glm.html#cb565-13"></a></span>
<span id="cb565-14"><a href="glm.html#cb565-14"></a>T_fixef</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">model</th>
<th align="left">Design</th>
<th align="left">Learning_unit</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Linearized regression</td>
<td align="left">DesignLegacy</td>
<td align="left">session</td>
<td align="right">0.772</td>
<td align="right">0.726</td>
<td align="right">0.824</td>
</tr>
<tr class="even">
<td align="left">Ordered factors</td>
<td align="left">DesignLegacy</td>
<td align="left">Step_1</td>
<td align="right">0.730</td>
<td align="right">0.653</td>
<td align="right">0.818</td>
</tr>
<tr class="odd">
<td align="left">Ordered factors</td>
<td align="left">DesignLegacy</td>
<td align="left">Step_2</td>
<td align="right">0.827</td>
<td align="right">0.726</td>
<td align="right">0.943</td>
</tr>
<tr class="even">
<td align="left">Linearized regression</td>
<td align="left">DesignNovel</td>
<td align="left">session</td>
<td align="right">0.773</td>
<td align="right">0.681</td>
<td align="right">0.876</td>
</tr>
<tr class="odd">
<td align="left">Ordered factors</td>
<td align="left">DesignNovel</td>
<td align="left">Step_1</td>
<td align="right">0.704</td>
<td align="right">0.558</td>
<td align="right">0.891</td>
</tr>
<tr class="even">
<td align="left">Ordered factors</td>
<td align="left">DesignNovel</td>
<td align="left">Step_2</td>
<td align="right">0.865</td>
<td align="right">0.656</td>
<td align="right">1.126</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb566"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb566-1"><a href="glm.html#cb566-1"></a>T_fixef <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb566-2"><a href="glm.html#cb566-2"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Learning_unit, <span class="dt">y =</span> center, <span class="dt">col =</span> Design, </span>
<span id="cb566-3"><a href="glm.html#cb566-3"></a>             <span class="dt">alpha =</span> model, <span class="dt">ymin =</span> lower, <span class="dt">ymax =</span> upper)) <span class="op">+</span></span>
<span id="cb566-4"><a href="glm.html#cb566-4"></a><span class="st">  </span><span class="kw">geom_crossbar</span>(<span class="dt">position =</span> <span class="st">&quot;dodge&quot;</span>, <span class="dt">width =</span> <span class="fl">.2</span>, <span class="dt">fill =</span> <span class="st">&quot;grey&quot;</span>) <span class="op">+</span></span>
<span id="cb566-5"><a href="glm.html#cb566-5"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">y =</span> <span class="st">&quot;rate of learning&quot;</span>) <span class="op">+</span></span>
<span id="cb566-6"><a href="glm.html#cb566-6"></a><span class="st">  </span><span class="kw">ylim</span>(<span class="dv">0</span>, <span class="fl">1.2</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-31-1.png" width="90%" /></p>
<p>Indeed, the coefficients Step_1 and Step_2 appear to be in a similar region for both designs, although learning at step 1 might be a little bit stronger. The OFM puts a huge amount of uncertainty on the coefficients. For Novel there even is quite some possibility that the learning rate falls in the region larger than One, which would mean performance gets worse. This makes little sense. What makes sense is that in both models the learning rate is the same for both designs. This indicates that the linearized regression model can even be reduced further, with just one parameter for learning rate (but with conditional intercepts):</p>
<div class="sourceCode" id="cb567"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb567-1"><a href="glm.html#cb567-1"></a>M_pois_lzrm &lt;-</span>
<span id="cb567-2"><a href="glm.html#cb567-2"></a><span class="st">  </span>D_agg <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb567-3"><a href="glm.html#cb567-3"></a><span class="st">  </span><span class="kw">brm</span>(deviations <span class="op">~</span><span class="st"> </span><span class="dv">0</span> <span class="op">+</span><span class="st"> </span>Design <span class="op">+</span><span class="st"> </span>session, <span class="dt">family =</span> <span class="st">&quot;poisson&quot;</span>, <span class="dt">data =</span> .)</span></code></pre></div>
<!-- ```{r} -->
<!-- T_pred <- -->
<!--   bind_rows( -->
<!--     post_pred(M_pois_ofm), -->
<!--     post_pred(M_pois_clzrm) -->
<!--   ) %>% -->
<!--   predict(scale = "lin_pred") %>% -->
<!--   left_join(D_Novel, by = "Obs") -->
<!-- T_pred %>% -->
<!--   ggplot(aes(x = Session, y = center, ymin = lower, ymax = upper, col = model)) + -->
<!--   geom_point(size = 3) + -->
<!--   geom_errorbar(width = .2, position = "dodge") -->
<!-- + -->
<!--   geom_jitter(data = D_Novel, aes(y = deviations)) -->
<!-- ``` -->
<p>We will return to this case in section <a href="wwm.html#ic">7.2.3</a> and show that the unconditional model, with just one learning rate parameter, is the best fit for the data. For now, the bottom line is that variables that a log-linearized model can remove nasty saturation effects and reduce model complexity. In addition, we replaced an ordered factor model with a smooth linear regression model, which has another advantage: The ordered factor model confines us to the number of learning sessions we have observed. From teh OFM, there is no way to predict, how the learning curve would progress with additional sessions. But, with a linearized regression model, we can produce <em>forecasts</em> for future sessions. For this purpose, we inject fabricated data (predictors only) into the model, extract the fitted responses and plot the results:</p>
<div class="sourceCode" id="cb568"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb568-1"><a href="glm.html#cb568-1"></a>D_forecast &lt;-<span class="st"> </span></span>
<span id="cb568-2"><a href="glm.html#cb568-2"></a><span class="st">  </span><span class="kw">expand_grid</span>(<span class="dt">session =</span> <span class="kw">c</span>(<span class="dv">0</span><span class="op">:</span><span class="dv">10</span>),</span>
<span id="cb568-3"><a href="glm.html#cb568-3"></a>              <span class="dt">Design =</span> <span class="kw">c</span>(<span class="st">&quot;Novel&quot;</span>, <span class="st">&quot;Legacy&quot;</span>),</span>
<span id="cb568-4"><a href="glm.html#cb568-4"></a>              <span class="dt">Part =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">50</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb568-5"><a href="glm.html#cb568-5"></a><span class="st">  </span><span class="kw">as_tbl_obs</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb568-6"><a href="glm.html#cb568-6"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Session =</span> <span class="kw">as.factor</span>(session <span class="op">+</span><span class="st"> </span><span class="dv">1</span>))</span>
<span id="cb568-7"><a href="glm.html#cb568-7"></a>  </span>
<span id="cb568-8"><a href="glm.html#cb568-8"></a>T_pred &lt;-<span class="st"> </span></span>
<span id="cb568-9"><a href="glm.html#cb568-9"></a><span class="st">  </span><span class="kw">post_pred</span>(M_pois_lzrm, <span class="dt">newdata =</span> D_forecast) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb568-10"><a href="glm.html#cb568-10"></a><span class="st">  </span><span class="kw">left_join</span>(D_forecast, <span class="dt">by =</span> <span class="st">&quot;Obs&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb568-11"><a href="glm.html#cb568-11"></a><span class="st">  </span><span class="kw">group_by</span>(Design, Session) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb568-12"><a href="glm.html#cb568-12"></a><span class="st">  </span><span class="kw">summarize</span>(<span class="dt">mean_deviations =</span> <span class="kw">mean</span>(value))</span>
<span id="cb568-13"><a href="glm.html#cb568-13"></a></span>
<span id="cb568-14"><a href="glm.html#cb568-14"></a>T_pred <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb568-15"><a href="glm.html#cb568-15"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">linetype =</span> <span class="kw">if_else</span>(<span class="kw">as.numeric</span>(Session) <span class="op">&lt;=</span><span class="st"> </span><span class="dv">3</span>, <span class="st">&quot;retrospective&quot;</span>, <span class="st">&quot;forecast&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb568-16"><a href="glm.html#cb568-16"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Session, <span class="dt">col =</span> Design, <span class="dt">y =</span> mean_deviations)) <span class="op">+</span></span>
<span id="cb568-17"><a href="glm.html#cb568-17"></a><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">group =</span> Design)) <span class="op">+</span></span>
<span id="cb568-18"><a href="glm.html#cb568-18"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">y =</span> deviations), <span class="dt">data =</span> D_pumps)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-34-1.png" width="90%" /></p>
<div class="sourceCode" id="cb569"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb569-1"><a href="glm.html#cb569-1"></a><span class="kw">detach</span>(IPump)</span></code></pre></div>
<p>Normally, fitted responses are just retrospective. Here, we extrapolate the learning curve by fake data and obtain real <em>forecasts</em>. We can make more interesting comparison of the two devices. For example, notice that initial performance with Novel is around five deviations. With Legacy this level is reached only in the seventh session. We can say that the Novel design is always seven sessions of training ahead of Legacy.</p>
<p>The conclusion is that log-linearized scales can reduce or even entirely remove saturation effects, such that we can go with a simpler models, that are easier to explain and potentially more useful. Potentially, because we can not generally construct parametric learning curves with log-linear models. The crucial property here is that the lower bound is Zero. Some measures have a positive lower bound, which is constant and known, and can be translated to a lower bound of Zero. For example, path length, the minimum number of steps to find something on the internet is One, Path length can just shifted by One, e.g. <code>mutate(addional_steps = steps + 1)</code> to create a lower bound of Zero. This is different for Time-on-Task, which always has a strictly positive lower bound, which we don’t know and which probably varies between individuals. Learning curves that approach strictly positive asymptotes have the following mathematical form:</p>
<div class="sourceCode" id="cb570"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb570-1"><a href="glm.html#cb570-1"></a><span class="co"># devtools::install_github(&quot;schmettow/asymptote&quot;)</span></span>
<span id="cb570-2"><a href="glm.html#cb570-2"></a>asymptote<span class="op">::</span>ARY</span></code></pre></div>
<pre><code>## perf ~ ampl * exp(-rate * trial) + asym
## &lt;environment: namespace:asymptote&gt;</code></pre>
<p>The offset to Zero is in the summand <code>asym</code>, and because it is a summand this term cannot be linearized in a straight-forward manner. For general learning curves a truly non-linear model is required, not just a linearized. This can be constructed with the Brm engine, but I considered this to be beyond the scope of this book.</p>
<!-- $$y_i = \phi\exp(\rho + x) + \omega$$ -->
<!-- where $\omega$ is the positive asymptote -->
<!-- ```{r} -->
<!-- # devtools::install_github("schmettow/asymptote") -->
<!-- asymptote::ARY -->
<!-- ``` -->
<!-- count errors, deviations or additional steps are great measures for experiments on learning. By model comparison against an OFM you first must assess whether the learning rate is approximately constant. If it is, then a log-linearized model renders a valid learning curve and can be used to predict future performance. When two designs are learned new, as in the IPump case, the difference can be interpreted as saved amount of training. Another interesting application would be how much training is needed for a novel design to actually beat a default design that users are highly trained with. -->
<!-- An interesting application arises, when a novel design is learned and compared against a default device, that users are highly trained with. A naive A/B study would come to te conclusion that teh default design is much better. With learning curves,  would An interesting application of learning curves is to find out  -->
<!-- #### Monotony and quantiles -->
<!-- The  transformation of coefficients to the original scale has been applied to the point and range estimates as produced by the `fixef` command, that is *after* summarizing the posterior distribution. One may wonder if this is valid.  Would we get the same estimates when applying the mean function to all draws of the MCMC chain and then summarize? The general answer is that applying the mean function after summarizing is allowed if the summary function is invariant under the exponential function. <!-- 101 -> -->
<!-- For all GLM, the link and mean functions are monotonically increasing, with the consequence that the order of observations is preserved. Formally, for any two MCMC iterations $i$ and  $j$ for a parameter $\beta_i$: -->
<!-- $$ -->
<!-- \beta_{1i} < \beta_{1j} \rightarrow \exp(\beta_{1i}) < \exp(\beta_{1j}) -->
<!-- $$ -->
<!-- Recall that throughout this book, center and interval estimates have been obtained by simple quantiles, marking the points where 2.5%, 50% and 97.5% of all iterations are smaller. Order does not change with monotonous transformations, if 2.5% (50%, 97.5%) of draws are smaller on the linear scale, they will still be after applying the mean function. Quantiles are not affected by monotonous transformation and transformation after summary is therefore valid. Some researchers prefer the mode of the posterior to represent its center location. The mode is the point of highest density and does not rely on ranks, it is therefore even invariant under all transformations that preserve identity. -->
<!-- This is different for higher order methods for obtaining point and interval estimates. Most notably the mean and the highest posterior density intervals are not invariant to mean functions. When using those, the mean function must be applied before summarizing the posterior, which is inconvenient and inconsistent. -->
<!-- #### Zero inflation -->
<!-- #### ZAP ME -->
<!-- Imagine a study that examines the frequency of visits to the social media website Fakebook <-- 102 ->. While other researchers already set out to find predictive factors, like extrovert personality and the like, we are here interested in the frequency of daily use.   -->
<!-- ```{r opts.label = "invisible"} -->
<!-- attach(Chapter_GLM) -->
<!-- ``` -->
<!-- ```{r sim_ZI, ops.label = "rtut"} -->
<!-- sim_zi <- function( -->
<!--   beta_0  = 2.5, -->
<!--   N_Part  = 120, -->
<!--   p_zero  = .2, # proportion of non-users -->
<!--   sd_Part = 0.8, # individual differences (lp scale) -->
<!--   seed    = 23 # parameters passed on to simulate_1 -->
<!-- ){ -->
<!--   set.seed(seed) -->
<!--   tibble(Part = 1:N_Part, -->
<!--              theta = rnorm(N_Part, beta_0, sd_Part), -->
<!--              mu = exp(theta), -->
<!--              is_user = rbinom(N_Part, 1, 1 - p_zero),  -->
<!--              visits = 0 + is_user *rpois(N_Part, mu)) -->
<!-- } -->
<!-- D_zi <- sim_zi() -->
<!-- D_zi %>%  -->
<!--   ggplot(aes(x = visits)) + -->
<!--   geom_histogram() -->
<!-- ``` -->
<!-- ```{r fit_ZI, opts.label = "future"} -->
<!-- M_zi <- -->
<!--   D_zi %>% brm(visits ~ 1 + (1|Part) , family = zero_inflated_poisson, data = .) -->
<!-- M_zi -->
<!-- sync_CE(Chapter_GLM, M_zi) -->
<!-- ``` -->
<!-- ```{r opts.label = "invisible"} -->
<!-- detach(Chapter_GLM) -->
<!-- sync_CE(Chapter_GLM, sim_zi, D_zi) -->
<!-- ``` -->
</div>
</div>
<div id="logistic-aka-binomial-regression-logistic-reg" class="section level3">
<h3><span class="header-section-number">6.2.2</span> Logistic (aka Binomial) regression (#logistic-reg)</h3>
<p>In the last section, we have seen how Poisson regression applies, when outcome variables are count numbers. More strictly, Poisson regression applies to count data, when there is no upper limit to counts (or if this limit is extremely large, as in the Smart Smurfer example). When the outcome variable is counts, but an upper limit exists and is known, <em>logistic regression</em> is an appropriate model. Such a situation often arises, when the counts are successes in a fixed number of trials. More brief, logistic regression has the following properties:</p>
<ol style="list-style-type: decimal">
<li>The outcome variable has a zero lower bound and a fixed upper bound, e.g. number of trials <span class="math inline">\(k\)</span>.</li>
<li>The linear predictors are on a <em>logit scale</em> also called <em>log-odds</em>, which is reversed by a <em>logistic function</em>.</li>
<li>The random component follows a <em>binomial distribution</em>.</li>
<li>Due to the former, the variance of randomness is largest at <span class="math inline">\(\mu = 0.5\)</span> or <span class="math inline">\(\eta = 1\)</span> and declines towards both boundaries, taking a characteristic cigar shape.</li>
</ol>
<p>Logistic regression applies for discrete outcomes, just like Poisson regression. The difference is that logistic regression has a finite number of possible outcomes, which is teh number of trials plus one. In the following section, I will first introduce logistic regression for when there is only one trial per observation, with two possible outcomes. That is called a <em>dichotomous outcomes</em>. This has some interesting applications for outcomes that are not exactly quantitative, like any other outcome variable covered in this book. Namely, we can apply logistic regression to variables that denote two classes. In the subsequent section, we will look at logistic regression for when there is more than one trial. The most difficult part of logistic regression is to report the estimated coefficients in an intelligible manner, which will be covered in the final section.</p>
<div id="dich-outcomes" class="section level4">
<h4><span class="header-section-number">6.2.2.1</span> Dichotomous outcomes</h4>
<p>The most simple form of successes-in-trials measure is when there is only one trial. This is called a <em>dichtotomous</em> variable, and that is very common:</p>
<ul>
<li>a user is successful at a task, or fails</li>
<li>a visitor returns to a website or does not</li>
<li>a usability problem is discovered or remains unseen</li>
<li>a driver brakes just in time or crashes</li>
<li>a customer recommends a product to a friend or does not</li>
<li>a user starts searching on a website by keyword or by traversing links</li>
</ul>
<p>Often, dichotomous outcome variables have a quantitative notion in the sense of more or less desireable.. When the outcome casts a positive light on the design, by convention it is coded as 1, otherwise 0. But, the dichotomy can also be two equal alternatives, such as whether a user starts a web inquiry by keyword search or by following a link. Let’s take this as an example. Research on search strategies of web users revealed that they are quite ecclectic regarding their method to find a piece of information. In particular, most people use keyword search and link navigation at occasion. Web users are also known to be impatient companions, who build a first judgment very quickly and swiftly turn to a competitor’s site, when the first impression leaves something to be desired. Therefore, it can be valuable to know what method the majority of users prefer, initially.</p>
<p>For this purpose, we can classify users by what method they start with when given a search task during a usability study. As there exist only two options, keyword search or following links, we can capture the outcome in a dichotomous response variable. Below is the simulation of a small data set, where 40% of users initially prefer keyword search:</p>
<!-- extremely impatient companions. They scan a page for visual features, rather than reading [REF: high school students information mall]. Visitors of websites build their first judgement in a time as short as 17ms [REF: Tuch presentation time]. For e-commerce that is a highly important fact to know about their customers and nowadays practically all commercial websites shine with a pleasing visual appearance. But, how would one measure the gratitude of a visitor who actually used the website and may have something to tell beyond visual appeal? -->
<!-- Let us consider an example: early research on foraging strategies of web users revealed that they are extremely impatient companions. They scan a page for visual features, rather than reading [REF: high school students information mall]. Visitors of websites build their first judgement in a time as short as 17ms [REF: Tuch presentation time]. For e-commerce that is a highly important fact to know about their customers and nowadays practically all commercial websites shine with a pleasing visual appearance. But, how would one measure the gratitude of a visitor who actually used the website and may have something to tell beyond visual appeal? -->
<!-- A simple measure for gratitude simply is when a visitor returns to buy more. This is usually a highly available measure, too, as any skilled web administrators can distill such data from the server logfiles with little effort.  -->
<!-- First, all unique visitors are extracted and if the same visitor returns within a given period of time, this is coded as a success (one) or otherwise failure (zero). We simulate such a data set:  -->
<div class="sourceCode" id="cb572"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb572-1"><a href="glm.html#cb572-1"></a><span class="kw">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb572-2"><a href="glm.html#cb572-2"></a>D_web &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">visitor =</span> <span class="kw">as.factor</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>),</span>
<span id="cb572-3"><a href="glm.html#cb572-3"></a>                <span class="dt">init_keyword =</span> <span class="kw">rbinom</span>(<span class="dv">100</span>, <span class="dv">1</span>, <span class="fl">.4</span>))</span>
<span id="cb572-4"><a href="glm.html#cb572-4"></a></span>
<span id="cb572-5"><a href="glm.html#cb572-5"></a>D_web <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sample_n</span>(<span class="dv">6</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">kable</span>()</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">visitor</th>
<th align="right">init_keyword</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">63</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">22</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">99</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">38</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">91</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">92</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb573"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb573-1"><a href="glm.html#cb573-1"></a>D_web <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb573-2"><a href="glm.html#cb573-2"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> init_keyword)) <span class="op">+</span></span>
<span id="cb573-3"><a href="glm.html#cb573-3"></a><span class="st">  </span><span class="kw">geom_bar</span>()</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-38-1.png" width="90%" /></p>
<p>For estimating the proportion of the two classes of users, we run a logistic regression grand mean model and inspect the coefficient table. Note that logistic regression is not called after its shape of randomness and we have to pass the <em>binomial family</em> to the regression engine.</p>
<div class="sourceCode" id="cb574"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb574-1"><a href="glm.html#cb574-1"></a>M_web &lt;-<span class="st"> </span>D_web <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">stan_glm</span>(init_keyword <span class="op">~</span><span class="st"> </span><span class="dv">1</span>, <span class="dt">data =</span> .,</span>
<span id="cb574-2"><a href="glm.html#cb574-2"></a>                            <span class="dt">family =</span> binomial)</span></code></pre></div>
<div class="sourceCode" id="cb575"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb575-1"><a href="glm.html#cb575-1"></a><span class="kw">clu</span>(M_web)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">parameter</th>
<th align="left">type</th>
<th align="left">fixef</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Intercept</td>
<td align="left">fixef</td>
<td align="left">Intercept</td>
<td align="right">-0.165</td>
<td align="right">-0.562</td>
<td align="right">0.261</td>
</tr>
</tbody>
</table>
<p>Clearly, the Intercept parameter is not a proportion, as that forbids negative values. Like with Poisson regression, the coefficient is on a linearized scale. As proportions are in a range from zero to one, a particular link function is sits between the measures and the linear term, stretching the bounded into an unbounded range. For logistic regression, the <em>logit</em> functions maps the fitted responses <span class="math inline">\(\mu_i \in [0;1]\)</span> onto the <em>linear predictor</em> scale <span class="math inline">\(\eta_i \in [-\infty; \infty]\)</span>.</p>
<p><span class="math display">\[
\eta_i = \textrm{logit}(\mu_i) = \log \frac{\mu_i} {1-\mu_i}
\]</span></p>
<p>Note that the fraction <span class="math inline">\(\frac{\mu_i} {1-\mu_i}\)</span> is the proportion of keyword search divided by the proportion of following links and is called an <em>odds</em>. The logit function is therefore often called <em>log-odds</em>. In section <a href="glm.html#talking-odds">6.2.2.3</a>, we will see how we can report logistic regression results as odds. In the case here, we can directly report the results as proportions, which requires to apply the <em>mean function</em>, which is the inverse of the logit, also known as the <em>logistic function</em>:</p>
<p><span class="math display">\[
\mu_i = \textrm{logit}^{-1}(\eta_i) = \frac{\exp \eta_1} {\exp \eta_i + 1}
\]</span></p>
<!-- \@ref(logit_logist) shows link and mean functions side-by-side. -->
<!-- ```{r logit_logist} -->
<!-- grid.arrange( -->
<!--   ggplot(data.frame(mu=c(0, 1)), aes(x = mu)) +  -->
<!--     stat_function(fun = mascutils::logit) + -->
<!--     xlab(expression(mu)) + ylab(expression(eta)) + -->
<!--     ggtitle("logit link function"), -->
<!--   ggplot(data.frame(eta=c(-5, 5)), aes(x = eta)) +  -->
<!--     stat_function(fun = mascutils::inv_logit) +  -->
<!--     xlab(expression(mu)) + ylab(expression(eta)) + -->
<!--     ggtitle("logistic mean function"), -->
<!--   nrow = 1) -->
<!-- ``` -->
<p>In GMM, <span class="math inline">\(\eta_i = \beta_0\)</span> and we can directly obtain the estimated proportion by applying the logistic function to the Intercept. The <code>clu</code> command lets you pass on a mean function. However, this logistic mean function is only useful for GMM and other absolute group means models, as we will see in <a href="glm.html#talking-odds">6.2.2.3</a>.</p>
<div class="sourceCode" id="cb576"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb576-1"><a href="glm.html#cb576-1"></a><span class="kw">posterior</span>(M_web) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">clu</span>(<span class="dt">mean.func =</span> inv_logit)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">model</th>
<th align="left">parameter</th>
<th align="left">type</th>
<th align="left">fixef</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">M_web</td>
<td align="left">Intercept</td>
<td align="left">fixef</td>
<td align="left">Intercept</td>
<td align="right">0.459</td>
<td align="right">0.363</td>
<td align="right">0.565</td>
</tr>
</tbody>
</table>
<p>From the GMM we retrieve one estimate that reflects the proportion to start by keyword search. As a side note, proportions could also be called probabilities, like “with 40% probability a user starts by keyword search.” However, I urge anyone to avoid speaking of logistic regression coefficients as probabilities. While mathematically this is correct, for the audience it can easily cause confusion with certainty or, beware of this, the p-value.</p>
<p>The apt reader may have noticed that the returners data set has been simulated with an exact return rate of 40%. Despite the sample size of 100, the center estimate seems rather off and hampered by considerable uncertainty. In computer science jargon, every dichotomous observation accounts to a <em>bit</em>, which is the smallest amount of information ever possible. Because the information of a single dichotomous observation is so sparse, large samples are important when dealing with dichtomous outcomes. Large samples can mean testing many users, or giving every user more than one trial.</p>
</div>
<div id="successes-trials" class="section level4">
<h4><span class="header-section-number">6.2.2.2</span> Successes in a number of trials</h4>
<p>If we repeatedly observe a dichotomous response, we can summarize the results as <em>successes-in trials</em>, like:</p>
<div class="sourceCode" id="cb577"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb577-1"><a href="glm.html#cb577-1"></a>responses &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb577-2"><a href="glm.html#cb577-2"></a><span class="kw">cat</span>(<span class="kw">sum</span>(responses), <span class="st">&quot;successes in&quot;</span>, <span class="kw">length</span>(responses), <span class="st">&quot;trials&quot;</span>)</span></code></pre></div>
<pre><code>## 4 successes in 6 trials</code></pre>
<p>Imagine we had conducted an extended version of the previous experiment, where users get set of ten tasks and we observe their initial behaviour every time they open a new website. As such tasks sometimes take very long, it may also happen that a participant cannot finish all ten tasks within time. That means, we potentially have a different number of attempts per participant, which we simulate as:</p>
<div class="sourceCode" id="cb579"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb579-1"><a href="glm.html#cb579-1"></a><span class="kw">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb579-2"><a href="glm.html#cb579-2"></a>D_web_ex &lt;-<span class="st"> </span></span>
<span id="cb579-3"><a href="glm.html#cb579-3"></a><span class="st">  </span><span class="kw">tibble</span>(</span>
<span id="cb579-4"><a href="glm.html#cb579-4"></a>    <span class="dt">Session =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">100</span>,</span>
<span id="cb579-5"><a href="glm.html#cb579-5"></a>    <span class="dt">trials =</span> <span class="kw">round</span>(<span class="kw">runif</span>(<span class="dv">100</span>, <span class="dv">7</span>, <span class="dv">10</span>), <span class="dv">0</span>),</span>
<span id="cb579-6"><a href="glm.html#cb579-6"></a>    <span class="dt">init_keyword =</span> <span class="kw">rbinom</span>(<span class="dv">100</span>, trials, <span class="fl">.4</span>),</span>
<span id="cb579-7"><a href="glm.html#cb579-7"></a>    <span class="dt">init_link =</span> trials <span class="op">-</span><span class="st"> </span>init_keyword) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb579-8"><a href="glm.html#cb579-8"></a><span class="st">  </span>mascutils<span class="op">::</span><span class="kw">as_tbl_obs</span>()</span>
<span id="cb579-9"><a href="glm.html#cb579-9"></a></span>
<span id="cb579-10"><a href="glm.html#cb579-10"></a>D_web_ex</span></code></pre></div>
<!-- Recall the fictional jump-and-run game  *smart smurfer* in \@ref(poisson_dist): the goal of the game is that players collect items and for the user experience it is crucial that this is neither too difficult nor too easy. Imagine, that for adjusting the difficulty level, the developers conduct a quick evaluation study, where they place a number of items (trials) in the game and the success rate of a single player is observed in a series of 15 game sessions. We simulate such a data set: -->
<!-- Per session the player has a number of opportunities for collecting an item, which makes it a repeated measures situation. One might expect that we need to include random effects into the model. Later we will see that this is necessary when the sessions were observed on a sample of players with different abilities. However, as long as one can reasonably assume the chance of catching an item to be constant across all sessions, plain logistic regression can deal with *successes in multiple trials*.  -->
<p>In order to estimate a model with more than one trial per observation, we have to specify how many trials there were. That is not done directly, but via the number of “failures”, in this case this is the number of trials where a link was followed. The response side of the model formula takes this in as an array with two columns, which is generally constructed as <code>cbind(successes, failures)</code>. We estimate a GMM:</p>
<div class="sourceCode" id="cb580"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb580-1"><a href="glm.html#cb580-1"></a>M_web_ex &lt;-<span class="st"> </span><span class="kw">stan_glm</span>(<span class="kw">cbind</span>(init_keyword, init_link) <span class="op">~</span><span class="st"> </span><span class="dv">1</span>,<span class="co"># &lt;--</span></span>
<span id="cb580-2"><a href="glm.html#cb580-2"></a>                   <span class="dt">family =</span> binomial,</span>
<span id="cb580-3"><a href="glm.html#cb580-3"></a>                   <span class="dt">data =</span> D_web_ex)</span></code></pre></div>
<div class="sourceCode" id="cb581"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb581-1"><a href="glm.html#cb581-1"></a><span class="kw">fixef</span>(M_web_ex, <span class="dt">mean.func =</span> inv_logit)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">model</th>
<th align="left">type</th>
<th align="left">fixef</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">object</td>
<td align="left">fixef</td>
<td align="left">Intercept</td>
<td align="right">0.407</td>
<td align="right">0.375</td>
<td align="right">0.441</td>
</tr>
</tbody>
</table>
<p>With repeated trials the estimate for tzhe proportion of initial keyword search gets much closer to real value and credibility intervals tighten up, too. By using teh inverse logit, we can readily report the results as proportions. But, make no mistake, when predictors come into play and additional coefficients are being estimated, reporting proportions does no longer work. Instead, we have to learn to speak in odds.</p>
</div>
<div id="talking-odds" class="section level4">
<h4><span class="header-section-number">6.2.2.3</span> Talking odds</h4>
<p>When presenting results of a statistical analysis, the linear predictor is likely to cause trouble, at least when the audience is interested in real quantities. Coefficients on a logit-linearized scale have only very general intuition:</p>
<ul>
<li>zero marks a 50% chance</li>
<li>positive values increase the chance, negative decrease</li>
<li>bigger effects have larger absolute values</li>
</ul>
<p>That is sufficient for purely ranking predictors by relative impact (if on a comparable scale of measurement), or plain hypothesis testing, but it does not connect well with quantities a decision maker is concerned with. Let’s see this at the example of the infusion pump study:</p>
<ol style="list-style-type: decimal">
<li>What is the expected frequency of failure on first use?</li>
<li>The novel design reduces failures, but is it sufficient?</li>
<li>Is frequency of failures sufficiently reduced after two training sessions?</li>
</ol>
<p>In the comparison of two medical infusion pumps (@ref(slope_RE)) 25 nurses completed a set of eight tasks repeatedly over three sessions. In @ref(slope_RE) a multi-level model was estimated on the workload outcome. It is tempting to apply the same structural model to success in task completion, using binomial random patterns and logit links.</p>
<pre><code>completion ~ Design*Session + (Design*Session|Part) + (Design*Session|Task)</code></pre>
<p>Such a model is practically impossible to estimate, because dichtomous variables are so scarce in information. Two populations encounter each other in the model: participants and tasks, with 6 observations per combination (6 bit). We should not expect to get reasonably certain estimates on that level and, in fact, the chains will not even mix well. The situation is a little better on the population level: every one of the six coefficients is estimated on 400 bit of raw information. We compromise here by estimating the full model on population level and do only intercept random effects to account for gross differences between participants and tasks.</p>
<div class="sourceCode" id="cb583"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb583-1"><a href="glm.html#cb583-1"></a><span class="kw">attach</span>(IPump)</span></code></pre></div>
<div class="sourceCode" id="cb584"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb584-1"><a href="glm.html#cb584-1"></a>M_cmpl &lt;-<span class="st">   </span></span>
<span id="cb584-2"><a href="glm.html#cb584-2"></a><span class="st">  </span>D_pumps <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb584-3"><a href="glm.html#cb584-3"></a><span class="st">  </span><span class="kw">stan_glmer</span>(completion <span class="op">~</span><span class="st"> </span>Design <span class="op">*</span><span class="st"> </span>Session <span class="op">+</span><span class="st"> </span></span>
<span id="cb584-4"><a href="glm.html#cb584-4"></a><span class="st">               </span>(<span class="dv">1</span><span class="op">|</span>Part) <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>Task),</span>
<span id="cb584-5"><a href="glm.html#cb584-5"></a>             <span class="dt">family =</span> binomial,</span>
<span id="cb584-6"><a href="glm.html#cb584-6"></a>             <span class="dt">data =</span> .)</span></code></pre></div>
<div class="sourceCode" id="cb585"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb585-1"><a href="glm.html#cb585-1"></a><span class="kw">fixef</span>(M_cmpl)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">fixef</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Intercept</td>
<td align="right">1.388</td>
<td align="right">0.069</td>
<td align="right">2.755</td>
</tr>
<tr class="even">
<td align="left">DesignNovel</td>
<td align="right">0.404</td>
<td align="right">0.092</td>
<td align="right">0.716</td>
</tr>
<tr class="odd">
<td align="left">Session2-1</td>
<td align="right">0.676</td>
<td align="right">0.147</td>
<td align="right">1.212</td>
</tr>
<tr class="even">
<td align="left">Session3-2</td>
<td align="right">-0.069</td>
<td align="right">-0.616</td>
<td align="right">0.463</td>
</tr>
<tr class="odd">
<td align="left">DesignNovel:Session2-1</td>
<td align="right">-0.300</td>
<td align="right">-1.055</td>
<td align="right">0.478</td>
</tr>
<tr class="even">
<td align="left">DesignNovel:Session3-2</td>
<td align="right">0.287</td>
<td align="right">-0.485</td>
<td align="right">1.068</td>
</tr>
</tbody>
</table>
<p>The result is one absoulte group mean, the Intercept, and five effects, which are mean differences on the logit-linearized scale <span class="math inline">\(\eta_i\)</span>. This is a boundless scale, where we can freely sum over effects to obtain group means. If we want to report group means, we can use the invers elogit function to obtain proportions, but for that we have to <em>first do the linear combination followed by the transformation</em>, for example:</p>
<ul>
<li>the completion rate in the first legacy session is 0.8</li>
<li>in novel/session 1: <code>logist(Intercept + DesignNovel)</code> = 0.857</li>
<li>in novel/session 2: <code>logist(Intercept + DesignNovel + Session2-1 + DesignNovel:Session2-1)</code> = 0.897</li>
<li>in legacy/session 3: <code>logist(Intercept + DesignNovel + Session2-1)</code> = 0.88</li>
</ul>
<p>Above we have used the mean logistic mean function to elevate the absolute group means to proportions. This is an intuitive scale, but unfortunately, the mean function does not apply to individual effects. It is for example, <em>incorrect</em> to apply it like: “the novel pumps proportion of failures in the first session increases by <code>logist(DesignNovel)</code> = 0.6”.</p>
<p>Now, it comes into play that the logit is a compound function, the logarithm of an odds. The inner part of the function, the <em>odds</em>, are the chance of success divided by the chance of failure. Especially in the anglo-american culture, odds are a rather common way to express ones chances in a game, say:</p>
<ul>
<li>odds are 1 against 1 that the coin flip produces Head. If you place €1 on Head, I put €1 on tail.</li>
<li>odds are 1 against 12 that Santa wins the dog race. If you place 1€ on Santa, I place €12 against.</li>
</ul>
<p>If the coefficients are log-odds, than we can extract the odds by the inverse of the logarithm, the exponential function, like in the following call of <code>fixef</code>:</p>
<div class="sourceCode" id="cb586"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb586-1"><a href="glm.html#cb586-1"></a><span class="kw">fixef</span>(M_cmpl, <span class="dt">mean.func =</span> exp)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">fixef</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Intercept</td>
<td align="right">4.008</td>
<td align="right">1.071</td>
<td align="right">15.72</td>
</tr>
<tr class="even">
<td align="left">DesignNovel</td>
<td align="right">1.497</td>
<td align="right">1.096</td>
<td align="right">2.05</td>
</tr>
<tr class="odd">
<td align="left">Session2-1</td>
<td align="right">1.966</td>
<td align="right">1.159</td>
<td align="right">3.36</td>
</tr>
<tr class="even">
<td align="left">Session3-2</td>
<td align="right">0.933</td>
<td align="right">0.540</td>
<td align="right">1.59</td>
</tr>
<tr class="odd">
<td align="left">DesignNovel:Session2-1</td>
<td align="right">0.741</td>
<td align="right">0.348</td>
<td align="right">1.61</td>
</tr>
<tr class="even">
<td align="left">DesignNovel:Session3-2</td>
<td align="right">1.333</td>
<td align="right">0.615</td>
<td align="right">2.91</td>
</tr>
</tbody>
</table>
<p>But is it legitimate to apply the transformation on individual coefficients in order to speak of changes of odds? The following arithmetic law tells that what is a sum on the log-odds scale, is multiplication on the scale of odds:</p>
<p><span class="math display">\[
\exp(x + y) = \exp(x)\exp(y)
\]</span></p>
<p>Consequently, we may speak of changes of odds using <em>multiplicative language</em>:</p>
<ul>
<li>If you place €100 on failure in the next task with the legacy design in session 1, I place €400.761 on success.</li>
<li>The odds of success with the novel design increase by <em>factor</em> 1.497. Now, I would place <span class="math inline">\(400.761 \times 1.497\)</span> = €599.962 on success.</li>
<li>On success with the novel design in session 2, I would place <span class="math inline">\(400.761 \times 1.497 \times 1.966 \times 0.741\)</span> = €874.002 on success.</li>
</ul>
<p>Once, we have transformed the coefficients to the odds scale, we can read coefficients as multipliers and speak of them in hard currency.</p>
<div class="sourceCode" id="cb587"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb587-1"><a href="glm.html#cb587-1"></a><span class="kw">detach</span>(IPump)</span></code></pre></div>
<p>To summarize: Logistic regression applies when the basic observations falls into two classes. For any research design involving such outcomes, repetition is highly recommended, and outcomes can be summarized into successes-in-trials. Reporting coefficients on the logit scale is only useful when nobody is interested in intelligible effects sizes. How to report the results depends on the research question. If one is interested in proportions per group, the inverse logit applies to the absolute group meeans and this can be easily understood. If one wants to talk about effects or differences, such as the amount of improvement with a novel design, only the logarithm of log-odds is inversed, and effects are reported as odds. Depending on teh audience, this may be more or less intuitive, but it can always be embedded in a wager for illustration.</p>
<p>While logistic regression is not easy to master it has important areas of application, as well as interesting extensions.</p>
<ul>
<li>In epidemiologic research, logistic regression is the undispensible tool for several central outcomes, such as hospitalization, mortality, infection and recovery.</li>
<li>In psychometrics, the famous Rasch model applies for measuring a persons ability by teh number of correct answers in a test. A Rasch model is just a cross-classified multilevel logistic regression <a href="mlm.html#designometrix">5.8.4</a>.</li>
<li>If the outcome is a classification with more than two classes, <em>multi-nomial regression</em> is an extension of logistic regression.</li>
<li>In section <a href="lm.html#ofm">4.3.5</a>, we will encounter <em>ordinal logistic regression</em>, which applies for classifications with an order, such as responses on Likert scales.</li>
</ul>
<p>One frequent problem when using logistic regression on successes-in-trials outcomes is that the assumption of a Binomial shape of randomness is violated by <em>over-dispersion</em>. Like Poisson distributions, Binomial distributions have a variance tightly linked to the mean, but frequently there is more variance than allowed, for example when tasks or test items vary in difficulty. In the following section two solutions to the problem are introduces: <em>beta-binomial regression</em> and <em>observation-level random effects</em>.</p>
<!--What is reported here, are the coefficients on the *linear predictor scale*, with the *logit* as link function. 
-->
<!-- \@ref(gross_task_completion) is chilling. Successful completion is the exception rather than the rule. Besides that, there is little difference between designs and sessions. For the mere sake of illustration we run a first logistic regression:





The Design factor is centered at zero and is highly uncertain. There simply is no clear conclusion possible. However, if Binomial distribution is underlying the logistic regression, should we not expect probability of success? Clearly that is not the case, as the lower credibility limits are negative. 



<!-- here: Infusion pump -->
<!-- For the two logistic GMMs, it seems one can short-circuit $\eta$ and state $\mu = \textrm{logist}(\eta)$ right away. This is not so, but a mere consequence of them only being the intercept $\mu = \beta_0$. It is also legitimate for absolute group mean models (AGMs), but not for any other coefficients with contrast settings. In order to find a specific predicted value, the mean function is applied to the linear combination of coefficients. For example, the 

However, in the final example of logistic regression we will see an alternative way to speak of logistic coefficients, *odds*, which allows transformation of individual coefficients.



<!--What is reported here, are the coefficients on the *linear predictor scale*, with the *logit* as link function. The logit is also called *log odds*: $\textrm{logit}(x) = \log(p(1-p))$. The inner part of the function, the *odds* are the chance of success divided by the chance of failure. That is a rather common way to express ones chances in a bet, say: "odds are 1 against 12 that Santa wins the dog race". If you place 1€ on Santa, I place €12 against.

If the coefficients are log odds, than we can convert them to odds by the inverse of the logarithm, the exponential function, like in the following call of `fixef`:



If you place €1000 on the next sequence with the legacy design being successful (which is the reference group), I place €103 against. What would be the odds of a successful sequence with the novel design? There are two ways to calculate. The first way is to do the respective linear combination on the linear predictor estimates, and than apply the exponential transformation:

$$\textrm(Odds(success|Novel)) = \exp(\beta_{0} + \beta_1) =
\exp(0) = 
1$$

The second way makes use of the following arithmetic law:

$$\exp(x + y) = \exp(x)\exp(y)$$

Once, we have transformed the linear predictor coefficients to the odds scale, we receive the combination of odds by multiplication:

$$\textrm(Odds(success|Novel)) = \exp(\beta_{0})\exp(\beta_1)) =
1$$
-->
<!--
Now we simulate a data set with the same research design, but the outcome variable is successes in ten tasks. For instance, the designs are three websites and participants are given ten information retrieval tasks.

Because the research design is the same, we can keep the two object data frames. The only issue is that $\beta$ and $\theta$ are on the scale of the linear predictor $\eta$, which is linked to fitted responses $\mu$ by the *logit link function*:

$$\eta = \log(\mu/(1-\mu))$$ 

fitted responses $\mu$ are computed from $\eta$ through the so-called *mean function*, which is the inverse of the logit, the *logistic tranformation*:

$$\mu = \exp(\eta)/{1 + \exp(eta)}$$ 

Note that values of \mu from the Gaussian data set above, are in the range of 300 - 350, which would result in chances of success of virtually 100%. To get more realistic values, we first divide $\eta$ by 100. 


from the experiment on Visual EDA:

Logistic regression coefficients are on a logit scale. For the interpretation, we want to back-transform to the original scale, which is: probability of the assumption to be rejected. In order to make statements on the probability scale, we backtransform using the inverse function of the logit, the logistic function:
mu_x = exp(eta_x)/(1 + exp(eta_x))
eta in the above formula is called the "linear predictor". The linear predictor is the value you get when you do operations on the coefficients. The most simple application is computing the overall probability for a rejection. This is represented by the Intercept parameter. In the table below, this parameter accounts to practically zero, on the logit scale. Back-transforming to the probability scale gives exactly a chance of 50%:
exp(0)/(1 + exp(0)) = 1/2
 So, an intercept of zero always means that chances are equal, which is the same as saying: the coefficient does not add any information. The same is true for parameters: the closer a coefficient is to zero, the less does it influence the prediction of rejection. Like with general linear models, parameters close to zero have "no effect" and can even be removed to simplify a model (we do not do that here).
Note, that the intercept in our model represents the situation that skew is 0 and N = 0. A more useful baseline rate for rejection rate would be at skew = 0 and the smallest sample size N = 10.
We can compute any prediction we want by combining the parameters with the predictors. But, the other coefficients in the model represent differences, not absolute values. Therefore, when predicting a certain event, let's say rejection at a skew = 0 and N = 50, we first have to compute the linear predictor eta, then transform to get predicted value mu. 
eta = 0.042 + Skew * 0 + Sampel size * 10
mu = logist(eta)
To give another example, the chance of rejection at skew = .5 and N = 50 is computed as:
eta = 0.042 + Skew * 0.5 + Sampel size * 50
mu = logist(eta)
Watch out! It is a severe mistake to transform coefficients individually, then do the linear combination.
eta = logist(Skew * 0.5) + logist(Sampel size * 50)
In any case, it is much preferred to report the estimates on the probability scale, as this is right-away interpretable. Credibility limits can be reported on the linear predictor scale. 

Group effects indicate how homogeneous the overall pattern (Table 1) is on individual units in the data set. Relevant units here are the participant and stimuli. For exanple, the relatively large coefficient skew means that people differ a lot in how much their response is influenced by skew (which is desireable): The strong effect on stimuli means that (after taking skew and N into account) stimuli differ a lot in which response they provoke. It sounds a bit paradox, but that means that stimuli differ systematically: there exist properties other then skew and N, that trigger a response.

-->
</div>
</div>
<div id="overdispersion" class="section level3">
<h3><span class="header-section-number">6.2.3</span> Modelling overdispersion</h3>
<!-- With count data, we usually expect the variance of randomness to be tied to the location (or magnitude). The Poisson distribution  is very strict in the sense that the variance equals the mean.  -->
<p>Poisson and binomial distributions are one-parameter distributions. As there is only one parameter, it is impossible to choose location and dispersion independently. In effect, both properties are tightly entangled. For Poisson distributions they are even the same.</p>
<p><span class="math display">\[
\begin{aligned}
Y &amp;\sim \textrm{Poisson} (\lambda) \\
\textrm {Var}(Y) &amp;= \textrm {Mean}(Y) = \lambda
\end{aligned}
\]</span></p>
<p>For binomial variables, mean and variance both depend on probability <span class="math inline">\(p\)</span> and are entangled in cigar shaped form, as the dispersion shrinks when approaching either two boundaries. Binomial variance is also affected by the number of trials <span class="math inline">\(k\)</span>, but that hardly matters as the value of <span class="math inline">\(k\)</span> is usually not up for estimation, but known a prior.</p>
<p><span class="math display">\[ 
\begin{aligned}
Y &amp;\sim \textrm{p, k} (\lambda) \\
\textrm {Mean}(Y) &amp;= kp \\
\textrm {Var}(Y) &amp;= kp(1 - p) 
\\&amp;= \textrm {Mean}(Y) (1-p)\\
\end{aligned}
\]</span></p>
<p>In real count data, we often see similar relationship between variance and mean, except that variance is inflated by some additional positive factor, which is called <em>overdispersion</em>. Poisson or Binomial distribution cannot render inflated data, and using them on over-dispersed data is a serious mistake. Fortunately, there exist two solutions to the problem, which I will introduce in the following three sections. In the first two sections, we will <em>replace the one-parameter distribution with a two-parameter distribution</em>, where the second parameter represents the factor of variance inflation. The second method is to use <em>observation-level ramdom effects</em>, which draws from the multi-level modelling toolbox.</p>
<p>Let me give you an example to illustrate the two methods. It is common saying that some people attract mosquito bites more than others. But is that really true? A simple lab experiment would do to test the “Sweet Blood” theory. A sample of participants are exposed to a pack of mosiquitos under carefully controlled conditions (time of day, environmental condition, hungriness of mosquitos). We don’t know the mechanisms that makes the blood sweeter, and hence cannot measure it. In the simulation below, it is just assumed that there is a such a property, but in a real study we would not know. Know imagine a study, where mosquito bites have been counted on 200 boy scouts after an expedition.</p>
<p>The following simulation function works by using a two-parameter distribution. Negative-binomial distributions are discrete distributions with a lower bound of zero, just like Poisson distributions. They also have the same location parameter <span class="math inline">\(\mu\)</span>, but a new parameter <code>size</code>, which re-scales the scale of measurement. When the scale of measurement is down-scaled, the distribution becomes relatively wider. When size approaches infinity, we are left with a plain Poisson variance. The following data simulation samples Sweet-blood data from a negative-binomial distribution with a size of 6.</p>
<div class="sourceCode" id="cb588"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb588-1"><a href="glm.html#cb588-1"></a><span class="kw">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb588-2"><a href="glm.html#cb588-2"></a>N =<span class="st"> </span><span class="dv">200</span></span>
<span id="cb588-3"><a href="glm.html#cb588-3"></a>avg_sweet &lt;-<span class="st"> </span><span class="dv">6</span></span>
<span id="cb588-4"><a href="glm.html#cb588-4"></a>size =<span class="st"> </span><span class="dv">3</span></span>
<span id="cb588-5"><a href="glm.html#cb588-5"></a></span>
<span id="cb588-6"><a href="glm.html#cb588-6"></a>Sweet_blood_nbin &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">Method =</span> <span class="st">&quot;NegBinomial&quot;</span>,</span>
<span id="cb588-7"><a href="glm.html#cb588-7"></a>                           <span class="dt">bites =</span> <span class="kw">rnbinom</span>(<span class="dt">n =</span> N, </span>
<span id="cb588-8"><a href="glm.html#cb588-8"></a>                                           <span class="dt">mu =</span> avg_sweet, </span>
<span id="cb588-9"><a href="glm.html#cb588-9"></a>                                           <span class="dt">size =</span> size))</span>
<span id="cb588-10"><a href="glm.html#cb588-10"></a></span>
<span id="cb588-11"><a href="glm.html#cb588-11"></a>Sweet_blood_nbin <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb588-12"><a href="glm.html#cb588-12"></a><span class="st">  </span><span class="kw">summarize</span>(<span class="kw">mean</span>(bites), <span class="kw">var</span>(bites))</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">mean(bites)</th>
<th align="right">var(bites)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">5.83</td>
<td align="right">16.7</td>
</tr>
</tbody>
</table>
<p>The next simulation first creates an observation-level indicator for blood sweetness, which in real data would not be known to the researcher; it is therefore a <em>latent variable</em>. Sweetness is sampled from a Gaussian distribution, that means we can get negative and positive values. Then Poisson random numbers are generated, but <code>lambda</code> is not a fixed value. Instead, every observation is sampled with a different value for sweetness. The exponentiation is simply to transform the Gaussian variable to the non-negative range of <code>lambda</code>. Taking the logarithm of the Gaussian <code>mu</code> puts the resulting distribution at approximately(!) the desired location of average sweetness.</p>
<div class="sourceCode" id="cb589"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb589-1"><a href="glm.html#cb589-1"></a><span class="kw">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb589-2"><a href="glm.html#cb589-2"></a>N =<span class="st"> </span><span class="dv">200</span></span>
<span id="cb589-3"><a href="glm.html#cb589-3"></a>avg_sweet &lt;-<span class="st"> </span><span class="dv">6</span></span>
<span id="cb589-4"><a href="glm.html#cb589-4"></a>size =<span class="st"> </span><span class="dv">4</span></span>
<span id="cb589-5"><a href="glm.html#cb589-5"></a></span>
<span id="cb589-6"><a href="glm.html#cb589-6"></a></span>
<span id="cb589-7"><a href="glm.html#cb589-7"></a>Sweet_blood_olre &lt;-</span>
<span id="cb589-8"><a href="glm.html#cb589-8"></a><span class="st">  </span><span class="kw">tibble</span>(<span class="dt">Method =</span> <span class="st">&quot;OLRE&quot;</span>,</span>
<span id="cb589-9"><a href="glm.html#cb589-9"></a>         <span class="dt">sweetness =</span> <span class="kw">rnorm</span>(N, <span class="dt">mean =</span> <span class="kw">log</span>(<span class="dv">6</span>), <span class="dt">sd =</span> <span class="fl">.5</span>),</span>
<span id="cb589-10"><a href="glm.html#cb589-10"></a>         <span class="dt">bites =</span> <span class="kw">rpois</span>(N, <span class="dt">lambda =</span> <span class="kw">exp</span>(sweetness)))</span>
<span id="cb589-11"><a href="glm.html#cb589-11"></a></span>
<span id="cb589-12"><a href="glm.html#cb589-12"></a>Sweet_blood_olre <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb589-13"><a href="glm.html#cb589-13"></a><span class="st">  </span><span class="kw">summarize</span>(<span class="kw">mean</span>(bites), <span class="kw">var</span>(bites))</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">mean(bites)</th>
<th align="right">var(bites)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">6.56</td>
<td align="right">16.5</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb590"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb590-1"><a href="glm.html#cb590-1"></a><span class="kw">bind_rows</span>(Sweet_blood_nbin,</span>
<span id="cb590-2"><a href="glm.html#cb590-2"></a>          Sweet_blood_olre) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb590-3"><a href="glm.html#cb590-3"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> bites)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb590-4"><a href="glm.html#cb590-4"></a><span class="st">  </span><span class="kw">geom_histogram</span>() <span class="op">+</span></span>
<span id="cb590-5"><a href="glm.html#cb590-5"></a><span class="st">  </span><span class="kw">facet_grid</span>(Method<span class="op">~</span><span class="dv">1</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-62-1.png" width="90%" /></p>
<p>The two methods for dealing with over-dispersion reverses either method of simulation. Either we choose a more flexible distribution, or we estimate the residuals on the linearized scale. The first method can be recommended, because it is leaner. Only one parameter is added, whereas OLRE results in one linearized residual for every observation. The advantage of OLRE is more of a conceptual kind. Not only does it appealing for researchers who are familiar with multi-level models, it also reverbs with the well-known concept of Gaussian residuals and lets us compare (linearized) residual variance to (linearized) ramdom effects variances.</p>
<!-- Overdispersion practically always happens in studies involving objects with complex dynamics, such as the human mind. Two solutions exist for overdispersed count data: we can either switch to a two-parameter response distribution, that gives variance more flexibility (see table below). -->
<!-- <!-- 46 make work with LaTeX -->
<p>–&gt;</p>
<!-- ```{r echo = F} -->
<!-- readxl::read_excel("Illustrations/GLM_distributions.xlsx", sheet = "plugin") -->
<!-- ``` -->
<!-- The alternative is to introduce an observation-level random effect, which is just like the Gaussian distributed variable Sweetness in the simulation above. -->
<!-- #### HERE -->
<div id="negbin-reg" class="section level4">
<h4><span class="header-section-number">6.2.3.1</span> Negative-binomial regression for overdispersed counts</h4>
<p>When Poisson regression is used for overdispersed count data, the model will produce reliable center estimates, but the credibility limits will be too narrow. The model suggests better certainty than there is. To explain that in simple terms: The model “sees” the location of a measure, which makes it seek errors in a region with precisely that variance. There will be many measures outside the likely region, but the model will hold on tight, regard these as (gradual) outliers and give them less weight. A solution to the problem is using a response distribution with <em>two parameters</em>. A second parameter usually gives variance of the distribution more flexibility, although only Gaussian models can set it loose, entirely. For the Poisson case (i.e. counts without an upper limit) <em>negative binomial distributions</em> do the job, for binomial distributions the beta-binomial applies. Both distributions are so-called <em>mixture distributions</em>. In mixture distributions, the parameter of the “outer” distribution is not constant, but allowed to vary by a distribution itself. Under this perspective, negative binomial distribution is equivalent to a Poisson distribution, if we let parameter <span class="math inline">\(\lambda\)</span> follow a gamma distribution, like this:</p>
<div class="sourceCode" id="cb591"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb591-1"><a href="glm.html#cb591-1"></a>rnegbinom &lt;-<span class="st"> </span><span class="cf">function</span>(n, mu, size){</span>
<span id="cb591-2"><a href="glm.html#cb591-2"></a>  shape &lt;-<span class="st"> </span>size</span>
<span id="cb591-3"><a href="glm.html#cb591-3"></a>  scale &lt;-<span class="st"> </span>mu<span class="op">/</span>size</span>
<span id="cb591-4"><a href="glm.html#cb591-4"></a>  lambdas &lt;-<span class="st"> </span><span class="kw">rgamma</span>(n, <span class="dt">shape =</span> shape, <span class="dt">scale =</span> scale)</span>
<span id="cb591-5"><a href="glm.html#cb591-5"></a>  <span class="kw">rpois</span>(n, <span class="dt">lambda =</span> lambdas)</span>
<span id="cb591-6"><a href="glm.html#cb591-6"></a>}</span>
<span id="cb591-7"><a href="glm.html#cb591-7"></a></span>
<span id="cb591-8"><a href="glm.html#cb591-8"></a><span class="kw">rnegbinom</span>(<span class="dv">1000</span>, <span class="dt">mu =</span> <span class="dv">3</span>, <span class="dt">size =</span> <span class="dv">2</span>) <span class="op">%&gt;%</span><span class="st">  </span><span class="kw">qplot</span>(<span class="dt">bins =</span> <span class="dv">20</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-63-1.png" width="90%" /></p>
<p>The figure below shows a negative binomial distribution and Poisson distribution with the same mean. The additional parameter size reduces the scale and makes the distribution wider.</p>
<div class="sourceCode" id="cb592"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb592-1"><a href="glm.html#cb592-1"></a><span class="kw">tibble</span>(<span class="dt">x =</span> <span class="dv">0</span><span class="op">:</span><span class="dv">15</span>,</span>
<span id="cb592-2"><a href="glm.html#cb592-2"></a>           <span class="dt">nbinom =</span> <span class="kw">dnbinom</span>(x, <span class="dt">mu =</span> <span class="dv">3</span>, <span class="dt">size =</span> <span class="dv">2</span>),</span>
<span id="cb592-3"><a href="glm.html#cb592-3"></a>           <span class="dt">poisson  =</span> <span class="kw">dpois</span>(x, <span class="dt">lambda =</span> <span class="dv">3</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb592-4"><a href="glm.html#cb592-4"></a><span class="st">  </span><span class="kw">gather</span>(distribution, prob, <span class="op">-</span>x) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb592-5"><a href="glm.html#cb592-5"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x, <span class="dt">y =</span> prob)) <span class="op">+</span></span>
<span id="cb592-6"><a href="glm.html#cb592-6"></a><span class="st">  </span><span class="kw">facet_grid</span>(distribution <span class="op">~</span><span class="st"> </span>.) <span class="op">+</span></span>
<span id="cb592-7"><a href="glm.html#cb592-7"></a><span class="st">  </span><span class="kw">geom_col</span>(<span class="dt">position =</span> <span class="st">&quot;dodge&quot;</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-64-1.png" width="90%" /></p>
<p>In <a href="glm.html#learning-curves">6.2.1.2</a> we have seen how log-linearization can accomodate learning curves, using a Poisson model. It is very likely that this data is over-dispersed and that the Poisson model was not correct. To demonstrate overdispersion, we estimate the unconditional learning curve model one more time, with a negative-binomial pattern of randomness:</p>
<div class="sourceCode" id="cb593"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb593-1"><a href="glm.html#cb593-1"></a><span class="kw">attach</span>(IPump)</span></code></pre></div>
<div class="sourceCode" id="cb594"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb594-1"><a href="glm.html#cb594-1"></a>M_negbin_lzrm &lt;-<span class="st"> </span></span>
<span id="cb594-2"><a href="glm.html#cb594-2"></a><span class="st">  </span>D_agg <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb594-3"><a href="glm.html#cb594-3"></a><span class="st">  </span><span class="kw">brm</span>(deviations <span class="op">~</span><span class="st"> </span><span class="dv">0</span> <span class="op">+</span><span class="st"> </span>Design <span class="op">+</span><span class="st"> </span>session, <span class="dt">family =</span> <span class="st">&quot;negbinomial&quot;</span>, <span class="dt">data =</span> .)</span></code></pre></div>
<div class="sourceCode" id="cb595"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb595-1"><a href="glm.html#cb595-1"></a><span class="kw">bind_rows</span>(</span>
<span id="cb595-2"><a href="glm.html#cb595-2"></a>  <span class="kw">posterior</span>(M_pois_lzrm),</span>
<span id="cb595-3"><a href="glm.html#cb595-3"></a>  <span class="kw">posterior</span>(M_negbin_lzrm)) <span class="op">%&gt;%</span><span class="st">   </span></span>
<span id="cb595-4"><a href="glm.html#cb595-4"></a><span class="st">  </span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;fixef&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb595-5"><a href="glm.html#cb595-5"></a><span class="st">  </span><span class="kw">clu</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb595-6"><a href="glm.html#cb595-6"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> fixef, <span class="dt">y =</span> center, <span class="dt">ymin =</span> lower, <span class="dt">ymax =</span> upper, <span class="dt">col =</span> model)) <span class="op">+</span></span>
<span id="cb595-7"><a href="glm.html#cb595-7"></a><span class="st">  </span><span class="kw">geom_crossbar</span>(<span class="dt">position =</span> <span class="st">&quot;dodge&quot;</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-68-1.png" width="90%" /></p>
<p>We observe that the center estimates are precisely the same. But, credibility limits are much wider with an underlying negative-binomial distribution. In the CLU table we also see that the neg-binomial model got another parameter <code>phi</code>, which is another name for the size parameter, controlling over-dispersion relative to a Poisson distribution as:</p>
<p><span class="math display">\[
\textrm{Variance} := \mu + \mu^2/\phi
\]</span></p>
<p>Due to the reciprocal term, the <em>smaller</em> <span class="math inline">\(\phi\)</span> gets, the <em>more</em> overdispersion had to be accounted for. From this formula alone it may seem that neg-binomial distributions could also account for under-dispersion, when we allow negative values. But, in most implementations <span class="math inline">\(\phi\)</span> must be non-negative. That is rarely a problem, as under-dispersion only occurs under very rare circumstances. Over-dispersion in count variables in contrast, is very common, if not ubiquitous. Negative-binomial regression solves the problem with just one additional parameter, which typically need not be interpreted. Reporting on coefficients uses the same principle as in plain Poisson regression: inversion by exponentiation and speaking <em>multiplicative</em>.</p>
<div class="sourceCode" id="cb596"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb596-1"><a href="glm.html#cb596-1"></a><span class="kw">detach</span>(IPump)</span></code></pre></div>
<!-- ```{r} -->
<!-- L_ovdsp_1 <- waic(M_ovdsp_1) -->
<!-- W_ovdsp_3 <- waic(M_ovdsp_2) -->
<!-- ``` -->
</div>
<div id="betabin-reg" class="section level4">
<h4><span class="header-section-number">6.2.3.2</span> Beta-binomial regression for successes in trials</h4>
<p>Beta-binomial regression follows a similar pattern as neg-binomial. A two parameter distribution allows to scale up the variance relative to a binomial model <a href="#logistic-reg"><strong>??</strong></a>. A beta-binomial distribution is created by replacing binomial parameter <span class="math inline">\(p\)</span> by a <span class="math inline">\(beta distribution\)</span>, with parameters <span class="math inline">\(a\)</span> and <span class="math inline">\(b\)</span>:</p>
<div class="sourceCode" id="cb597"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb597-1"><a href="glm.html#cb597-1"></a>rbetabinom &lt;-<span class="st"> </span><span class="cf">function</span>(n, size, a, b) <span class="kw">rbinom</span>(n, size, <span class="kw">rbeta</span>(n, a, b))</span>
<span id="cb597-2"><a href="glm.html#cb597-2"></a><span class="kw">rbetabinom</span>(<span class="dv">1000</span>, <span class="dv">10</span>, <span class="dv">1</span>, <span class="dv">2</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">qplot</span>()</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-70-1.png" width="90%" /></p>
<!-- Predictions and interpretation of coefficients of negative-binomial and beta-binomial models are just as with their counterparts, using the same link functions. There is just a tiny difference: a single parameter has been added to the model, which modifies the dispersion. In a standard analysis these parameters have very little meaning, even less than the standard error in a Gaussian model. No misunderstanding: these parameters are *not* the constant standard deviation of residuals. They act as scalers for the "natural" dispersion at any point. -->
<p>The Brms regression engine currently only implements the negative binomial, but not the beta-binomial family. That is a good opportunity to applaud the author of the Brms package for his ingenious architecture, which allows custom families to be defined by the user. The only requirement is that the distribution type is implemented in Stan [%Stan], which is the underlying general-purpose engine behind Brms. The following code is directly taken from the Brms documentation and adds beta-binomial distribution to the available shapes of randomness.</p>
<div class="sourceCode" id="cb598"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb598-1"><a href="glm.html#cb598-1"></a><span class="co"># define a custom beta-binomial family</span></span>
<span id="cb598-2"><a href="glm.html#cb598-2"></a>beta_binomial2 &lt;-<span class="st"> </span><span class="kw">custom_family</span>(</span>
<span id="cb598-3"><a href="glm.html#cb598-3"></a>  <span class="st">&quot;beta_binomial2&quot;</span>, <span class="dt">dpars =</span> <span class="kw">c</span>(<span class="st">&quot;mu&quot;</span>, <span class="st">&quot;phi&quot;</span>),</span>
<span id="cb598-4"><a href="glm.html#cb598-4"></a>  <span class="dt">links =</span> <span class="kw">c</span>(<span class="st">&quot;logit&quot;</span>, <span class="st">&quot;log&quot;</span>), <span class="dt">lb =</span> <span class="kw">c</span>(<span class="ot">NA</span>, <span class="dv">0</span>),</span>
<span id="cb598-5"><a href="glm.html#cb598-5"></a>  <span class="dt">type =</span> <span class="st">&quot;int&quot;</span>, <span class="dt">vars =</span> <span class="st">&quot;trials[n]&quot;</span></span>
<span id="cb598-6"><a href="glm.html#cb598-6"></a>)</span>
<span id="cb598-7"><a href="glm.html#cb598-7"></a></span>
<span id="cb598-8"><a href="glm.html#cb598-8"></a><span class="co"># define custom stan functions</span></span>
<span id="cb598-9"><a href="glm.html#cb598-9"></a>bb_stan_funs &lt;-<span class="st"> &quot;</span></span>
<span id="cb598-10"><a href="glm.html#cb598-10"></a><span class="st">  real beta_binomial2_lpmf(int y, real mu, real phi, int N) {</span></span>
<span id="cb598-11"><a href="glm.html#cb598-11"></a><span class="st">    return beta_binomial_lpmf(y | N, mu * phi, (1 - mu) * phi);</span></span>
<span id="cb598-12"><a href="glm.html#cb598-12"></a><span class="st">  }</span></span>
<span id="cb598-13"><a href="glm.html#cb598-13"></a><span class="st">  int beta_binomial2_rng(real mu, real phi, int N) {</span></span>
<span id="cb598-14"><a href="glm.html#cb598-14"></a><span class="st">    return beta_binomial_rng(N, mu * phi, (1 - mu) * phi);</span></span>
<span id="cb598-15"><a href="glm.html#cb598-15"></a><span class="st">  }</span></span>
<span id="cb598-16"><a href="glm.html#cb598-16"></a><span class="st">&quot;</span></span></code></pre></div>
<p><em>Note</em> that Beta-binomial distribution are usually parametrized with two shape paramneter <span class="math inline">\(a\)</span> and <span class="math inline">\(b\)</span>, which have a rather convoluted relationship with mean and variance. For a GLM a parametrization is required that has a mean parameter (for <span class="math inline">\(\mu_i\)</span>). Note, how the author of this code created a <code>beta_binomial2</code> distribution family, which takes <span class="math inline">\(\mu\)</span> and a scale parameter <span class="math inline">\(\phi\)</span>.</p>
<p>Defining the two function is sufficient to estimate beta-binomial models with Brms. In the following I simulate two outcomes from nine trials, <code>y</code> is sampled from a beta-binomial distribution, whereas <code>ybin</code> is from a Binomial distribution. Both have the same mean of <span class="math inline">\(.1\)</span> (10% correct). Subsequently, a beta-binomial and a binomial grand mean models are estimated.</p>
<div class="sourceCode" id="cb599"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb599-1"><a href="glm.html#cb599-1"></a><span class="kw">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb599-2"><a href="glm.html#cb599-2"></a>D_betabin &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">y =</span> VGAM<span class="op">::</span><span class="kw">rbetabinom</span>(<span class="dv">1000</span>, <span class="dv">9</span>, <span class="dt">prob =</span> <span class="fl">.1</span>, <span class="dt">rho =</span> <span class="fl">.3</span>),</span>
<span id="cb599-3"><a href="glm.html#cb599-3"></a>                    <span class="dt">n =</span> <span class="dv">9</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb599-4"><a href="glm.html#cb599-4"></a><span class="st">  </span><span class="kw">as_tbl_obs</span>()</span>
<span id="cb599-5"><a href="glm.html#cb599-5"></a></span>
<span id="cb599-6"><a href="glm.html#cb599-6"></a>M_betabin &lt;-<span class="st"> </span></span>
<span id="cb599-7"><a href="glm.html#cb599-7"></a><span class="st">  </span>D_betabin <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb599-8"><a href="glm.html#cb599-8"></a><span class="st">  </span><span class="kw">brm</span>(y <span class="op">|</span><span class="st"> </span><span class="kw">trials</span>(n) <span class="op">~</span><span class="st"> </span><span class="dv">1</span>, <span class="dt">family =</span> beta_binomial2, </span>
<span id="cb599-9"><a href="glm.html#cb599-9"></a>      <span class="dt">stan_funs =</span> bb_stan_funs, <span class="dt">data =</span> .)</span>
<span id="cb599-10"><a href="glm.html#cb599-10"></a></span>
<span id="cb599-11"><a href="glm.html#cb599-11"></a>M_bin &lt;-<span class="st"> </span><span class="kw">brm</span>(y <span class="op">|</span><span class="kw">trials</span>(n) <span class="op">~</span><span class="st"> </span><span class="dv">1</span>, <span class="dt">family =</span> <span class="st">&quot;binomial&quot;</span>, <span class="dt">data =</span> D_betabin)</span></code></pre></div>
<p>The following CLU table collects the estimates from both models, the true beta-binomial and the binomial, which does not account for over-dispersion in the data.</p>
<div class="sourceCode" id="cb600"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb600-1"><a href="glm.html#cb600-1"></a><span class="kw">bind_rows</span>(</span>
<span id="cb600-2"><a href="glm.html#cb600-2"></a>  <span class="kw">posterior</span>(M_bin),</span>
<span id="cb600-3"><a href="glm.html#cb600-3"></a>  <span class="kw">posterior</span>(M_betabin)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb600-4"><a href="glm.html#cb600-4"></a><span class="st">  </span><span class="kw">clu</span>(<span class="dt">mean.func =</span> inv_logit)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">model</th>
<th align="left">parameter</th>
<th align="left">type</th>
<th align="left">fixef</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">M_betabin</td>
<td align="left">b_Intercept</td>
<td align="left">fixef</td>
<td align="left">Intercept</td>
<td align="right">0.706</td>
<td align="right">0.663</td>
<td align="right">0.747</td>
</tr>
<tr class="even">
<td align="left">M_betabin</td>
<td align="left">phi</td>
<td align="left">shape</td>
<td align="left">NA</td>
<td align="right">0.998</td>
<td align="right">0.983</td>
<td align="right">1.000</td>
</tr>
<tr class="odd">
<td align="left">M_bin</td>
<td align="left">b_Intercept</td>
<td align="left">fixef</td>
<td align="left">Intercept</td>
<td align="right">0.706</td>
<td align="right">0.679</td>
<td align="right">0.734</td>
</tr>
</tbody>
</table>
<p>When comparing the two intercept estimates, we notice that the center estimate is not affected by over-dispersion. But, just like with Poisson models, the binomial model is too optimistic about the level of certainty.</p>
<p>We have seen To summarize: one-parameter distributions usually cannot be used to model count data due to extra variance. One solution to the problem is to switch to a family with a second parameter. These exist for the most common situations. When we turn to modelling durations, we will use the Gamma family to extend the Exponential distribution <a href="glm.html#exp-gam-reg">6.3.1</a>. Gamma distributions have another problem: while extra variance can be accounted by a scale parameter, we will see that another property of distribution families can be to rigid, the skew. The solution will be to switch to a three-parameter distribution family to gain more flexibility.</p>
<p>Another technique to model over-dispersion does not require to find (or define) a two-parametric distribution. Instead, <em>observation-level random effects</em> borrow concepts from multi-level modelling and allow to keep the one-parameter distributions.</p>
</div>
<div id="olre" class="section level4">
<h4><span class="header-section-number">6.2.3.3</span> Using observation-level random effects</h4>
<p>As we have seen in chapter <a href="mlm.html#mlm">5</a>, random effects are often interpreted towards variance in a population, with a Gaussian distribution. On several occasions were multi-level models used to separate sources of variance, such as between teams and participants in CUE8. <em>Observation-level random effect</em> (OLRE) use the same approach by just calling the set of observation a <em>a population</em>.</p>
<p>Using random effects with GLMs is straight-forward, because random effects (or their dummy variable representation, to be precise), are part of the linear term, and undergo the log or logit linearization just like any population-level effect in the model.</p>
<!-- Recall how we regard variation between members of a population as normally distributed deviations from the population mean, by the example of a Poisson grand mean model with a participant-level ($p$) random effect: -->
<!-- $$ -->
<!-- \theta_{pi} = \beta_0 + x_p\beta_{0p} \\ -->
<!-- \mu_{pi} = \exp(\theta_{pi})\\ -->
<!-- \beta_{0p} \sim N(\mu_{p}, \sigma_p)\\ -->
<!-- y_{p} \sim \textrm{Pois}(\mu_{ij}) -->
<!-- $$ -->
<!-- The OLRE is normally distributed but does not cause any bounded range, as it is added on the level of the linear predictor before applying the exponential transformation. Observation-level random effects are completely analogous, except that every observation becomes its own group, in a Poisson grand mean model with added variation: -->
<!-- $$ -->
<!-- \theta_{i} = \beta_0 + \beta_{i} \\ -->
<!-- \mu_i = \exp(\theta_i)\\ -->
<!-- y_{ij} \sim \textrm{Pois}(\mu_{ij}) -->
<!-- $$ -->
<!-- See, how $\beta_i$ is a unique deviation per observation $i$, and how a variance parameter $\sigma$ appears in an otherwise purely Poisson model. Observation-level random effects are on the linear predictor level, and therefore additive. Compare this to the negative binomial distribution where variance is scaled up, which is multiplication. We find a resemblance with how sums on the linear predictor become multiplications on the fitted responses scale. -->
<p>For demonstration of the concept, we simulate from an overdispersed Poisson grand mean model with participant-level variation and observation-level variation.</p>
<div class="sourceCode" id="cb601"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb601-1"><a href="glm.html#cb601-1"></a>sim_ovdsp &lt;-<span class="st"> </span><span class="cf">function</span>(</span>
<span id="cb601-2"><a href="glm.html#cb601-2"></a>  <span class="dt">beta_0 =</span> <span class="dv">2</span>,   <span class="co"># mu = 8</span></span>
<span id="cb601-3"><a href="glm.html#cb601-3"></a>  <span class="dt">sd_Obs =</span> <span class="fl">.3</span>, </span>
<span id="cb601-4"><a href="glm.html#cb601-4"></a>  <span class="dt">sd_Part   =</span> <span class="fl">.5</span>,</span>
<span id="cb601-5"><a href="glm.html#cb601-5"></a>  <span class="dt">N_Part =</span> <span class="dv">30</span>,</span>
<span id="cb601-6"><a href="glm.html#cb601-6"></a>  <span class="dt">N_Rep  =</span> <span class="dv">20</span>, </span>
<span id="cb601-7"><a href="glm.html#cb601-7"></a>  <span class="dt">N_Obs =</span>  N_Part <span class="op">*</span><span class="st"> </span>N_Rep,</span>
<span id="cb601-8"><a href="glm.html#cb601-8"></a>  <span class="dt">seed =</span> <span class="dv">1</span>){</span>
<span id="cb601-9"><a href="glm.html#cb601-9"></a>  <span class="kw">set.seed</span>(seed)</span>
<span id="cb601-10"><a href="glm.html#cb601-10"></a>  Part &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">Part =</span> <span class="dv">1</span><span class="op">:</span>N_Part,</span>
<span id="cb601-11"><a href="glm.html#cb601-11"></a>                     <span class="dt">beta_0p =</span> <span class="kw">rnorm</span>(N_Part, <span class="dv">0</span>, sd_Part)) <span class="co">## participant-level RE</span></span>
<span id="cb601-12"><a href="glm.html#cb601-12"></a>  D &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">Obs  =</span> <span class="dv">1</span><span class="op">:</span>N_Obs,</span>
<span id="cb601-13"><a href="glm.html#cb601-13"></a>                    <span class="dt">Part =</span> <span class="kw">rep</span>(<span class="dv">1</span><span class="op">:</span>N_Part, N_Rep),</span>
<span id="cb601-14"><a href="glm.html#cb601-14"></a>                    <span class="dt">beta_0i =</span> <span class="kw">rnorm</span>(N_Obs, <span class="dv">0</span>, sd_Obs),   <span class="co">## observeration-level RE</span></span>
<span id="cb601-15"><a href="glm.html#cb601-15"></a>                    <span class="dt">beta_0 =</span> beta_<span class="dv">0</span>) <span class="op">%&gt;%</span></span>
<span id="cb601-16"><a href="glm.html#cb601-16"></a><span class="st">    </span><span class="kw">left_join</span>(Part) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb601-17"><a href="glm.html#cb601-17"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">theta_i =</span> beta_<span class="dv">0</span> <span class="op">+</span><span class="st"> </span>beta_0p <span class="op">+</span><span class="st"> </span>beta_0i,</span>
<span id="cb601-18"><a href="glm.html#cb601-18"></a>           <span class="dt">mu_i    =</span> <span class="kw">exp</span>(theta_i),               <span class="co">## inverse link function</span></span>
<span id="cb601-19"><a href="glm.html#cb601-19"></a>           <span class="dt">y_i     =</span> <span class="kw">rpois</span>(N_Obs, mu_i))</span>
<span id="cb601-20"><a href="glm.html#cb601-20"></a>  D <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tbl_obs</span>()</span>
<span id="cb601-21"><a href="glm.html#cb601-21"></a>}</span>
<span id="cb601-22"><a href="glm.html#cb601-22"></a></span>
<span id="cb601-23"><a href="glm.html#cb601-23"></a>D_ovdsp &lt;-<span class="st"> </span><span class="kw">sim_ovdsp</span>()</span>
<span id="cb601-24"><a href="glm.html#cb601-24"></a></span>
<span id="cb601-25"><a href="glm.html#cb601-25"></a>D_ovdsp <span class="op">%&gt;%</span></span>
<span id="cb601-26"><a href="glm.html#cb601-26"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> y_i)) <span class="op">+</span></span>
<span id="cb601-27"><a href="glm.html#cb601-27"></a><span class="st">  </span><span class="kw">geom_histogram</span>(<span class="dt">bins =</span> <span class="dv">15</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-77-1.png" width="90%" /></p>
<p>The above code is instructive to how OLREs work:</p>
<ol style="list-style-type: decimal">
<li>A participant-level random effect is created as <code>beta_0p</code>. This random effect can be recovered, because we have repeated measures. This variation will not contaminate Poisson variance.</li>
<li>An observation-level random effect is created in much the same way.</li>
<li>Both random effects are on the linearized scale. The linear predictor <code>theta_i</code> is just the sum of random effects (and Intercept). It could take negative values, but …</li>
<li>… applying the inverse link function (<code>exp(theta_i)</code>) ensures that all responses are positive.</li>
</ol>
<p>The extra variation comes from two sources: participant-level, with repeated measures and observation-level, without repeated measures. When random effects were introduced in the previous chapter, I said that estimating random effects requires repeated measures. For Gaussian models, that is true (and probably also for other two-parametric families). In contrast, one-parameter distributions precisely specify their variance at any point of location and therefore the OLREs can be recovered. The following model contains an participant-level random effects and an OLRE:</p>
<div class="sourceCode" id="cb602"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb602-1"><a href="glm.html#cb602-1"></a>M_ovdsp &lt;-<span class="st"> </span></span>
<span id="cb602-2"><a href="glm.html#cb602-2"></a><span class="st">  </span>D_ovdsp <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb602-3"><a href="glm.html#cb602-3"></a><span class="st">  </span><span class="kw">stan_glmer</span>(y_i <span class="op">~</span><span class="st"> </span><span class="dv">1</span> <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>Part) <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>Obs), <span class="dt">data =</span> .,</span>
<span id="cb602-4"><a href="glm.html#cb602-4"></a>             <span class="dt">family =</span> poisson)</span></code></pre></div>
<div class="sourceCode" id="cb603"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb603-1"><a href="glm.html#cb603-1"></a><span class="kw">grpef</span>(M_ovdsp)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">re_factor</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Obs</td>
<td align="right">0.305</td>
<td align="right">0.264</td>
<td align="right">0.350</td>
</tr>
<tr class="even">
<td align="left">Part</td>
<td align="right">0.536</td>
<td align="right">0.403</td>
<td align="right">0.719</td>
</tr>
</tbody>
</table>
<p>Let’s first take a look at the two random effect standard errors above. It seems that we got a fair recovery on the center estimates (for standard deviation). For OLRE certainty is also good, even better than for the participant-level, which is simply due to the fact that there are more levels. Random effect variation is accurately recovered from the simulated data, but can we also recover the full vector of factor levels? The following plot shows that In the following I am extracting observation-level random effects and plot them against the simulated (linearized) coefficients:</p>
<div class="sourceCode" id="cb604"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb604-1"><a href="glm.html#cb604-1"></a>OLRE &lt;-<span class="st"> </span></span>
<span id="cb604-2"><a href="glm.html#cb604-2"></a><span class="st">  </span><span class="kw">posterior</span>(M_ovdsp) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb604-3"><a href="glm.html#cb604-3"></a><span class="st">  </span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;ranef&quot;</span>, re_factor <span class="op">==</span><span class="st"> &quot;Obs&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb604-4"><a href="glm.html#cb604-4"></a><span class="st">  </span><span class="kw">clu</span>()</span>
<span id="cb604-5"><a href="glm.html#cb604-5"></a></span>
<span id="cb604-6"><a href="glm.html#cb604-6"></a>D_ovdsp <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb604-7"><a href="glm.html#cb604-7"></a><span class="st">  </span><span class="kw">bind_cols</span>(OLRE) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb604-8"><a href="glm.html#cb604-8"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> beta_0i, <span class="dt">y =</span> center)) <span class="op">+</span></span>
<span id="cb604-9"><a href="glm.html#cb604-9"></a><span class="st">  </span><span class="kw">geom_point</span>() <span class="op">+</span></span>
<span id="cb604-10"><a href="glm.html#cb604-10"></a><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">se =</span> F)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-81-1.png" width="90%" /></p>
<p>This shows, that even the observation-level deviations can be recovered quite well. For every single observation, the model can determine how much it has been pushed by unrecorded sources of variation. These OLRE levels can be seen as generalized residuals, or <em>linearized residuals</em>.</p>
<p>Linearized residuals can be used for different purposes, such as outlier detection or to compare sources of variation. Frequently, I reminded the reader to interpret parameters quantitatively by translating their magnitude to statements of practical relevance. For random effects variance this is not always straight forward, especially when we are on a linearized scale. One way is to make comparative statements on the sources of variance, like “the variance due to individual differences exceeds the measurement error”. OLREs are on the same scale as all other random effects in the model, which makes it a suitable reference source of variation.</p>
<!-- A non-default comparison of sources of variance is the one of Dennis Egan, that people cause more variance than designs do. With <!-- #111 GLMMs, the marriage of LMM and GLM this claim is testable. -->
<!-- Up to this point, this would only make sense for Gaussian models, as only those have this parameter. And only with Gaussian models does it make sense to talk about residuals in the sense of `y_i = \mu_i + \epsilon_i`. Residuals are summands on the response scale, whereas Poisson models are multiplicative on the response scale. OLREs operate  -->
<!-- From LMM we borrow a surprisingly simple and general solution, observation-level random effects. So, most of the time we will not need one of those twisted two parameter random distributions to account for overdispersion. With OLRE models we get an estimate that is very similar to *residuals*, which has proven very useful in model criticism. -->
<pre><code>## [1] &quot;sim_ovdsp&quot; &quot;D_ovdsp&quot;</code></pre>
</div>
</div>
</div>
<div id="duration-measures" class="section level2">
<h2><span class="header-section-number">6.3</span> Duration measures</h2>
<p>Time is a highly accessible measure, as clocks are all around us: on your wrist, in transport stations, in your computers and a very big (and accurate) one is hosted at the Physikalisch-Technischen Bundesanstalt in Braunschweig (Physical-technological federal institute in Braunschweig, Germany). Duration measures often carry useful information; especially, <em>Reaction time (RT)</em> measures are prime in experimental cognitive studies and have revealed fascinating phenomena of the human mind, such as the Stroop effect, memory priming, motor learning and the structure of attention. In design research, reaction times are also frequently used in experiments, but more common is <em>time-on-task (ToT)</em> as a measure of task efficiency. Formally, both outcome types measure a period of time. I am deliberately making a distinction between the two, because the data generating process of reacting to a simple task (like naming a color) may be different to a complex task, like fidning information on a website.</p>
<p>Temporal variables are practically continuous (as long as one measures with sufficient precision), but always have lower bounds. First, I will introduce two classic types of models that use exponentially or Gamma distributed error terms. Both distribution families assume a lower bound at zero, which is problematic, as we will see. Modern Bayesian estimation engines offer an increasing variety of more exotic response distributions. Among those are Exgaussian response distributions, which works well when the lower bound is positive.</p>
<div id="exp-gam-reg" class="section level3">
<h3><span class="header-section-number">6.3.1</span> Exponential and Gamma regression</h3>
<p>Exponential distributions arise from basic random processes under some very idealized conditions. First, the lower boundary must be zero and second, the rate at which events happen is assumed to be constant rate, just like Poisson distributions assumes a constant <span class="math inline">\(\lambda\)</span>.</p>
<!-- This leads to the property called *being memoryless*, which can be mind-boggling as . In many situations we have the intuition that the longer we wait, the shorter our expected remaining time-to-wait. Examples of such memoryful processes are plenty, such as: -->
<!-- + the longer a little girl waits, the closer is her birthday -->
<!-- + the longer you wait in a queue, the more likely you'll be called in, soon. -->
<!-- + earthquakes are essentially relaxations in the earth crust. Once it happened, it is less likely to reoccur in near future. -->
<p>Reconsider the subway smurfer example <a href="ebs.html#distributions">3.4.2</a>, where players collect items in a jump and run game. We have already seen how collection counts can be modelled using Poisson or binomial regression. Another way to look at it is the time between two events of item collection.
For demonstration only, we assume such idealized conditions in the subway smurfer example and generate a data set. Exponential distributions are determined by one parameter, the <em>rate</em> parameter <span class="math inline">\(\lambda\)</span>, which is strictly positive. The mean of an exponential distribution is the reciprocal <span class="math inline">\(1/\lambda\)</span> and the variance is <span class="math inline">\(\textrm{Var} = 1/\lambda^2\)</span>. Like with Poisson regression, variance is strictly tied to the mean.</p>
<div class="sourceCode" id="cb606"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb606-1"><a href="glm.html#cb606-1"></a><span class="kw">set.seed</span>(<span class="dv">20</span>)</span>
<span id="cb606-2"><a href="glm.html#cb606-2"></a>D_exp &lt;-<span class="st"> </span></span>
<span id="cb606-3"><a href="glm.html#cb606-3"></a><span class="st">  </span><span class="kw">tibble</span>(<span class="dt">Obs =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">100</span>,</span>
<span id="cb606-4"><a href="glm.html#cb606-4"></a>  <span class="dt">time =</span> <span class="kw">rexp</span>(<span class="dv">100</span>, <span class="dt">rate =</span> <span class="dv">1</span><span class="op">/</span><span class="dv">3</span>))</span>
<span id="cb606-5"><a href="glm.html#cb606-5"></a></span>
<span id="cb606-6"><a href="glm.html#cb606-6"></a>D_exp <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb606-7"><a href="glm.html#cb606-7"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> time)) <span class="op">+</span></span>
<span id="cb606-8"><a href="glm.html#cb606-8"></a><span class="st">  </span><span class="kw">geom_histogram</span>(<span class="dt">bins =</span> <span class="dv">10</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-84-1.png" width="90%" /></p>
<div class="sourceCode" id="cb607"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb607-1"><a href="glm.html#cb607-1"></a><span class="kw">mean</span>(D_exp<span class="op">$</span>time)</span></code></pre></div>
<pre><code>## [1] 3.29</code></pre>
<div class="sourceCode" id="cb609"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb609-1"><a href="glm.html#cb609-1"></a><span class="kw">var</span>(D_exp<span class="op">$</span>time)</span></code></pre></div>
<pre><code>## [1] 12.4</code></pre>
<p>As the <code>stan_glm</code> engine does not support exponential response distributions, we use <code>brm</code>, instead, and recover the parameter.</p>
<div class="sourceCode" id="cb611"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb611-1"><a href="glm.html#cb611-1"></a>M_exp &lt;-<span class="st"> </span><span class="kw">brm</span>(time <span class="op">~</span><span class="st"> </span><span class="dv">1</span>, </span>
<span id="cb611-2"><a href="glm.html#cb611-2"></a>             <span class="dt">family =</span> <span class="st">&quot;exponential&quot;</span>, </span>
<span id="cb611-3"><a href="glm.html#cb611-3"></a>             <span class="dt">data =</span> D_exp)</span></code></pre></div>
<div class="sourceCode" id="cb612"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb612-1"><a href="glm.html#cb612-1"></a><span class="kw">fixef</span>(M_exp, <span class="dt">mean.func =</span> exp)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">type</th>
<th align="left">fixef</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">fixef</td>
<td align="left">Intercept</td>
<td align="right">3.29</td>
<td align="right">2.73</td>
<td align="right">4</td>
</tr>
</tbody>
</table>
<p>Exponential disribution are rarely used in practice for two shortcomings: first, the strict mean-variance relation makes it prone to over-dispersion. This can be resolved by using observation-level random effects <a href="glm.html#olre">6.2.3.3</a> or using Gamma distributions, which accounts for extra variance by a second parameter. The second problem is the lower boundary of Zero, which will be resolved by using Exgaussian error distributions, instead.</p>
<p>Exponential regression has a single parameter and therefore has the same problem as seen with Poisson and binomial regression before. Only if all events have the same rate to occur, will an exponential distribution arise, which means for behavioural research: never. A general solution to the problem is introducing an observation-level random effect <a href="glm.html#olre">6.2.3.3</a>. Here we will tackle the problem by using continuous, zero-bounded distributions with two parameters, the Gamma family of distributions. While the two parameters rate and shape do not directly translate into location and dispersion as with Gaussian, it provides the extra degree of freedom to set them almost independently. The only limitation is that variance rises with the mean, but as we have argued in <a href="#assume-constant-variance"><strong>??</strong></a>, this is rather a desired feature than a problem. In the following, we simulate Gamma distributed observations.</p>
<div class="sourceCode" id="cb613"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb613-1"><a href="glm.html#cb613-1"></a><span class="kw">set.seed</span>(<span class="dv">20</span>)</span>
<span id="cb613-2"><a href="glm.html#cb613-2"></a>D_gam &lt;-<span class="st"> </span></span>
<span id="cb613-3"><a href="glm.html#cb613-3"></a><span class="st">  </span><span class="kw">tibble</span>(<span class="dt">Obs =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">100</span>,</span>
<span id="cb613-4"><a href="glm.html#cb613-4"></a>  <span class="dt">time =</span> <span class="kw">rgamma</span>(<span class="dv">100</span>, <span class="dt">rate =</span> <span class="dv">1</span><span class="op">/</span><span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">2</span>))</span>
<span id="cb613-5"><a href="glm.html#cb613-5"></a></span>
<span id="cb613-6"><a href="glm.html#cb613-6"></a>D_gam <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb613-7"><a href="glm.html#cb613-7"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> time)) <span class="op">+</span></span>
<span id="cb613-8"><a href="glm.html#cb613-8"></a><span class="st">  </span><span class="kw">geom_histogram</span>(<span class="dt">bins =</span> <span class="dv">10</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-88-1.png" width="90%" /></p>
<div class="sourceCode" id="cb614"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb614-1"><a href="glm.html#cb614-1"></a><span class="kw">mean</span>(D_gam<span class="op">$</span>time)</span></code></pre></div>
<pre><code>## [1] 7.61</code></pre>
<p>In comparison to the exponential distribution above, a significant difference is that the mode of the gamma distribution (its peak) is not fixed at zero, but can move along the x-axis. That makes it appear a much more realistic choice for temporal data in behavioural research. We estimate a simple gamma GMM on the simulated data. For historical reasons, <code>brm</code> uses the inverse link function (<span class="math inline">\(\theta = 1/\mu\)</span>) for Gamma regression per default, but that does not actually serve the purpose of link functions to stretch <span class="math inline">\(\mu\)</span> into the range of real numbers. Instead, we explicitly demand a log link, which makes this a multiplicative model.</p>
<div class="sourceCode" id="cb616"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb616-1"><a href="glm.html#cb616-1"></a>M_gam &lt;-<span class="st"> </span><span class="kw">brm</span>(time <span class="op">~</span><span class="st"> </span><span class="dv">1</span>, </span>
<span id="cb616-2"><a href="glm.html#cb616-2"></a>             <span class="dt">family =</span> <span class="kw">Gamma</span>(<span class="dt">link =</span> log), </span>
<span id="cb616-3"><a href="glm.html#cb616-3"></a>             <span class="dt">data =</span> D_gam)</span>
<span id="cb616-4"><a href="glm.html#cb616-4"></a></span>
<span id="cb616-5"><a href="glm.html#cb616-5"></a>M_gam</span></code></pre></div>
<div class="sourceCode" id="cb617"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb617-1"><a href="glm.html#cb617-1"></a><span class="kw">fixef</span>(M_gam, <span class="dt">mean.func =</span> exp)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">type</th>
<th align="left">fixef</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">fixef</td>
<td align="left">Intercept</td>
<td align="right">7.64</td>
<td align="right">6.7</td>
<td align="right">8.71</td>
</tr>
</tbody>
</table>
<p>Both, Exponential and Gamma distributions support the range of real numbers including zero. The weak point of both models is that they have zero as their natural starting point. As we will see in the following section, this assumption is usually violated with RT and ToT data. So, what are they good for, after all? These two models are routinely used for the <em>time intervals (TI)</em> between events that are triggered independently. In nuclear physics the individual triggers are atoms, each one <em>deciding on their own</em> when to decay. If you measure the interval between two decays the time interval is exponentially distributed. (And if you count the neutrons per time interval, the result is a Poisson distribution).</p>
<p>Analog situations can be found in service design and logistics. Take the exanmple of for customer support systems. Customers are like atoms in that their decision to file a request is usually independent from each other. Just by chance it can truly happen that two customers call the service center practically the same moment, so that the lower bound of Zero can actually be reached by some observations. Overwhelmed hotline queues do not make people happy, if they have technical problems. When planning a support system, the risk of angry customers has to be weighed against the costs of over-staffing. A good design would hit a certain sweet spot and in the ideal case there would be a predictive model of inflow rate of customers.</p>
</div>
<div id="exgauss-reg" class="section level3">
<h3><span class="header-section-number">6.3.2</span> ExGaussian regression</h3>
<p>The problem with RT and ToT data is that Zero is not a possible outcome, as any task uses up a minimum time to complete. For example, the table below shows the minimum ToT for finding the academic calendar on ten university websites (case Egan). This varies a lot between designs, but is never even close to zero. The last column puts the minimum observed ToT in relation to the observed range. On two of the websites, the offset was even larger than the obvserved range itself, hence the problem of positive lower boundaries is real in user studies.</p>
<div class="sourceCode" id="cb618"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb618-1"><a href="glm.html#cb618-1"></a><span class="kw">attach</span>(Egan)</span></code></pre></div>
<div class="sourceCode" id="cb619"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb619-1"><a href="glm.html#cb619-1"></a>D_egan <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb619-2"><a href="glm.html#cb619-2"></a><span class="st">  </span><span class="kw">filter</span>(success,</span>
<span id="cb619-3"><a href="glm.html#cb619-3"></a>         Task <span class="op">==</span><span class="st"> &quot;academic calendar&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb619-4"><a href="glm.html#cb619-4"></a><span class="st">  </span><span class="kw">group_by</span>(Task, Design) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb619-5"><a href="glm.html#cb619-5"></a><span class="st">  </span><span class="kw">summarize</span>(<span class="dt">min_time =</span> <span class="kw">min</span>(ToT),</span>
<span id="cb619-6"><a href="glm.html#cb619-6"></a>            <span class="dt">range =</span> <span class="kw">max</span>(ToT) <span class="op">-</span><span class="st"> </span>min_time,</span>
<span id="cb619-7"><a href="glm.html#cb619-7"></a>            min_time<span class="op">/</span>range) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb619-8"><a href="glm.html#cb619-8"></a><span class="st">  </span><span class="kw">kable</span>()</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">Task</th>
<th align="left">Design</th>
<th align="right">min_time</th>
<th align="right">range</th>
<th align="right">min_time/range</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">academic calendar</td>
<td align="left">VU Brussel</td>
<td align="right">21</td>
<td align="right">130</td>
<td align="right">0.162</td>
</tr>
<tr class="even">
<td align="left">academic calendar</td>
<td align="left">KU Leuven</td>
<td align="right">52</td>
<td align="right">40</td>
<td align="right">1.300</td>
</tr>
<tr class="odd">
<td align="left">academic calendar</td>
<td align="left">UGent</td>
<td align="right">8</td>
<td align="right">70</td>
<td align="right">0.114</td>
</tr>
<tr class="even">
<td align="left">academic calendar</td>
<td align="left">University of Antwerp</td>
<td align="right">3</td>
<td align="right">7</td>
<td align="right">0.429</td>
</tr>
<tr class="odd">
<td align="left">academic calendar</td>
<td align="left">UHasselt</td>
<td align="right">21</td>
<td align="right">48</td>
<td align="right">0.438</td>
</tr>
<tr class="even">
<td align="left">academic calendar</td>
<td align="left">Leiden University</td>
<td align="right">130</td>
<td align="right">181</td>
<td align="right">0.718</td>
</tr>
<tr class="odd">
<td align="left">academic calendar</td>
<td align="left">VU Amsterdam</td>
<td align="right">207</td>
<td align="right">188</td>
<td align="right">1.101</td>
</tr>
<tr class="even">
<td align="left">academic calendar</td>
<td align="left">RUG</td>
<td align="right">119</td>
<td align="right">132</td>
<td align="right">0.902</td>
</tr>
<tr class="odd">
<td align="left">academic calendar</td>
<td align="left">University Tilburg</td>
<td align="right">24</td>
<td align="right">39</td>
<td align="right">0.615</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb620"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb620-1"><a href="glm.html#cb620-1"></a><span class="kw">detach</span>(Egan)</span></code></pre></div>
<p>On the first glance, that does not seem to pose a major problem for Gamma distributions, as the left tail vanishes the more a Gamma distribution is shifted to the right, the impossible regions get smaller. However, Gamma distributions inevitably become more symmetric at larger values. A Gamma distribution far to the right has almost equally long tails and we may eventually use a Gaussian distribution, instead. As there is no separate parameter controlling the skewness of the curve it may happen that the random component captures the amount of variance, but overdoes the left tail, which introduces a bias on the coefficients. The following graphic illustrates the mean-variance-skew relationship on three Gamma distributions that move from left to right (<code>M</code>), keeping the variance constant (<code>V</code>):</p>
<div class="sourceCode" id="cb621"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb621-1"><a href="glm.html#cb621-1"></a>M =<span class="st"> </span><span class="kw">c</span>(<span class="dv">100</span>,<span class="dv">200</span>, <span class="dv">400</span>)</span>
<span id="cb621-2"><a href="glm.html#cb621-2"></a>V =<span class="st"> </span><span class="dv">8000</span></span>
<span id="cb621-3"><a href="glm.html#cb621-3"></a></span>
<span id="cb621-4"><a href="glm.html#cb621-4"></a><span class="co">## gamma</span></span>
<span id="cb621-5"><a href="glm.html#cb621-5"></a>rate =<span class="st"> </span>M<span class="op">/</span>V</span>
<span id="cb621-6"><a href="glm.html#cb621-6"></a>shape =<span class="st"> </span>rate<span class="op">^</span><span class="dv">2</span> <span class="op">*</span><span class="st"> </span>V</span>
<span id="cb621-7"><a href="glm.html#cb621-7"></a></span>
<span id="cb621-8"><a href="glm.html#cb621-8"></a><span class="kw">ggplot</span>(<span class="kw">data.frame</span>(<span class="dt">x =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">3000</span>)), <span class="kw">aes</span>(<span class="dt">x =</span> x)) <span class="op">+</span></span>
<span id="cb621-9"><a href="glm.html#cb621-9"></a><span class="st">  </span><span class="kw">stat_function</span>(<span class="dt">fun =</span> dgamma, </span>
<span id="cb621-10"><a href="glm.html#cb621-10"></a>                <span class="dt">args =</span> <span class="kw">list</span>(<span class="dt">rate =</span> rate[<span class="dv">1</span>], <span class="dt">shape =</span> shape[<span class="dv">1</span>])) <span class="op">+</span></span>
<span id="cb621-11"><a href="glm.html#cb621-11"></a><span class="st">  </span><span class="kw">stat_function</span>(<span class="dt">fun =</span> dgamma, </span>
<span id="cb621-12"><a href="glm.html#cb621-12"></a>                <span class="dt">args =</span> <span class="kw">list</span>(<span class="dt">rate =</span> rate[<span class="dv">1</span>], <span class="dt">shape =</span> shape[<span class="dv">2</span>])) <span class="op">+</span></span>
<span id="cb621-13"><a href="glm.html#cb621-13"></a><span class="st">  </span><span class="kw">stat_function</span>(<span class="dt">fun =</span> dgamma, </span>
<span id="cb621-14"><a href="glm.html#cb621-14"></a>                <span class="dt">args =</span> <span class="kw">list</span>(<span class="dt">rate =</span> rate[<span class="dv">1</span>], <span class="dt">shape =</span> shape[<span class="dv">3</span>])) <span class="op">+</span></span>
<span id="cb621-15"><a href="glm.html#cb621-15"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;ToT&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;density&quot;</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-96-1.png" width="90%" /></p>
<p>We have seen so far, that distributions with one parameter (Poisson, binomial, exponential) have a fixed relationship between location and dispersion. In order to vary location and dispersion independently, a second parameter is needed (neg-binomial, beta-binomial, Gamma, Gaussian). Only three-parameter distributions can do the trick of setting skewness separately. So called <em>exponentially modified Gaussian</em> (Exgaussian) distributions are convolutions of a Gaussian distribution and exponential distribution and have three parameters, <span class="math inline">\(\mu\)</span>, <span class="math inline">\(\sigma\)</span> and rate <span class="math inline">\(\beta\)</span>. Very roughly, the Gaussian component controls location and dispersion whereas the exponential part adjusts the skew. When <span class="math inline">\(\beta\)</span> is large in comparison to <span class="math inline">\(\mu\)</span>, the distribution is more left skewed. With this additional degree of freedom we can simulate (and estimate) distributions that are far to the right, have strong dispersion <em>and</em> strong skewness. The following plot repeats the right-furthest Gamma distribution from above and adds a Gaussian and Exgaussian distributions with all three having the exact same mean and variance.</p>
<div class="sourceCode" id="cb622"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb622-1"><a href="glm.html#cb622-1"></a>M =<span class="st"> </span><span class="dv">400</span></span>
<span id="cb622-2"><a href="glm.html#cb622-2"></a>V =<span class="st"> </span><span class="dv">8000</span></span>
<span id="cb622-3"><a href="glm.html#cb622-3"></a></span>
<span id="cb622-4"><a href="glm.html#cb622-4"></a><span class="co">## Exgaussian</span></span>
<span id="cb622-5"><a href="glm.html#cb622-5"></a>mu =<span class="st"> </span>M</span>
<span id="cb622-6"><a href="glm.html#cb622-6"></a>beta =<span class="st"> </span><span class="dv">80</span></span>
<span id="cb622-7"><a href="glm.html#cb622-7"></a>sigma =<span class="st"> </span><span class="kw">sqrt</span>(V <span class="op">-</span><span class="st"> </span>beta<span class="op">^</span><span class="dv">2</span>)</span>
<span id="cb622-8"><a href="glm.html#cb622-8"></a></span>
<span id="cb622-9"><a href="glm.html#cb622-9"></a><span class="co">## Gamma</span></span>
<span id="cb622-10"><a href="glm.html#cb622-10"></a>rate =<span class="st"> </span>M<span class="op">/</span>V</span>
<span id="cb622-11"><a href="glm.html#cb622-11"></a>shape =<span class="st"> </span>rate<span class="op">^</span><span class="dv">2</span> <span class="op">*</span><span class="st"> </span>V</span>
<span id="cb622-12"><a href="glm.html#cb622-12"></a></span>
<span id="cb622-13"><a href="glm.html#cb622-13"></a><span class="kw">ggplot</span>(<span class="kw">data.frame</span>(<span class="dt">x =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">800</span>)), <span class="kw">aes</span>(<span class="dt">x =</span> x)) <span class="op">+</span></span>
<span id="cb622-14"><a href="glm.html#cb622-14"></a><span class="st">  </span><span class="kw">stat_function</span>(<span class="dt">fun =</span> dgamma, </span>
<span id="cb622-15"><a href="glm.html#cb622-15"></a>                <span class="dt">args =</span> <span class="kw">list</span>(<span class="dt">rate =</span> rate, <span class="dt">shape =</span> shape), </span>
<span id="cb622-16"><a href="glm.html#cb622-16"></a>                <span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">colour =</span> <span class="st">&quot;Gamma&quot;</span>)) <span class="op">+</span></span>
<span id="cb622-17"><a href="glm.html#cb622-17"></a><span class="st">  </span><span class="kw">stat_function</span>(<span class="dt">fun =</span> dnorm, </span>
<span id="cb622-18"><a href="glm.html#cb622-18"></a>                <span class="dt">args =</span> <span class="kw">list</span>(<span class="dt">mean =</span> M, <span class="dt">sd =</span> <span class="kw">sqrt</span>(V)), </span>
<span id="cb622-19"><a href="glm.html#cb622-19"></a>                <span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">colour =</span> <span class="st">&quot;Gaussian&quot;</span>)) <span class="op">+</span></span>
<span id="cb622-20"><a href="glm.html#cb622-20"></a><span class="st">  </span><span class="kw">stat_function</span>(<span class="dt">fun =</span> brms<span class="op">::</span>dexgaussian, </span>
<span id="cb622-21"><a href="glm.html#cb622-21"></a>                <span class="dt">args =</span> <span class="kw">list</span>(<span class="dt">mu =</span> M, </span>
<span id="cb622-22"><a href="glm.html#cb622-22"></a>                            <span class="dt">sigma =</span> sigma,</span>
<span id="cb622-23"><a href="glm.html#cb622-23"></a>                            <span class="dt">beta =</span> beta), </span>
<span id="cb622-24"><a href="glm.html#cb622-24"></a>                <span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">colour =</span> <span class="st">&quot;Exgaussian&quot;</span>)) <span class="op">+</span></span>
<span id="cb622-25"><a href="glm.html#cb622-25"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">colour=</span><span class="st">&quot;Distribution&quot;</span>, <span class="dt">x =</span> <span class="st">&quot;ToT&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;density&quot;</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-97-1.png" width="90%" /></p>
<p>The Gamma distribution in this example starts approaching a the perfect bell curve of the Gaussian distribution. In contrast, the exgaussian distribution takes a steep left climb followed by a long right tail, which is caused by its pronounced exponential component. We do the usual exercise to simulate a grand mean model and recover the parameters with the help of the <code>brm</code> engine:</p>
<div class="sourceCode" id="cb623"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb623-1"><a href="glm.html#cb623-1"></a><span class="kw">attach</span>(Chapter_GLM)</span></code></pre></div>
<div class="sourceCode" id="cb624"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb624-1"><a href="glm.html#cb624-1"></a>D_exg &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">Y =</span> <span class="kw">rexgaussian</span>(<span class="dv">100</span>, <span class="dt">mu =</span> <span class="dv">100</span>, <span class="dt">sigma =</span> <span class="dv">20</span>, <span class="dt">beta =</span> <span class="dv">30</span>))</span>
<span id="cb624-2"><a href="glm.html#cb624-2"></a><span class="kw">qplot</span>(D_exg<span class="op">$</span>Y)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-99-1.png" width="90%" /></p>
<div class="sourceCode" id="cb625"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb625-1"><a href="glm.html#cb625-1"></a>M_exg &lt;-<span class="st"> </span><span class="kw">brm</span>(Y <span class="op">~</span><span class="st"> </span><span class="dv">1</span>,</span>
<span id="cb625-2"><a href="glm.html#cb625-2"></a>             <span class="dt">family =</span> exgaussian,</span>
<span id="cb625-3"><a href="glm.html#cb625-3"></a>             <span class="dt">data =</span> D_exg)</span></code></pre></div>
<div class="sourceCode" id="cb626"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb626-1"><a href="glm.html#cb626-1"></a><span class="kw">fixef</span>(M_exg)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">type</th>
<th align="left">fixef</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">fixef</td>
<td align="left">Intercept</td>
<td align="right">99.4</td>
<td align="right">92.9</td>
<td align="right">107</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb627"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb627-1"><a href="glm.html#cb627-1"></a><span class="kw">detach</span>(Chapter_GLM)</span></code></pre></div>
<p>Noteworthy, for Exgaussian models the brm engine uses the identity link function by default. While this is rather convenient for interpretation, it could theoretically lead to impossible predictions. As we will see later, the exgaussian is not immune, but <em>robust to impossible predictions</em> because of its tiny left tail. Any linear impact factor, like an experimental treatment can push it 150 ms to the left with insignificant risk of impossible predictions.</p>
<p>All GLM family members introduced so far are established, have been presented in many textbooks and are routinely used in research of all kinds. The Exgaussian is a newcomer. It does not come with a solid background in physical systems and may very well be considered just a hack. The following two sections examine Exgaussian models more closely and sets them in competition against Gamma and Gaussian error terms. We will be using primarily graphical methods here, but will come back these cases in chapter <a href="wwm.html#choose-dist">7.2.4</a> with a more formal approach.</p>
<!-- In chapter \@ref(crossover) we have tested Egan's assumption that user differences produce more variation in performance than designs do. In order to "normalize" the ToT outcome, log transformation was applied. Using an exgaussian regression as part of the GLM framework, we are now set to do without this humble trick. -->
<div id="rt" class="section level4">
<h4><span class="header-section-number">6.3.2.1</span> Reaction times</h4>
<p>In experimental studies, the <em>inertia of the nervous system</em> sets a limit larger than zero for reaction times. This is partly due to to some hard electrochemical and biomechanical limits of the peripheral systems. (Octopuses have decentralized nervous system for a reason!) Nerve cells and muscle fibers are slow working horses. The same goes for our minds. Reportedly, they are blazingly fast at complex tasks, such as recognizing colors and written words. Still, there always is a minimum time necessary to collect an idea from the memories and activate the surrounding nodes. Therefore, experimental reaction times have a positive minimum and it is not much of a stretch to imagine that this is a sharp limit.</p>
<p>In the Hugme case, we tried to pin down the hypothetical Geek personality. We used Need-for-cognition as a predictor for reaction times from the semantic Stroop task. Like in the original Participants must name the color of words, but these are non-color words from two categories (geek/non-geek). These words are preceded by Geek/non-geek pictures. The theory goes that a real geek, when seeing an open computer case followed by “root” will briefly reminisce and be distracted from the primary task. It did not work this way at all and we only saw a few miniscule effects. That is good news for the analysis here. The only effect was that Geek primes caused a minimal delay. Because there are no other effects, we can use a rather simple multi-level CGM to compare how Gaussian, Gamma and Exgaussian fit reaction times. Let’s take a first look some parts of the data:</p>
<div class="sourceCode" id="cb628"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb628-1"><a href="glm.html#cb628-1"></a><span class="kw">attach</span>(Hugme)</span></code></pre></div>
<p>Most of the participant-level frequency distributions of RT have a clear cut-off at around .25 seconds. The steepness of the left climb varies between participants, but some at least are rather sharp, with a right tail that is leveling off slowly. When compared to the illustrations above, it seems that an Exgaussian model could accomodate this data well.</p>
<div class="sourceCode" id="cb629"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb629-1"><a href="glm.html#cb629-1"></a>D_hugme <span class="op">%&gt;%</span></span>
<span id="cb629-2"><a href="glm.html#cb629-2"></a><span class="st">  </span><span class="kw">group_by</span>(Part) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb629-3"><a href="glm.html#cb629-3"></a><span class="st">  </span><span class="kw">filter</span>(Part <span class="op">&lt;=</span><span class="st"> </span><span class="dv">10</span>) <span class="op">%&gt;%</span></span>
<span id="cb629-4"><a href="glm.html#cb629-4"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> RT, <span class="dt">color =</span> PrimeGeek)) <span class="op">+</span></span>
<span id="cb629-5"><a href="glm.html#cb629-5"></a><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>Part, <span class="dt">nrow =</span> <span class="dv">2</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb629-6"><a href="glm.html#cb629-6"></a><span class="st">  </span><span class="kw">geom_density</span>(<span class="dt">adjust =</span> <span class="dv">2</span>) <span class="op">+</span></span>
<span id="cb629-7"><a href="glm.html#cb629-7"></a><span class="st">  </span><span class="kw">xlim</span>(<span class="dv">0</span>, <span class="fl">1.5</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-107-1.png" width="90%" /></p>
<p>Even the efect of geek primes is barely visible, but we clearly observe a left skew in most of the participants. In the following, we run three CGM models with Exgaussian, Gamma or Gaussian response distributions. For the subsequent analysis, multi-model posterior distributions and posterior predictive distributions are extracted and merged into one multi-model posterior object <code>P_1</code>.</p>
<div class="sourceCode" id="cb630"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb630-1"><a href="glm.html#cb630-1"></a><span class="kw">memory.limit</span>(<span class="dv">16000</span>)</span>
<span id="cb630-2"><a href="glm.html#cb630-2"></a></span>
<span id="cb630-3"><a href="glm.html#cb630-3"></a>F_<span class="dv">1</span> &lt;-<span class="st"> </span><span class="kw">formula</span>(RT <span class="op">~</span><span class="st"> </span><span class="dv">1</span> <span class="op">+</span><span class="st"> </span>PrimeGeek <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span> <span class="op">+</span><span class="st"> </span>PrimeGeek<span class="op">|</span>Part))</span>
<span id="cb630-4"><a href="glm.html#cb630-4"></a></span>
<span id="cb630-5"><a href="glm.html#cb630-5"></a>M_<span class="dv">1</span>_gau &lt;-<span class="st"> </span>D_hugme <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb630-6"><a href="glm.html#cb630-6"></a><span class="st">  </span><span class="kw">brm</span>(F_<span class="dv">1</span>,</span>
<span id="cb630-7"><a href="glm.html#cb630-7"></a>      <span class="dt">family =</span> gaussian,</span>
<span id="cb630-8"><a href="glm.html#cb630-8"></a>      <span class="dt">data =</span> .)</span>
<span id="cb630-9"><a href="glm.html#cb630-9"></a></span>
<span id="cb630-10"><a href="glm.html#cb630-10"></a>M_<span class="dv">1</span>_gam &lt;-<span class="st"> </span>D_hugme <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb630-11"><a href="glm.html#cb630-11"></a><span class="st">  </span><span class="kw">brm</span>(F_<span class="dv">1</span>,</span>
<span id="cb630-12"><a href="glm.html#cb630-12"></a>      <span class="dt">family =</span> <span class="kw">Gamma</span>(<span class="dt">link =</span> identity),</span>
<span id="cb630-13"><a href="glm.html#cb630-13"></a>      <span class="dt">data =</span> .)</span>
<span id="cb630-14"><a href="glm.html#cb630-14"></a></span>
<span id="cb630-15"><a href="glm.html#cb630-15"></a>M_<span class="dv">1</span>_exg &lt;-<span class="st"> </span>D_hugme <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb630-16"><a href="glm.html#cb630-16"></a><span class="st">  </span><span class="kw">brm</span>(F_<span class="dv">1</span>,</span>
<span id="cb630-17"><a href="glm.html#cb630-17"></a>      <span class="dt">family =</span> exgaussian,</span>
<span id="cb630-18"><a href="glm.html#cb630-18"></a>      <span class="dt">data =</span> .)</span>
<span id="cb630-19"><a href="glm.html#cb630-19"></a></span>
<span id="cb630-20"><a href="glm.html#cb630-20"></a>P_<span class="dv">1</span> &lt;-<span class="st"> </span><span class="kw">bind_rows</span>( </span>
<span id="cb630-21"><a href="glm.html#cb630-21"></a>  <span class="kw">posterior</span>(M_<span class="dv">1</span>_gau),</span>
<span id="cb630-22"><a href="glm.html#cb630-22"></a>  <span class="kw">posterior</span>(M_<span class="dv">1</span>_gam),</span>
<span id="cb630-23"><a href="glm.html#cb630-23"></a>  <span class="kw">posterior</span>(M_<span class="dv">1</span>_exg)</span>
<span id="cb630-24"><a href="glm.html#cb630-24"></a>)</span>
<span id="cb630-25"><a href="glm.html#cb630-25"></a></span>
<span id="cb630-26"><a href="glm.html#cb630-26"></a>T_<span class="dv">1</span>_predict &lt;-<span class="st"> </span></span>
<span id="cb630-27"><a href="glm.html#cb630-27"></a><span class="st">  </span><span class="kw">bind_rows</span>( </span>
<span id="cb630-28"><a href="glm.html#cb630-28"></a>    <span class="kw">post_pred</span>(M_<span class="dv">1</span>_gau, <span class="dt">thin =</span> <span class="dv">5</span>),</span>
<span id="cb630-29"><a href="glm.html#cb630-29"></a>    <span class="kw">post_pred</span>(M_<span class="dv">1</span>_gam, <span class="dt">thin =</span> <span class="dv">5</span>),</span>
<span id="cb630-30"><a href="glm.html#cb630-30"></a>    <span class="kw">post_pred</span>(M_<span class="dv">1</span>_exg, <span class="dt">thin =</span> <span class="dv">5</span>)</span>
<span id="cb630-31"><a href="glm.html#cb630-31"></a>  ) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb630-32"><a href="glm.html#cb630-32"></a><span class="st">  </span><span class="kw">predict</span>()</span></code></pre></div>
<p><em>Note</em> that the predictive posterior distributions runs over thousands of observation, which creates very large objects in your computer’s RAM. To prevent running into a memory limit, we crank up the memory limit (<code>memory.limit</code>) and thin out the number of posterior predictive samples by factor 5.</p>
<p>The below plot shows the population-level effects for the three models. The center estimates are very close, which means that neither of the models has a significant bias. However, the Exgaussian model produces much tighter credibility intervals. We have seen such an effect before, when on over-dispersed data, a Poisson model produced tighter intervals than the Negbinomial model. Here it is the other way round: the model with more parameters produces better levels of certainty.</p>
<div class="sourceCode" id="cb631"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb631-1"><a href="glm.html#cb631-1"></a><span class="kw">fixef</span>(P_<span class="dv">1</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb631-2"><a href="glm.html#cb631-2"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> center, <span class="dt">ymin =</span> lower, <span class="dt">ymax =</span> upper, </span>
<span id="cb631-3"><a href="glm.html#cb631-3"></a>             <span class="dt">x =</span> fixef, </span>
<span id="cb631-4"><a href="glm.html#cb631-4"></a>             <span class="dt">color =</span> model)) <span class="op">+</span><span class="st">  </span></span>
<span id="cb631-5"><a href="glm.html#cb631-5"></a><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>fixef, <span class="dt">scales =</span> <span class="st">&quot;free&quot;</span>) <span class="op">+</span></span>
<span id="cb631-6"><a href="glm.html#cb631-6"></a><span class="st">  </span><span class="kw">geom_crossbar</span>(<span class="dt">width =</span> <span class="fl">.2</span>, <span class="dt">position =</span> <span class="st">&quot;dodge&quot;</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-110-1.png" width="90%" /></p>
<!-- The following residual plot give us a hint why this might be so: Apparently, the Exgaussian model takes a much steeper left climb, whereas  -->
<p>If the Exgaussian model has a better fit, we should primarily see that in how the residuals are shaped. The Exgaussian distribution has one more degree of freedom, which can be used to set an arbitrary skew. The following reveals that the extra flexibility of the Exgaussian has been employed. Both, Gaussian and Gamma are almost symmetric, whereas the Exgausian takes a steeper left climb. The three distributions have almost the same right tail, but the left tail of the Exgaussian is smaller and the extra probability mass has moved to the center.</p>
<div class="sourceCode" id="cb632"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb632-1"><a href="glm.html#cb632-1"></a>D_hugme &lt;-<span class="st"> </span>D_hugme <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb632-2"><a href="glm.html#cb632-2"></a><span class="st">  </span><span class="kw">left_join</span>(T_<span class="dv">1</span>_predict) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb632-3"><a href="glm.html#cb632-3"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">resid =</span> RT <span class="op">-</span><span class="st"> </span>center)</span>
<span id="cb632-4"><a href="glm.html#cb632-4"></a></span>
<span id="cb632-5"><a href="glm.html#cb632-5"></a>D_hugme <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb632-6"><a href="glm.html#cb632-6"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> resid, <span class="dt">color =</span> model)) <span class="op">+</span></span>
<span id="cb632-7"><a href="glm.html#cb632-7"></a><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>PrimeGeek) <span class="op">+</span><span class="st"> </span></span>
<span id="cb632-8"><a href="glm.html#cb632-8"></a><span class="st">  </span><span class="kw">geom_density</span>() <span class="op">+</span></span>
<span id="cb632-9"><a href="glm.html#cb632-9"></a><span class="st">  </span><span class="kw">labs</span>()</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-111-1.png" width="90%" /></p>
<p>We can carefully conclude that the Exgaussian may be very useful for analyzing pychological experiments as it seems to better accomodate reaction times. Given the novelty of Exgaussian models, it is recommended that researchers carry out a careful multi-model analysis. In <a href="wwm.html#choose-dist">7.2.4</a> we will come back to this case with a more formal approach and confirm that from the three response distributions, the Exgaussian has the best predictive accuracy.</p>
<div class="sourceCode" id="cb633"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb633-1"><a href="glm.html#cb633-1"></a><span class="kw">detach</span>(Hugme)</span></code></pre></div>
</div>
<div id="tot" class="section level4">
<h4><span class="header-section-number">6.3.2.2</span> Time-on-task</h4>
<p>Experimental psychologists call the Stroop task a complex one. But, essentially it is a decision between three options and minimal processing time is rather short. Compared to tasks in usability studies, such as finding information on websites or renting cars online, this is almost nothing. Also, the complex dynamics of the task are rather different. For example, a single user error at the begin of a task sequence can have dramatic consequences, such as getting lost on a website. While ToT data also has a strictly positive lower boundary (the fastest way of achieving the goal), it often has a much wider spread and more pronounced than in RT data. In the following we will repeat the informal model comparison from the last section for ToT data.</p>
<p>We compare the three patterns of randomness on the CUE8 data set, which contains ToT measures on five tasks on a car rental website. In this study 14 professional teams took part with two conditions: remote and moderated sessions. As data from the remote condition is contaminated with cheaters, we only use the moderated sessions. In order to compare the impact of the chosen distribution on the coefficient estimates, we include the factor Task as a fixed effect (with treatment contrasts), despite this not being the most meaningful.</p>
<div class="sourceCode" id="cb634"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb634-1"><a href="glm.html#cb634-1"></a><span class="kw">attach</span>(CUE8)</span></code></pre></div>
<div class="sourceCode" id="cb635"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb635-1"><a href="glm.html#cb635-1"></a>D_cue8_mod &lt;-<span class="st"> </span>D_cue8 <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb635-2"><a href="glm.html#cb635-2"></a><span class="st">  </span><span class="kw">filter</span>(Condition <span class="op">==</span><span class="st"> &quot;moderated&quot;</span>, <span class="op">!</span><span class="kw">is.na</span>(ToT)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb635-3"><a href="glm.html#cb635-3"></a><span class="st">  </span><span class="kw">as_tbl_obs</span>()</span></code></pre></div>
<div class="sourceCode" id="cb636"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb636-1"><a href="glm.html#cb636-1"></a>F_<span class="dv">4</span> &lt;-<span class="st"> </span><span class="kw">formula</span>(ToT <span class="op">~</span><span class="st"> </span><span class="dv">1</span> <span class="op">+</span><span class="st"> </span>Task <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>Part))</span>
<span id="cb636-2"><a href="glm.html#cb636-2"></a></span>
<span id="cb636-3"><a href="glm.html#cb636-3"></a>M_<span class="dv">4</span>_gau &lt;-<span class="st"> </span>D_cue8_mod <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb636-4"><a href="glm.html#cb636-4"></a><span class="st">  </span><span class="kw">brm</span>(F_<span class="dv">4</span>,</span>
<span id="cb636-5"><a href="glm.html#cb636-5"></a>      <span class="dt">family =</span> <span class="kw">gaussian</span>(<span class="dt">link =</span> log),</span>
<span id="cb636-6"><a href="glm.html#cb636-6"></a>      <span class="dt">data =</span> ., <span class="dt">iter =</span> <span class="dv">2000</span>)</span>
<span id="cb636-7"><a href="glm.html#cb636-7"></a></span>
<span id="cb636-8"><a href="glm.html#cb636-8"></a>M_<span class="dv">4</span>_exg &lt;-<span class="st"> </span>D_cue8_mod <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb636-9"><a href="glm.html#cb636-9"></a><span class="st">  </span><span class="kw">brm</span>(F_<span class="dv">4</span>,</span>
<span id="cb636-10"><a href="glm.html#cb636-10"></a>      <span class="dt">family =</span> <span class="kw">exgaussian</span>(<span class="dt">link =</span> log),</span>
<span id="cb636-11"><a href="glm.html#cb636-11"></a>      <span class="dt">data =</span> ., <span class="dt">iter =</span> <span class="dv">2000</span>)</span>
<span id="cb636-12"><a href="glm.html#cb636-12"></a></span>
<span id="cb636-13"><a href="glm.html#cb636-13"></a>M_<span class="dv">4</span>_gam &lt;-<span class="st"> </span>D_cue8_mod <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb636-14"><a href="glm.html#cb636-14"></a><span class="st">  </span><span class="kw">brm</span>(F_<span class="dv">4</span>,</span>
<span id="cb636-15"><a href="glm.html#cb636-15"></a>      <span class="dt">family =</span> <span class="kw">Gamma</span>(<span class="dt">link =</span> log),</span>
<span id="cb636-16"><a href="glm.html#cb636-16"></a>      <span class="dt">data =</span> ., <span class="dt">iter =</span> <span class="dv">2000</span>)</span>
<span id="cb636-17"><a href="glm.html#cb636-17"></a></span>
<span id="cb636-18"><a href="glm.html#cb636-18"></a>P_<span class="dv">4</span> &lt;-<span class="st"> </span><span class="kw">bind_rows</span>( </span>
<span id="cb636-19"><a href="glm.html#cb636-19"></a>  <span class="kw">posterior</span>(M_<span class="dv">4</span>_gau),</span>
<span id="cb636-20"><a href="glm.html#cb636-20"></a>  <span class="kw">posterior</span>(M_<span class="dv">4</span>_gam) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">value =</span> <span class="kw">if_else</span>(value <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;fixef&quot;</span>, <span class="st">&quot;ranef&quot;</span>), <span class="kw">exp</span>(value), value)),</span>
<span id="cb636-21"><a href="glm.html#cb636-21"></a>  <span class="kw">posterior</span>(M_<span class="dv">4</span>_exg)</span>
<span id="cb636-22"><a href="glm.html#cb636-22"></a>)</span>
<span id="cb636-23"><a href="glm.html#cb636-23"></a></span>
<span id="cb636-24"><a href="glm.html#cb636-24"></a></span>
<span id="cb636-25"><a href="glm.html#cb636-25"></a>T_<span class="dv">4</span>_predict &lt;-<span class="st"> </span><span class="kw">bind_rows</span>( </span>
<span id="cb636-26"><a href="glm.html#cb636-26"></a>  <span class="kw">post_pred</span>(M_<span class="dv">4</span>_gau, <span class="dt">thin =</span> <span class="dv">5</span>),</span>
<span id="cb636-27"><a href="glm.html#cb636-27"></a>  <span class="kw">post_pred</span>(M_<span class="dv">4</span>_gam, <span class="dt">thin =</span> <span class="dv">5</span>),</span>
<span id="cb636-28"><a href="glm.html#cb636-28"></a>  <span class="kw">post_pred</span>(M_<span class="dv">4</span>_exg, <span class="dt">thin =</span> <span class="dv">5</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb636-29"><a href="glm.html#cb636-29"></a><span class="st">  </span><span class="kw">predict</span>()</span></code></pre></div>
<p><em>Note</em> that the Gamma model caused trouble when estimated with an identity link. For this reason, all three models were estimated on a log-linearized scale. This makes the back-transformed coefficients multiplicative, which actually makes more sense, as we have seen in <a href="glm.html#speaking-multipliers">6.2.1.1</a>.</p>
<div class="sourceCode" id="cb637"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb637-1"><a href="glm.html#cb637-1"></a><span class="kw">fixef</span>(P_<span class="dv">4</span>, <span class="dt">mean.func =</span> exp) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb637-2"><a href="glm.html#cb637-2"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> center, <span class="dt">ymin =</span> lower, <span class="dt">ymax =</span> upper, </span>
<span id="cb637-3"><a href="glm.html#cb637-3"></a>             <span class="dt">x =</span> <span class="st">&quot; &quot;</span>, </span>
<span id="cb637-4"><a href="glm.html#cb637-4"></a>             <span class="dt">color =</span> model)) <span class="op">+</span><span class="st">  </span></span>
<span id="cb637-5"><a href="glm.html#cb637-5"></a><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>fixef, <span class="dt">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="dt">nrow =</span> <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb637-6"><a href="glm.html#cb637-6"></a><span class="st">  </span><span class="kw">geom_crossbar</span>(<span class="dt">width =</span> <span class="fl">.2</span>, <span class="dt">position =</span> <span class="st">&quot;dodge&quot;</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-117-1.png" width="90%" /></p>
<p>In the pr4evious section, we have seen, that the three models agreed on the center estimates. For the ToT data here, three models produce rather different coefficients. It seems that the models disgree on all but the Intercept.</p>
<p>Inspecting the residual distributions yields a similar patter, once again: there is visible left skew, and the Exgaussian model has the sharpest left rise.</p>
<div class="sourceCode" id="cb638"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb638-1"><a href="glm.html#cb638-1"></a><span class="kw">left_join</span>(T_<span class="dv">4</span>_predict, D_cue8_mod, <span class="dt">by =</span> <span class="st">&quot;Obs&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb638-2"><a href="glm.html#cb638-2"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">resid =</span> ToT <span class="op">-</span><span class="st"> </span>center) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb638-3"><a href="glm.html#cb638-3"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> resid, <span class="dt">color =</span> model)) <span class="op">+</span></span>
<span id="cb638-4"><a href="glm.html#cb638-4"></a><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>Task, <span class="dt">nrow =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb638-5"><a href="glm.html#cb638-5"></a><span class="st">  </span><span class="kw">geom_density</span>(<span class="dt">adjust =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-118-1.png" width="90%" /></p>
<div class="sourceCode" id="cb639"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb639-1"><a href="glm.html#cb639-1"></a><span class="kw">detach</span>(CUE8)</span></code></pre></div>
<p>In general, it seems that Exgaussian models for RT and ToT accomodates left skewness better and produces estimates that are more conservative and certain at the same time. But, These are just informal comparisons. In chapter <a href="wwm.html#wwm">7</a>, we will apply formal criteria for selecting between distributions. As will turn out, Gamma distribution is the preferred distribution for ToT in CUE8.</p>
<!-- Could it be true, that Gaussian and Gamma models overestimate group mean differences for left skewed RT and ToT responses? We examine this possibility by simulating a well-known experiment using an exgaussian distribution. -->
<!-- #### MAKE Stroop -->
<!-- ```{r} -->
<!-- sim_Stroop <- function(beta_0 = 500, # congruent -->
<!--                        beta_1 = 50,  # neutral -->
<!--                        beta_2 = 120, # incongr -->
<!--                        sigma = 20,   -->
<!--                        beta = 70, -->
<!--                        N = 400,      # obs per condition -->
<!--                        seed = 42){ -->
<!--   set.seed(seed) -->
<!--   tibble(Condition = rep(c("con", "neu", "inc"), N), -->
<!--              mu = beta_0 + beta_1 * (Condition == "neu") + beta_2 * (Condition == "inc"), -->
<!--              RT = rexgaussian(N * 3, mu, sigma, beta)) %>%  -->
<!--     as_tbl_obs() -->
<!--   } -->
<!-- D_stroop <- sim_Stroop() -->
<!-- D_stroop %>%  -->
<!--   group_by(Condition) %>%  -->
<!--   summarize(mean(RT)) %>%  -->
<!--   kable() -->
<!-- D_stroop %>%  -->
<!--   ggplot(aes(x = RT, col = Condition)) + -->
<!--   geom_density() -->
<!-- ``` -->
<!-- In the same way as above, we recover the effects by three models that have the same likelihood, but differ in their response distribution. -->
<!-- ```{r opts.label = "mcmc"} -->
<!-- F_stroop <- formula(RT ~ Condition) -->
<!-- M_stroop_gau <- D_stroop %>%  -->
<!--   brm(F_stroop, -->
<!--       family = gaussian, -->
<!--       data = .) -->
<!-- M_stroop_exg <- D_stroop %>%  -->
<!--   brm(F_stroop, -->
<!--       family = exgaussian, -->
<!--       data = .) -->
<!-- M_stroop_gam <- D_stroop %>%  -->
<!--   brm(F_stroop, -->
<!--       family = Gamma(link = identity), -->
<!--       data = .) -->
<!-- P <- bind_rows(  -->
<!--   posterior(M_stroop_gau), -->
<!--   posterior(M_stroop_gam), -->
<!--   posterior(M_stroop_exg) -->
<!-- ) -->
<!-- T_stroop_predict <- bind_rows(  -->
<!--   post_pred(M_stroop_gau, thin = 4), -->
<!--   post_pred(M_stroop_gam, thin = 4), -->
<!--   post_pred(M_stroop_exg, thin = 4) -->
<!-- ) %>%  -->
<!--   predict() -->
<!-- ``` -->
<!-- ```{r opts.label = "mcsync"} -->
<!-- sync_CE(Chapter_GLM, M_stroop_gau, M_stroop_gam, M_stroop_exg, P_stroop, T_stroop_predict, D_stroop) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- fixef(P_stroop) %>%  -->
<!--   ggplot(aes(y = fixef, x = center, xmin = lower, xmax = upper, color = model)) + -->
<!--   facet_wrap(~fixef, ncol = 1, scales = "free") + -->
<!--   geom_point(size = 2) + -->
<!--   geom_errorbarh() -->
<!-- ``` -->
<!-- What is confirmed by the simulation is that the exgaussian model, when it is the true one, produces more certain estimates.  -->
<!-- ```{r} -->
<!-- D_stroop  %>%    -->
<!--   left_join(T_stroop_predict) %>%  -->
<!--   mutate(resid = RT - center) %>%  -->
<!--   ggplot(aes(x = resid, color = model)) + -->
<!--   facet_wrap(~Condition) + -->
<!--   geom_density(adjust = 2) + -->
<!--   xlim(-200, 350) -->
<!-- ``` -->
<!-- Different to what has been observed above, the shapes of the residual distributions do not differ much, except for a shift to the right. What could be accountable for that is that the simulation only contained three homogenous groups, rather than the many groups in the previous multi-level data sets (random factors). It remains to be clarified what precisely the biases and drags are caused by, ill-specified response distribution for RT or ToT in complex research designs. Despite these question marks, it has been confirmed that the superior certainty of estimates is not just an artifact of the exgaussian model, but is real and likely to make quantitative inference from RT and ToT data more efficient. -->
<p>One last issue remains to get clarified: using the identity link for Exgaussian models is very convenient and is probably much safer as compared to Gaussian models with their longer left tails. Still, impossible predictions can arise. But, how much of a risk is there? We check this on the posterior predictive distributions of both studies, CUE8 and Hugme. The following table shows the proportion of observations, that get a negative 2.5% credibility limit assigned.</p>
<div class="sourceCode" id="cb640"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb640-1"><a href="glm.html#cb640-1"></a><span class="kw">bind_rows</span>(Hugme<span class="op">$</span>T_<span class="dv">1</span>_predict, CUE8<span class="op">$</span>T_<span class="dv">4</span>_predict) <span class="op">%&gt;%</span></span>
<span id="cb640-2"><a href="glm.html#cb640-2"></a><span class="st">  </span><span class="kw">group_by</span>(model) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb640-3"><a href="glm.html#cb640-3"></a><span class="st">  </span><span class="kw">summarize</span>(<span class="kw">mean</span>(lower <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb640-4"><a href="glm.html#cb640-4"></a><span class="st">  </span><span class="kw">kable</span>()</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">model</th>
<th align="right">mean(lower &lt; 0)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">M_1_exg</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">M_1_gam</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">M_1_gau</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">M_4_exg</td>
<td align="right">0.015</td>
</tr>
<tr class="odd">
<td align="left">M_4_gam</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">M_4_gau</td>
<td align="right">0.712</td>
</tr>
</tbody>
</table>
<p>For our RT data, impossible predictions is not a big issue with any of the models, as all 2.5% quantiles are positive. That is different for ToT: while the gamma model is inherently immune to negative predictions, the exgaussian model produced a few impossible lower 2.5% limits (around 3%). The Gaussian model is extremely off: more than 70% of all predictions have impossible lower 2.5% limits.</p>
<p>In the scientific literature, the coverage on what random pattern to use for RT and ToT data is meager at this moment. Probably, that is due to the lack of user-friendly engines supporting the more exotic GLM family members, gamma or exgaussian regression. The brms engine covers a much broader set of distributions than any other implementation before and researchers have the choice. This chapter attempted to provide theoretical arguments as well as empirical indications that the exgaussian regression can be a better choice than Gaussian and gamma. First of all, it accomodates the strong left skew of RT and ToT much better than the gamma, which takes a too symmetric form when far from the left boundary. Second, it is reasonably robust to impossible predictions, even when using the convenient identity link function. Third, and that is almost too good to be true, it massively improves certainty in predictors. It seems that Exgaussian models are more efficient for carving out delicate effects in experimental studies.</p>
<p>However, as the discussion has not even fully started, to declare it settled would be premature. In contrast, the aim of this section was to illustrate a semi-formal approach that reseachers can follow to choose among the candidate models for their specific RT and ToT data. Data from other RT paradigms might take different shapes. For example, when measuring RT by events in EEG signals (rather than actual key presses), motor time plays a much smaller role, pushing RTs closer to the left boundary. Then, the exgaussian model might produce higher rates of impossible predictions and the gamma model could sufficiently accomodate the left skewness. Note that even using a log link on the exgaussian model can produce visits to the negative range. When both accomodate the left skew equally well, the gamma model is to be preferred as it never produces impossible predictions and is more parsimomous.</p>
<!-- That being said, the brms engine offers even more opportunities. First, it supports two more distributions with an offset component: the shifted log-normal and the Wiener distribution. Interestingly, the latter grounds on one of the few formally specified cognitive process models, the diffusion model for simple choice tasks. All four parameters of the Wiener distribution are directly linked to individual elements of the cognitive process. This  brings us to another  extension of Brms, which I will briefly cover in the last section of this chapter. *Distributional models* put linearized predictor terms not just on the distribution mean, but also on other parameters on . The vast majority of statistical analysis capitalizes on the location parameters. We ask whether an increase in a continuous predictor causes an increase in the outcome or if one group has a higher average than the other.  -->
<!-- Only in the analysis of random effects have we drawn conclusions from dispersion parameters, such as to test Egans claim. In design research, the variance in performance is a crucial issue. To give another example: reportedly, several areas of cognitive functioning deteriorate with age, on average, but variance typically increases. It was the very idea of Dennis Egan that designs should not just improve performance on average, but also keep variance at a minimum. Hence, linking variance to predictors, such as design and age, can be a fruitful endeavour under the paradigm of robust designs. For RT the beforementioned Wiener distribution matches RTs in simple choice tasks and every parameter corresponds with an element in the so called  diffusion model. In design research, such ideas are almost unexplored. Perhaps, one day a researcher finds that the gaussian and the exponential component are influenced by different design features. -->
<!-- ### Literature -->
<!--The theoretical thoughts at the beginning of the chapter spoke of two components of tasks that have different properties: the motor part is slow, but higly predictable, whereas the cognitive part is very fast, but performance varies a lot. Possibly, motor time could be mapped to the gaussian component and the cognitive time be assigned to the exponential component. In a distributional model, all parameters of the distribution can get their own set of predictors. For example, in a Stroop experiment, the exgaussian $\beta$ could  be linked to the experimental conditions, whereas $\mu$ gets a random effect to accomodate individual differences in motor speed.-->
<!--
#### Dealing with upper boundaries

As we have seen, the exgaussian linear model gracefully deals with lower boundaries larger than zero. However, so far it maintains the assumption that there is no upper boundary for durations. This assumption is uncritical when the empirical setup gives all participants enough headroom to finish their tasks. When most durations are rather short, fractions of a second to just a few minutes, one does not mind waiting for a few that take longer. However, in user research tasks are often rather complex and purportedly routine tasks, like renting a car online can be surprisingly long-running. In order to reduce the risk of user frustration, user studies are often imposing a limit for task completion, say five minutes, and give participants the freedom to give up whenever they want. As a consequence, there will be a number of *censorded* observations. These cannot be taken at face value, such as "user $i$ took five minutes  to complete task $j$", but as a minimum for an otherwise unknown duration: "it would have taken user $i$ *at least* five minutes to complete the task".

Had the researchers in the CUE8 study limited the time per task to ten minutes, the distribution would take the following shape:
-->
<!--
For tasks 1 and 4, the smooth running right tail of the distribution is replaced by a sudden peak. So, how to deal with such observations? A brute-force approach would be to regard them as outliers and completely remove them from the data. However, that would induce a serious bias as those observations are real and they are even the most crucial to pin-point problems in the design. In consequence, one would get a highly optimistic estimate for ToT. The optimistic bias would be reduced, but not removed, if one would take the censored observations at face value. In addition to the location bias, all dispersion estimates (residuals, random effects variance) will be underestimated, too.

The `brm` engine allows to specify censored #114 -->
<!--observations by another formula extension, *additional response information*. For censoring, a Boolean indicator variable needs to be added to the data frame  (`cens`, see above) and the formula specified as follows: 

-->
<!--
We compare the results of the (artificially) censored data with the original. At least for this data set, I had to seriously crank up the number of iterations to obtain adequatly effective samples.-->
<!--
In the above example, we have used a generic censoring rule for all observations, but in fact, all individual observations may have their own limit, which is very convenient as in user studies participants typically have the freedom to give up at any time they want and you never know in advance when that is. Furthermore, the censoring function can also deal with lower bound censoring and even interval censoring, but it is difficult to come up with convincing examples from design research.

When observations are censored, the regression engine interpolates the missing parts from the random distribution. Review the comparison of residuals from the previous section. It is the real observations on the long right tail, that drag the notoriously symmetric Gaussian distribution into some skewness. If the far right observations were not available due to censoring, a Gaussian model would snap back to a more symmetric form. Therefore, when data is censored an adequate distributional choice is more important than ever.

-->
<!-- ### Exercises: -->
<!-- 1. Review the literature on reaction times. What is the shortest time length you can find? <!-- #113 ->  -->
<!-- 1. The user can only do things one after the other and therefore, RT and ToT will never come close to zero. Conceive examples of independently triggered events that *happen* to users and effect user satisfaction.  -->
</div>
</div>
</div>
<div id="rating-scales" class="section level2">
<h2><span class="header-section-number">6.4</span> Rating scales</h2>
<p>In classic design research of the last millenium, die-hard Human Factors researchers have mainly been asking objectively sounding questions, like:</p>
<ul>
<li>Can a user achieve accurate results with the system?</li>
<li>Can they do so in less time?</li>
<li>Is the number of errors reasonably confined?</li>
</ul>
<p>For professional, and especially critical, systems, these are highly valid questions, indeed. The purpose of a medical infusion pump is to improve the health status of a patient by accurately and reliably delivering medication into the bloodstream and the extent to which this can be measured dircetly.</p>
<p>However, starting with the 1990s, wave after wave of novel electronic entertainment systems and digital gadgets rolled over the comsumer mass market. The purpose of a video recorder or a smartphone is to deliver joy … and to be sold in large batches. The sole purpose of a commercial website is to sell and nothing else. With the new millenium, design researchers began to recognize what consumer psychologists had discovered two decades earlier: users are not rational decision makers in a utilitarian sense. When people decide to adopt (or buy) a new system, this is only partly driven by their expectation of productivity. These additional expectations are commonly called <em>hedonistic values</em> and cover a broad class of human needs, such as:</p>
<ul>
<li>positive emotions</li>
<li>expression of self</li>
<li>social connectedness</li>
<li>aesthetic perception</li>
<li>personal growth</li>
</ul>
<p>Whether or not these concepts are well-defined from a psychological point-of-view is beyond the scope of this book. What matters is that these concepts are so elusive that the most sincere researchers have not yet found objective criteria to measure them. Instead, almost everyone resorts to use of <em>self-report rating scales</em>, like this one:</p>
<p>How beautiful do you perceive the user interface to be?</p>
<p>unattractive 1 – 2 – 3 – 4 – 5 a piece of art</p>
<p>If you use a 5 point rating scale like this one to measure perceived beauty, participants have to convert their gut feeling into a number, which involves the following three processes somewhere in their minds:</p>
<ol style="list-style-type: decimal">
<li>anchoring</li>
<li>introspection</li>
<li>binning</li>
</ol>
<p>By <em>anchoring</em> participants establish an idea of how ugly or beautiful something has be to get an extreme rating of 1 or 5. These imaginary endpoints define the <em>absolute range</em> of the rating scale. The researcher might early on give explicit or implicit cues to let the participant guess the range the researcher has in mind. If an experiment is overtly about web design, then probably “very ugly” means the least attractive commercial website the participant can think of. However, participants old enough to remember web design in its infancy (say the early attempts of disney.com), may end up with a lower anchor than today’s kids. If too few cues are given upfront, participants will probably adjust their anchors to the stimuli they see throughout the experiment. Probably, it will make a difference for what 1 or 5 mean, when the set of stimuli contain just websites, or websites <em>and</em> impressionist paintings <em>and</em> screenshots from 1980 splatter <!-- #105 --> movies.</p>
<p>By <em>introspection</em> participants intuitively assess the intensity of their <em>real feelings</em> as compared to the anchors. Reportedly, feelings are influenced by:</p>
<ol style="list-style-type: decimal">
<li>visual simplicity</li>
<li>prototypicality</li>
<li>second exposure</li>
<li>Gestalt principles</li>
<li>fluency of processing</li>
<li>attribute substitution heuristics</li>
<li>color aesthetics</li>
<li>fashion</li>
<li>previous stimuli</li>
<li>current mood</li>
<li>a person’s history</li>
<li>and cultural background <!-- #104 --></li>
</ol>
<p>By <em>binning</em> the participant mentally divides the absolute range into five categories that are either fuzzy or defined by stereotypes, like “It must look at least as elegant as a certain other website to get a 4.”</p>
<p>As the outcome of anchoring, introspection and binning are not under the control of the researcher, the response patterns can vary between participants. Let’s consider a few possible patterns of participants (and their dramatic stories):</p>
<ol style="list-style-type: decimal">
<li>A is undecisive and stays in the center region</li>
<li>B has a crush on the experimenter and responds slightly more positive</li>
<li>C politely avoids the negative extremes</li>
<li>D is a human rights activist and habitually treats the seven bins equally.</li>
<li>E is annoyed by the experiment (rightly so) and falls into a dichotomous response pattern: 1 or 5</li>
<li>F is a surrealist and has a completely unique way to “look at things”.</li>
</ol>
<p>Many unkowns are in the game. Only one special case we can alleviate with the multilevel modelling tools in our hands. Anchoring can (but not necessarily does) result in a constant <em>shift</em> between participants. Compare participants A and B: A collects almost all stimuli in categories 2, 3 and 4, whereas B uses 3, 4 and 5. This is not much else than a participant-level intercept random effect. Unfortunately, the situation can be more difficult than that. When participants differ in how extreme they set their endpoints, like C and D, than their responses will differ in variance. The maximum variance, however, will be found in participant E.</p>
<p>To speak of a real case: In the IPump study, a single-item rating scale was used to measure mental workload. The results suggest that all participants used the lower range of the scale, but differed vastly in where they set their upper point. Figure XY orders participants by the maximum value they used. This is obviously related to variance, but seemingly not so much with location. It does not suffice to use a response distribution with mean-variance relationship, as we used to. All these issues make rating scales peculiar and we should not pretend they have the same neat arithmetic properties as objective measures.</p>
<div class="sourceCode" id="cb641"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb641-1"><a href="glm.html#cb641-1"></a><span class="kw">attach</span>(IPump)</span>
<span id="cb641-2"><a href="glm.html#cb641-2"></a></span>
<span id="cb641-3"><a href="glm.html#cb641-3"></a>D_pumps <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb641-4"><a href="glm.html#cb641-4"></a><span class="st">  </span><span class="kw">group_by</span>(Part) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb641-5"><a href="glm.html#cb641-5"></a><span class="st">  </span><span class="kw">summarize</span>(<span class="dt">min =</span> <span class="kw">min</span>(workload),</span>
<span id="cb641-6"><a href="glm.html#cb641-6"></a>            <span class="dt">max =</span> <span class="kw">max</span>(workload),</span>
<span id="cb641-7"><a href="glm.html#cb641-7"></a>            <span class="dt">median =</span> <span class="kw">median</span>(workload)) <span class="op">%&gt;%</span></span>
<span id="cb641-8"><a href="glm.html#cb641-8"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Part_ord =</span> <span class="kw">rank</span>(max, <span class="dt">ties.method =</span> <span class="st">&quot;first&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb641-9"><a href="glm.html#cb641-9"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Part_ord, <span class="dt">ymax =</span> max, <span class="dt">ymin =</span> min, <span class="dt">y =</span> median)) <span class="op">+</span></span>
<span id="cb641-10"><a href="glm.html#cb641-10"></a><span class="st">  </span><span class="kw">geom_errorbar</span>() <span class="op">+</span></span>
<span id="cb641-11"><a href="glm.html#cb641-11"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-128-1.png" width="90%" /></p>
<div class="sourceCode" id="cb642"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb642-1"><a href="glm.html#cb642-1"></a><span class="kw">detach</span>(IPump)</span></code></pre></div>
<p>Setting the idiosyncratic rating scale responses aside, how does a common rating scale appear in our framework of link functions and patterns of randomness? Rating scales are bounded on two sides and we already know what that means: a suitable model for rating scales will likely contain a logit link function and a distribution of randomness that is bounded on two sides.</p>
<p>A real problem with rating scales is that they often are discrete. Most scales force participants to give their answer as a choice between five or seven ordered levels. When the response variable has just a few levels, <em>ordinal regression</em> is a good choice. Ordinal regression itself is a generalization of logistic regression.</p>
<p>However, with most properly designed self-report instruments, a scale comprises several items, because only that can sufficiently reduce measurement error and allow for in-depth psychometric assessments. Take the well-known Cronbach <span class="math inline">\(\alpha\)</span>, which assesses reliability of a scale by correlating every items score with the sum score. Obviously, that only makes sense when there are multiple items.
While from a psychometric perspective, single-item scales are susceptible, there can be situations where a researcher may use a validated single item scale for pragmatic reasons. Especially, when measures happen in situ, such as during a usability test or even a real operation, being brief and unobtrusive might be more important than good quality of measures.</p>
<!-- 40 write a separate subsection on multilevel models with multi-item scales -->
<p>With multi-item rating scales, one also has the possibility to build a psychometric multi-level model, where items are considered a sample of a population of possible items. That is actually a very good idea, as the item-level random effects control for differences in item location. For example, the following item is likely to produce generally lower beauty ratings than the one shown earlier, because the anchors have been moved downwards:</p>
<p>How beautiful do you perceive the user interface?</p>
<p>like a screenshot from a splatter <!-- #105 -->movie 1 – 2 – 3 – 4 – 5 quite attractive</p>
<p>Unless one builds such a psychometric multi-level model, ordinal regression is not very suitable for multi-item scales and here is why: The sum (or mean) score is still binned, but more finely grained. A sum score over three seven-binned items already has 21 bins, which would result in an inflation of number of parameters in ordinal regression.</p>
<p>As a rescue, one might well regard a measure with 21 bins as continuous <!-- #115 -->. Furthermore, there actually is no strong reason to use binned rating scales at all. So called <em>visual analog scales</em> let participants make continuous choices by either drawing a cross on a line or move a slider control. For sum scores and visual analogue scales, the problem of choice reduces to a logit link function (they still have two boundaries) and a continuous distribution bounded on both sides. That is precisely what is behind <em>beta regression</em> and, as we shall see, this distribution is flexible enough to smooth over several of the rating scale pathologies that were just discussed.</p>
<!--Moreover, they are not directly linked 
to the purpose. When testing a novel design of a commercial website against the current version
-->
<div id="ord-logist-reg" class="section level3">
<h3><span class="header-section-number">6.4.1</span> Ordered logistic regression</h3>
<p>When the ordinal response has a low number of response categories (between 4 and 7), ordinal regression applies. Recall logistic regression: the response falls into one of two categories, which are coded as 0 and 1. Although not in a strict sense, the two categories can often be thought of as in an order: success is better than failure, presence more than absence and a return better than staying away. Instead of two categories, we can also conceive the situation as a <em>threshold</em> between the categories, that needs force to jump over it. Any positive impact factor <span class="math inline">\(x_i\)</span> can then be thought of as such a force that pushes a response probability to the higher category, by the amount <span class="math inline">\(\beta_i\)</span> (on logit scale). At the same time, the intercept <span class="math inline">\(\beta_0\)</span> represents the basic log-odds of falling into category 1 in a default state, that is <span class="math inline">\(x_i = 0\)</span>.</p>
<p>In ordinal regression, this idea extends to cases with more than two ordered response categories. The only arising complication is that with two categories, we have one <em>threshold</em> to overcome, whereas with three categories there are two thresholds and generally, with <span class="math inline">\(c\)</span> categories, there are <span class="math inline">\(c - 1\)</span> thresholds. Ordinal regression deals with the problem by estimating <span class="math inline">\(c - 1\)</span> intercept estimates <span class="math inline">\(\beta_{0[k]}\)</span>. Each threshold intercept <span class="math inline">\(\beta_{0[k]}\)</span> represents the probability (on logit scale) that the response falls into category <span class="math inline">\(k\)</span> <em>or lower</em>, or formally:</p>
<p><span class="math display">\[
\text{logit}(P(y_i \leq k)) = \beta_{0[k]}
\]</span></p>
<p>Let’s see this at the example of the BrowsingAB case, first. User ratings have been simulated with seven levels:</p>
<div class="sourceCode" id="cb643"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb643-1"><a href="glm.html#cb643-1"></a><span class="kw">attach</span>(BrowsingAB)</span></code></pre></div>
<div class="sourceCode" id="cb644"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb644-1"><a href="glm.html#cb644-1"></a>BAB1 <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb644-2"><a href="glm.html#cb644-2"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> rating)) <span class="op">+</span></span>
<span id="cb644-3"><a href="glm.html#cb644-3"></a><span class="st">  </span><span class="kw">facet_grid</span>(Design<span class="op">~</span>.) <span class="op">+</span></span>
<span id="cb644-4"><a href="glm.html#cb644-4"></a><span class="st">  </span><span class="kw">geom_histogram</span>() <span class="op">+</span></span>
<span id="cb644-5"><a href="glm.html#cb644-5"></a><span class="st">  </span><span class="kw">xlim</span>(<span class="dv">1</span>,<span class="dv">7</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-130-1.png" width="90%" /></p>
<p>The brms regression engine implements ordinal regression by the family <code>cratio</code> (cumulative odds ratio <!-- #116 -->) with a default logit link function.</p>
<div class="sourceCode" id="cb645"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb645-1"><a href="glm.html#cb645-1"></a>M_ord_<span class="dv">1</span> &lt;-<span class="st"> </span></span>
<span id="cb645-2"><a href="glm.html#cb645-2"></a><span class="st">  </span>BAB1 <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb645-3"><a href="glm.html#cb645-3"></a><span class="st">  </span><span class="kw">brm</span>(rating <span class="op">~</span><span class="st"> </span>Design, </span>
<span id="cb645-4"><a href="glm.html#cb645-4"></a>      <span class="dt">family =</span> <span class="st">&quot;cratio&quot;</span>, </span>
<span id="cb645-5"><a href="glm.html#cb645-5"></a>      <span class="dt">data =</span> .) </span></code></pre></div>
<!--We have to keep in mind, though, that the constant shift of thresholds is an assumption [EXPLAIN ASSUMPTION FIRST EXPLICITLY], which may fail. For example, when participants choose different anchor points for their judgements, such as in the IPump study, this is no longer the case. [WHY NOT?] -->
<div class="sourceCode" id="cb646"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb646-1"><a href="glm.html#cb646-1"></a><span class="kw">fixef</span>(M_ord_<span class="dv">1</span>)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">fixef</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">NA</td>
<td align="right">-13.411</td>
<td align="right">-53.335</td>
<td align="right">-5.928</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="right">-13.326</td>
<td align="right">-53.333</td>
<td align="right">-5.896</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="right">-4.053</td>
<td align="right">-5.020</td>
<td align="right">-3.251</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="right">-2.281</td>
<td align="right">-2.778</td>
<td align="right">-1.796</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="right">-1.121</td>
<td align="right">-1.525</td>
<td align="right">-0.734</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="right">1.321</td>
<td align="right">0.846</td>
<td align="right">1.851</td>
</tr>
<tr class="odd">
<td align="left">DesignB</td>
<td align="right">-0.856</td>
<td align="right">-1.317</td>
<td align="right">-0.410</td>
</tr>
</tbody>
</table>
<p>The six intercepts correspond with the thresholds between the seven levels. It is no coincidence that the intercept estimates increase by order, as they are cumulative (the “c” in cratio). The first intercept estimate represents (the logit of) the proportion of responses <span class="math inline">\(y_i \leq 1\)</span>, the second <span class="math inline">\(y_i \leq 2\)</span> etc.
The Design effect has the usual interpretation as compared to logistic regression, an increase in logit. The only difference is that it refers to all six reference points. The expected proportion of responses equal to or smaller than 2 for design A is:</p>
<p><span class="math display">\[
\begin{aligned}
\pi(y_i \leq 2|A) &amp;= 
\text{logit}^{-1}(\beta_{0[2]}) \\ 
&amp;=\text{logit}^{-1}(-13.3) = 
1.674\times 10^{-6}
\end{aligned}
\]</span></p>
<p>The expected proportion of responses equal to or smaller than 2 for design B we get by the usual linear combination:</p>
<p><span class="math display">\[
\begin{aligned}
\pi(y_i \leq 2|B) &amp;= \text{logit}^{-1}(\beta_{0[2]} + \beta_1) \\
&amp;= 
\text{logit}^{-1}(-14.2) = 
6.808\times 10^{-7}
\end{aligned}
\]</span></p>
<p>All coefficients are shifting all thresholds by the same amount (on the linear predictor scale). You can picture this as a single puppetier controlling multiple puppets by just one stick, making them dance synchronously. As long as the ordinal scale has only a low number of bins, that keeps the number of parameters at a reasonable level. Just imagine, you were estimating an ordinal multilevel model and all participant-level effects were five or seven-folded, too. However, the equidistancy of effects on bin thresholds is an assumption by itself, and in the presence of response styles on rating scales, it cannot be taken for granted <!-- #117 -->.</p>
<p>Besides that, the ordinal model appears very snug to the structure of the data. It does not wipe over the fact that the response is discrete and the thresholds represent the order. Conveniently, effects are represented by a single estimate, which one can use to communicate direction and certainty of effects. On the downside, communicating absolute performance (that is, including the intercept) is more complicated. When presenting predictions from an ordinal model one actually has to present all thresholds, rather than a single mean. In practice that probably is less relevant than one might think at first, because predictions on self-report scales is less useful than metric performance data. Ordinal data also does not lend itself so much to further calculations. For example, you can use ToT measures on infusion pumps in calculating the required staffing of an intensive care unit, because seconds are metric and can be summed and divided. In contrast, it does not make sense to calculate the cognitive workload of a team of nurses by summing their self-report scores. The only possibility is to compare the strengths of predictors, but that does not require predictions.</p>
<div class="sourceCode" id="cb647"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb647-1"><a href="glm.html#cb647-1"></a><span class="kw">detach</span>(BrowsingAB)</span></code></pre></div>
</div>
<div id="beta-reg" class="section level3">
<h3><span class="header-section-number">6.4.2</span> Beta regression</h3>
<p>One of the most futile discussions in methodology research to my mind is whether one should use a four, five or seven binned Likert scale. From a pure measurement point of view, more bins give better resolution, the ultimate consequence being not to bin at all, that is using continuous rating scales. At the same time, many rating responses come from multiple item scales, which multiplies the number of bins. Speaking of ordinal regression, it seems reasonable to have seven intercepts for a single item scale, but who would want 14 or 21 for a two or three-item scale? And most scales have more items than that, which is good from a psychometric perspective.</p>
<p>In fact, psychometric research in the process of developing rating scales routinely uses a method called <em>confirmatory factor analysis</em>, which derives from the Gaussian linear model and inherits its assumptions. Not surprisingly, most research applying the very same instruments also use plain linear models. It seems fair enough to take a multi-item scale as a continuous measure, but given the framework of GLM, it is unnecessary (to put it mildly) to go along with the assumptions of Normality and linearity. While the link function for a double bounded response variable is simply the logit, the only missing ingredient is a double bounded error distribution. Enter beta distribution!</p>
<p>We demonstrate beta regression on rating scales at the example of the CUE8 study. This study aimed at assessing whether remote usability testing arrives at the same ToT measures as in moderated sessions. As we have seen in <a href="mlm.html#nested-re">5.6</a>, the difference is marginal. But, rating scales are susceptible for all kinds of cognitive and social biases. For that reason, a golden rule for user test moderators is to constantly remind participants to not blame themselves for errors. Reportedly, test moderators also do help participants (after counting to 10) in order to minimize frustration (and maximize information flow). What could the presence or absence of a moderator do to satisfaction ratings? Perhaps, remote participants feel the lack of assurance and support as higher levels of frustration. Furthermore, it is not unlikely that satisfaction ratings are sensitive to idiosyncratics in the process and setting of the user test, such that we could even expect differences between teams.</p>
<!--




MOVE TO MODEL SELECTION







-->
<p>Before we build the model, there are two issues to regard: First, the boundaries are 0 and 1, which requires a rescaling of responses into this interval. The SUS <!-- #119 -->scores are on a scale from 0 to 100 and a divisor of 100 would produce the desired interval. Second, the responses must actually lie <em>strictly between 0 and 1</em>, excluding the boundaries. On (quasi)continuous scales, it seems not very likely to have 0 or 1 as response, but it can happen. Indeed, participants in the CUE8 study have responded with a satisfaction rating of 100 quite often.</p>
<p>A practical solution is to scale the responses in such a way as to avoid the two boundaries, which is what the following hack does.</p>
<ol style="list-style-type: decimal">
<li>add a tiny value to all responses</li>
<li>create a divisor by adding twice that value to the maximum value the responses can take</li>
<li>divide all responses by that divisor</li>
</ol>
<p>You may find it inappropriate to mangle a response variable in such an arbitrary way. However, keep in mind that the levels of ordinal responses are highly arbitrary. In terms of measurement theory, all transformations that maintain the order are permitted for ordinal scales. For the following analysis, the data set was further reduced by averaging the scores across tasks and excluding probable cheaters with a ToT &lt; 30s.</p>
<div class="sourceCode" id="cb648"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb648-1"><a href="glm.html#cb648-1"></a><span class="kw">attach</span>(CUE8)</span>
<span id="cb648-2"><a href="glm.html#cb648-2"></a></span>
<span id="cb648-3"><a href="glm.html#cb648-3"></a>D_cue8_SUS &lt;-</span>
<span id="cb648-4"><a href="glm.html#cb648-4"></a><span class="st">  </span>D_cue8 <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb648-5"><a href="glm.html#cb648-5"></a><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(SUS)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb648-6"><a href="glm.html#cb648-6"></a><span class="st">  </span><span class="kw">group_by</span>(Part, Team, Condition) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb648-7"><a href="glm.html#cb648-7"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarize</span>(<span class="dt">ToT =</span> <span class="kw">sum</span>(ToT),</span>
<span id="cb648-8"><a href="glm.html#cb648-8"></a>            <span class="dt">SUS =</span> <span class="kw">mean</span>(SUS)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb648-9"><a href="glm.html#cb648-9"></a><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></span>
<span id="cb648-10"><a href="glm.html#cb648-10"></a><span class="st">  </span><span class="kw">filter</span>(ToT <span class="op">&gt;</span><span class="st"> </span><span class="dv">30</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb648-11"><a href="glm.html#cb648-11"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">SUS =</span> (SUS <span class="op">+</span><span class="st"> </span><span class="dv">1</span>)<span class="op">/</span>(<span class="dv">100</span> <span class="op">+</span><span class="st"> </span><span class="dv">2</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="co">## rescaling to ]0;1[</span></span>
<span id="cb648-12"><a href="glm.html#cb648-12"></a><span class="st">  </span><span class="kw">as_tbl_obs</span>()</span>
<span id="cb648-13"><a href="glm.html#cb648-13"></a></span>
<span id="cb648-14"><a href="glm.html#cb648-14"></a>D_cue8_SUS <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb648-15"><a href="glm.html#cb648-15"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Team, <span class="dt">y =</span> SUS, <span class="dt">fill =</span> Condition)) <span class="op">+</span></span>
<span id="cb648-16"><a href="glm.html#cb648-16"></a><span class="st">  </span><span class="kw">geom_violin</span>() <span class="op">+</span></span>
<span id="cb648-17"><a href="glm.html#cb648-17"></a><span class="st">  </span><span class="kw">geom_jitter</span>()</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-143-1.png" width="90%" /></p>
<div class="sourceCode" id="cb649"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb649-1"><a href="glm.html#cb649-1"></a>M_<span class="dv">5</span>_bet &lt;-</span>
<span id="cb649-2"><a href="glm.html#cb649-2"></a><span class="st">  </span>D_cue8_SUS <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb649-3"><a href="glm.html#cb649-3"></a><span class="st">  </span><span class="kw">brm</span>(SUS <span class="op">~</span><span class="st"> </span>Condition <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span> <span class="op">|</span><span class="st"> </span>Team),</span>
<span id="cb649-4"><a href="glm.html#cb649-4"></a>      <span class="dt">family =</span> <span class="kw">Beta</span>(<span class="dt">link =</span> <span class="st">&quot;logit&quot;</span>), <span class="dt">iter =</span> <span class="dv">0</span>, <span class="dt">chains =</span> <span class="dv">1</span>,</span>
<span id="cb649-5"><a href="glm.html#cb649-5"></a>      <span class="dt">data =</span> .)</span>
<span id="cb649-6"><a href="glm.html#cb649-6"></a></span>
<span id="cb649-7"><a href="glm.html#cb649-7"></a>M_<span class="dv">5</span>_bet &lt;-</span>
<span id="cb649-8"><a href="glm.html#cb649-8"></a><span class="st">  </span>D_cue8_SUS <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb649-9"><a href="glm.html#cb649-9"></a><span class="st">  </span><span class="kw">brm</span>(<span class="dt">fit =</span> M_<span class="dv">5</span>_bet, <span class="dt">data =</span> .)</span>
<span id="cb649-10"><a href="glm.html#cb649-10"></a></span>
<span id="cb649-11"><a href="glm.html#cb649-11"></a><span class="kw">sync_CE</span>(M_<span class="dv">5</span>_bet, <span class="dt">Env =</span> CUE8)</span></code></pre></div>
<div class="sourceCode" id="cb650"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb650-1"><a href="glm.html#cb650-1"></a>M_<span class="dv">5</span>_bet</span></code></pre></div>
<pre><code>##  Family: beta 
##   Links: mu = logit; phi = identity 
## Formula: SUS ~ Condition + (1 | Team) 
##    Data: . (Number of observations: 363) 
## Samples: 4 chains, each with iter = 2000; warmup = 1000; thin = 1;
##          total post-warmup samples = 4000
## 
## Group-Level Effects: 
## ~Team (Number of levels: 7) 
##               Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
## sd(Intercept)     0.81      0.39     0.34     1.95 1.01      554      231
## 
## Population-Level Effects: 
##                    Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
## Intercept              0.79      0.52    -0.51     1.78 1.00      567      234
## Conditionmoderated    -0.34      0.70    -1.70     1.11 1.00     1124     1088
## 
## Family Specific Parameters: 
##     Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
## phi     4.13      0.30     3.56     4.74 1.00     3390     2547
## 
## Samples were drawn using sampling(NUTS). For each parameter, Bulk_ESS
## and Tail_ESS are effective sample size measures, and Rhat is the potential
## scale reduction factor on split chains (at convergence, Rhat = 1).</code></pre>
<div class="sourceCode" id="cb652"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb652-1"><a href="glm.html#cb652-1"></a><span class="kw">fixef</span>(M_<span class="dv">5</span>_bet)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">fixef</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Intercept</td>
<td align="right">0.798</td>
<td align="right">-0.506</td>
<td align="right">1.78</td>
</tr>
<tr class="even">
<td align="left">Conditionmoderated</td>
<td align="right">-0.343</td>
<td align="right">-1.699</td>
<td align="right">1.11</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb653"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb653-1"><a href="glm.html#cb653-1"></a><span class="kw">grpef</span>(M_<span class="dv">5</span>_bet)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">type</th>
<th align="left">fixef</th>
<th align="left">re_factor</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">grpef</td>
<td align="left">Intercept</td>
<td align="left">Team</td>
<td align="right">0.71</td>
<td align="right">0.344</td>
<td align="right">1.95</td>
</tr>
</tbody>
</table>
<p>Do participants in remote sessions feel less satisfied? There seems to be a slight disadvantage, but we cannot confirm this with sufficient certainty. In contrast, the variation between teams is substantial, which indicates that SUS ratings are not independent of the particular setting. That is rather concerning for a widely used and allegedly validated rating scale.</p>
<!-- Is that a realistic model for the SUS responses? Another glance at the violin plot suggests another pathology: the teams seem to differ in variation. In the closing section I will briefly demonstrate how to deal with differences in variance (rather than mean) by a *distributional model*. -->
</div>
</div>
<div id="distributional-models" class="section level2">
<h2><span class="header-section-number">6.5</span> Beyond mean: distributional models</h2>
<p>The framework of GLM, as flexible as it has proven to be up to this point, has one major limitation: it renders the relationship between predictors and the location of the response. We only think in terms of impact factors that improve (or damage) average response times, error rates, satisfaction ratings etc. As we have seen multiple times in chapter <a href="mlm.html#mlm">5</a>, variance matters, too. With multi-level models we can estimate variance within a sample and even compare variance across samples, like in the CUE8 case, where more variance is due to teams rather than due to participants. What cannot be done with plain multi-level models is estimate effects on variance, like: “Do teams in CUE8 differ in the variation of responses?”. That brings me to the final feature of modern regression modelling in the scope of this book. With <em>distributional models</em>, we can put predictors on any distribution parameter that we want, not just location <span class="math inline">\(\mu\)</span>.</p>
<!-- That brings me to the final feature of modern regression modelling in the scope of this book. GLMs greatly enhanced the scope of modelling by giving us the choice of response distributions and linearizing functions. Still, all models introduced so far establish an association between predictors and location ($\mu$), only.  -->
<p>All but the one-parameter distributions come with additional parameters that allow to accomodate the dispersion of the error distribution, or even its skew in the case of Exgaussian models. In teh following I will present two application scenarios for distributional models. We start with a designometric problem when using bi-polar rating scales (see <a href="glm.html#rating-scales">6.4</a>.) In the next section I will illustrate the problem on simulated data from two items with different anchoring. Subsequently, we will see on a real data set, that participants differ in how they exploit the range of a rating scale. by a simple simulation will show what effect item anchoring can have and how a distributional model can deal with such a situation.</p>
<div id="item-anchoring" class="section level3">
<h3><span class="header-section-number">6.5.1</span> Item-level anchoring in rating scales</h3>
<p>As a first illustration, imagine two versions of a continuous rating scale for visual beauty that differ in how their extreme levels are labelled:</p>
<ol style="list-style-type: decimal">
<li>like the ugliest website I have ever seen: 0 ——– 1 like the most beautiful website</li>
<li>distgusting as a screenshot from a splatter movie: 0 ——– 1 like the most beautiful sunset</li>
</ol>
<!-- The second scale has the lower and upper anchors moved to more extremes and two questions arise: -->
<!-- 1. Is the overall location of the scale untouched, that is, have both anchors been moved by the same distance outwards? -->
<!-- 1. What range do websites cover as compared to all thinkable visual impressions? -->
<p>For the sake of simplicity (not for a strong research design), let us assume that one participant has rated a sample of 400 websites in two conditions: narrow anchoring and wide anchoring. The following simulates data such that both anchorings have the same location (<code>mu</code>), but the narrow anchoring condition produces a much more exploitation of the range. Note that parameter <code>phi</code> does not increase variance, but just the opposite: the range of the scale is expanded, which lets the variance shrink. It is comparable to the number of trials in a binomial process. The more trials there are, the relatively tighter the distribution becomes.</p>
<div class="sourceCode" id="cb654"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb654-1"><a href="glm.html#cb654-1"></a><span class="kw">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb654-2"><a href="glm.html#cb654-2"></a></span>
<span id="cb654-3"><a href="glm.html#cb654-3"></a>N =<span class="st"> </span><span class="dv">400</span></span>
<span id="cb654-4"><a href="glm.html#cb654-4"></a></span>
<span id="cb654-5"><a href="glm.html#cb654-5"></a>Conditions &lt;-</span>
<span id="cb654-6"><a href="glm.html#cb654-6"></a><span class="st">  </span><span class="kw">tribble</span>(<span class="op">~</span>Anchoring, <span class="op">~</span>mu, <span class="op">~</span>phi,</span>
<span id="cb654-7"><a href="glm.html#cb654-7"></a>             <span class="st">&quot;wide&quot;</span>, <span class="fl">.3</span>, <span class="dv">18</span>,</span>
<span id="cb654-8"><a href="glm.html#cb654-8"></a>             <span class="st">&quot;narrow&quot;</span>, <span class="fl">.3</span>, <span class="dv">6</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb654-9"><a href="glm.html#cb654-9"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">a =</span> mu <span class="op">*</span><span class="st"> </span>phi,</span>
<span id="cb654-10"><a href="glm.html#cb654-10"></a>         <span class="dt">b =</span> phi <span class="op">-</span><span class="st"> </span>mu <span class="op">*</span><span class="st"> </span>phi)</span>
<span id="cb654-11"><a href="glm.html#cb654-11"></a></span>
<span id="cb654-12"><a href="glm.html#cb654-12"></a>D_Anchor &lt;-<span class="st"> </span></span>
<span id="cb654-13"><a href="glm.html#cb654-13"></a><span class="st">  </span><span class="kw">tibble</span>(<span class="dt">Obs =</span> <span class="dv">1</span><span class="op">:</span>N,</span>
<span id="cb654-14"><a href="glm.html#cb654-14"></a>             <span class="dt">Anchoring =</span> <span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;wide&quot;</span>, <span class="st">&quot;narrow&quot;</span>), N<span class="op">/</span><span class="dv">2</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb654-15"><a href="glm.html#cb654-15"></a><span class="st">  </span><span class="kw">left_join</span>(Conditions) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb654-16"><a href="glm.html#cb654-16"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">rating =</span> <span class="kw">rbeta</span>(N, a, b))</span>
<span id="cb654-17"><a href="glm.html#cb654-17"></a></span>
<span id="cb654-18"><a href="glm.html#cb654-18"></a>D_Anchor <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb654-19"><a href="glm.html#cb654-19"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Anchoring, <span class="dt">y =</span> rating)) <span class="op">+</span></span>
<span id="cb654-20"><a href="glm.html#cb654-20"></a><span class="st">  </span><span class="kw">geom_violin</span>() <span class="op">+</span></span>
<span id="cb654-21"><a href="glm.html#cb654-21"></a><span class="st">  </span><span class="kw">ylim</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-147-1.png" width="90%" /></p>
<p><em>Note</em> that the <code>rbeta</code> command uses a different parametrization of beta distribution, with parameters <code>a</code> and <code>b</code>, which have are linked to the distribution mean and variance in rather convoluted ways.</p>
<p>The two response distributions have the same location, but the more narrow anchoring produces a wider dispersion of responses. How would we confirm this statistically? The Brms engine can link predictors to <em>any</em> other parameter of the response distribution, which teh author of the package calls <em>distributional models</em>. They have an immense potential as they relax another assumption of GLM, namely that all variance parameters must strictly follow the mean-variance relationship demanded by a distribution family. As we have seen, one can easily create a case where this assumption is violated.</p>
<!-- ```{r} -->
<!-- plot_rbinom <- function(N, size, p) { -->
<!--   tibble(succ = rbinom(N, size, p)) %>%  -->
<!--   ggplot(aes(x = succ)) + -->
<!--   geom_histogram(bins = size + 1) + -->
<!--   xlim(0, size)   -->
<!-- } -->
<!-- plot_rbinom(1000, 10, .5) -->
<!-- plot_rbinom(1000, 50, .5) -->
<!-- ``` -->
<p>For Beta distributions, a large <span class="math inline">\(\phi\)</span> results in reduced dispersion and vive versa. Accordingly, when used in a distributional model, a positive effect decreases variance.</p>
<p>When estimating dispersion or scale parameters, we have to regard that these are positive, strictly. The Brms engine simply extends the principle of link functions to parameters other than the <span class="math inline">\(\mu\)</span> and sets a default log link for <span class="math inline">\(\phi\)</span>. In order to estimate the changes in <span class="math inline">\(\mu\)</span> and <span class="math inline">\(\phi\)</span> simultaneously, the brms engine receives two regression formulas. Having multiple formulas for a regresson model is a notable extension of the R model specification language, which is why Brms brings its own command <code>bf()</code> to collect these. We run a distributional Beta regression on the simulated rating scale responses:</p>
<div class="sourceCode" id="cb655"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb655-1"><a href="glm.html#cb655-1"></a>M_beta &lt;-<span class="st"> </span><span class="kw">brm</span>(<span class="kw">bf</span>(rating <span class="op">~</span><span class="st"> </span><span class="dv">0</span> <span class="op">+</span><span class="st"> </span>Anchoring,</span>
<span id="cb655-2"><a href="glm.html#cb655-2"></a>                 phi <span class="op">~</span><span class="st"> </span><span class="dv">0</span> <span class="op">+</span><span class="st"> </span>Anchoring),</span>
<span id="cb655-3"><a href="glm.html#cb655-3"></a>              <span class="dt">family =</span> <span class="kw">Beta</span>(),</span>
<span id="cb655-4"><a href="glm.html#cb655-4"></a>              <span class="dt">data =</span> D_Anchor)</span></code></pre></div>
<p>The parameter table below contains the two regular coefficients on location and, as expected, there is little difference in location. The intercept on scale parameter <span class="math inline">\(\phi\)</span> is the scale in the narrow condition (with wider variance). The treatment effect on <span class="math inline">\(\phi\)</span> is positive on the log scale, which means it is deflates variance, just as expected.</p>
<div class="sourceCode" id="cb656"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb656-1"><a href="glm.html#cb656-1"></a><span class="kw">posterior</span>(M_beta) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb656-2"><a href="glm.html#cb656-2"></a><span class="st">  </span><span class="kw">select</span>(parameter, value) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb656-3"><a href="glm.html#cb656-3"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">value =</span> <span class="kw">if_else</span>(<span class="kw">str_detect</span>(parameter, <span class="st">&quot;phi&quot;</span>), </span>
<span id="cb656-4"><a href="glm.html#cb656-4"></a>                         <span class="kw">exp</span>(value), </span>
<span id="cb656-5"><a href="glm.html#cb656-5"></a>                         <span class="kw">inv_logit</span>(value))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb656-6"><a href="glm.html#cb656-6"></a><span class="st">  </span><span class="kw">group_by</span>(parameter) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb656-7"><a href="glm.html#cb656-7"></a><span class="st">  </span><span class="kw">summarize</span>(<span class="dt">center =</span> <span class="kw">median</span>(value),</span>
<span id="cb656-8"><a href="glm.html#cb656-8"></a>            <span class="dt">lower =</span> <span class="kw">quantile</span>(value, <span class="fl">.025</span>),</span>
<span id="cb656-9"><a href="glm.html#cb656-9"></a>            <span class="dt">upper =</span> <span class="kw">quantile</span>(value, <span class="fl">.975</span>))</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">parameter</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">b_Anchoringwide</td>
<td align="right">0.371</td>
<td align="right">0.313</td>
<td align="right">0.432</td>
</tr>
<tr class="even">
<td align="left">b_Intercept</td>
<td align="right">0.421</td>
<td align="right">0.367</td>
<td align="right">0.476</td>
</tr>
<tr class="odd">
<td align="left">b_phi_Anchoringwide</td>
<td align="right">4.501</td>
<td align="right">2.632</td>
<td align="right">7.562</td>
</tr>
<tr class="even">
<td align="left">b_phi_Intercept</td>
<td align="right">5.103</td>
<td align="right">3.481</td>
<td align="right">7.288</td>
</tr>
</tbody>
</table>
<p><em>Note</em> that as of writing this, the Bayr package has not yet evolved to handle distributional models. One issue is that the two parameters are on different scales ()</p>
<p>With such a distributional model we can discover differences in anchoring. And, even better, we can account for it. It intuitively makes sense to mix items with extreme and modest anchoring. By merging a distributional model with a designometric multi-level model <a href="mlm.html#designometrix">5.8.4</a>, we can evaluate and use such heteogeneous scales. The general concept is shown in the next section, where we account for participant-level employment of scale.</p>
</div>
<div id="part-employment" class="section level3">
<h3><span class="header-section-number">6.5.2</span> Participant-level employment of scale</h3>
<p>In the previous section, we have seen how to discover differences in anchoring of items. What is seen frequently in designometric studies is different response patterns in participants, in particular how much they tend to use the extremes. We can account for that by letting the variance parameter vary across participants. The last model in this chapter is putting it all together:</p>
<ol style="list-style-type: decimal">
<li>a structural part for the mean with a fixed-effects polynomial term and a participant-level random effect for the response curve.</li>
<li>a structural part for scale parameter <code>rho</code> with a participant-level random effect on the scale parameter <code>rho</code></li>
<li>a Beta shape of randomness</li>
</ol>
<div class="sourceCode" id="cb657"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb657-1"><a href="glm.html#cb657-1"></a><span class="kw">attach</span>(Uncanny)</span></code></pre></div>
<div class="sourceCode" id="cb658"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb658-1"><a href="glm.html#cb658-1"></a>RK_<span class="dv">1</span> &lt;-</span>
<span id="cb658-2"><a href="glm.html#cb658-2"></a><span class="st">  </span>RK_<span class="dv">1</span> <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb658-3"><a href="glm.html#cb658-3"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">response_unit =</span> mascutils<span class="op">::</span><span class="kw">rescale_centered</span>(response, <span class="dt">scale =</span> <span class="fl">.99</span>) <span class="op">+</span><span class="st"> </span><span class="dv">1</span>)</span>
<span id="cb658-4"><a href="glm.html#cb658-4"></a></span>
<span id="cb658-5"><a href="glm.html#cb658-5"></a>M_poly_<span class="dv">3</span>_beta  &lt;-</span>
<span id="cb658-6"><a href="glm.html#cb658-6"></a><span class="st">  </span><span class="kw">brm</span>(<span class="dt">formula =</span> response_unit <span class="op">~</span><span class="st"> </span><span class="dv">1</span> <span class="op">+</span><span class="st"> </span>huMech1 <span class="op">+</span><span class="st"> </span>huMech2 <span class="op">+</span><span class="st"> </span>huMech3 <span class="op">+</span><span class="st"> </span></span>
<span id="cb658-7"><a href="glm.html#cb658-7"></a><span class="st">        </span>(<span class="dv">1</span> <span class="op">+</span><span class="st"> </span>huMech1 <span class="op">+</span><span class="st"> </span>huMech2 <span class="op">+</span><span class="st"> </span>huMech3 <span class="op">|</span><span class="st"> </span>Part),</span>
<span id="cb658-8"><a href="glm.html#cb658-8"></a>      <span class="dt">family =</span> <span class="kw">Beta</span>(),</span>
<span id="cb658-9"><a href="glm.html#cb658-9"></a>      <span class="dt">data =</span> RK_<span class="dv">1</span>,</span>
<span id="cb658-10"><a href="glm.html#cb658-10"></a>      <span class="dt">inits =</span> <span class="dv">0</span>)</span>
<span id="cb658-11"><a href="glm.html#cb658-11"></a></span>
<span id="cb658-12"><a href="glm.html#cb658-12"></a>M_poly_<span class="dv">3</span>_beta_dist  &lt;-</span>
<span id="cb658-13"><a href="glm.html#cb658-13"></a><span class="st">  </span><span class="kw">brm</span>(<span class="dt">formula =</span> <span class="kw">bf</span>(response_unit <span class="op">~</span><span class="st"> </span><span class="dv">1</span> <span class="op">+</span><span class="st"> </span>huMech1 <span class="op">+</span><span class="st"> </span>huMech2 <span class="op">+</span><span class="st"> </span>huMech3 <span class="op">+</span><span class="st"> </span></span>
<span id="cb658-14"><a href="glm.html#cb658-14"></a><span class="st">        </span>(<span class="dv">1</span> <span class="op">+</span><span class="st"> </span>huMech1 <span class="op">+</span><span class="st"> </span>huMech2 <span class="op">+</span><span class="st"> </span>huMech3 <span class="op">|</span><span class="st"> </span>Part),</span>
<span id="cb658-15"><a href="glm.html#cb658-15"></a>                   phi <span class="op">~</span><span class="st"> </span><span class="dv">1</span> <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>Part)),</span>
<span id="cb658-16"><a href="glm.html#cb658-16"></a>      <span class="dt">family =</span> <span class="kw">Beta</span>(),</span>
<span id="cb658-17"><a href="glm.html#cb658-17"></a>      <span class="dt">data =</span> RK_<span class="dv">1</span>,</span>
<span id="cb658-18"><a href="glm.html#cb658-18"></a>      <span class="dt">inits =</span> <span class="dv">0</span>)</span></code></pre></div>
<p><em>Note</em> that on such more exotic and complex models, Brm sometimes has difficulties in finding valid good starting values for the MCMC walk. Like here, it often helps to fix all starting values to Zero.</p>
<p>If participants show different employment of scale, we should see that on the participant-level standard deviation of <code>rho</code>.</p>
<!-- We return to this case with a formal and a theoretical conclusion in \@ref(choos_resp). -->
<div class="sourceCode" id="cb659"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb659-1"><a href="glm.html#cb659-1"></a><span class="kw">bind_rows</span>(</span>
<span id="cb659-2"><a href="glm.html#cb659-2"></a>  <span class="kw">posterior</span>(M_poly_<span class="dv">3</span>_beta),</span>
<span id="cb659-3"><a href="glm.html#cb659-3"></a>  <span class="kw">posterior</span>(M_poly_<span class="dv">3</span>_beta_dist)</span>
<span id="cb659-4"><a href="glm.html#cb659-4"></a>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb659-5"><a href="glm.html#cb659-5"></a><span class="st">  </span><span class="kw">filter</span>(type <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;grpef&quot;</span>, <span class="st">&quot;fixef&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb659-6"><a href="glm.html#cb659-6"></a><span class="st">  </span><span class="kw">group_by</span>(model, parameter) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb659-7"><a href="glm.html#cb659-7"></a><span class="st">  </span><span class="kw">summarize</span>(<span class="dt">center =</span> <span class="kw">median</span>(value),</span>
<span id="cb659-8"><a href="glm.html#cb659-8"></a>            <span class="dt">lower =</span> <span class="kw">quantile</span>(value, <span class="fl">.025</span>),</span>
<span id="cb659-9"><a href="glm.html#cb659-9"></a>            <span class="dt">upper =</span> <span class="kw">quantile</span>(value, <span class="fl">.975</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb659-10"><a href="glm.html#cb659-10"></a><span class="st">  </span><span class="kw">ungroup</span>()</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">model</th>
<th align="left">parameter</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">M_poly_3_beta</td>
<td align="left">b_huMech1</td>
<td align="right">4.945</td>
<td align="right">3.320</td>
<td align="right">6.613</td>
</tr>
<tr class="even">
<td align="left">M_poly_3_beta</td>
<td align="left">b_huMech2</td>
<td align="right">-14.596</td>
<td align="right">-18.284</td>
<td align="right">-11.072</td>
</tr>
<tr class="odd">
<td align="left">M_poly_3_beta</td>
<td align="left">b_huMech3</td>
<td align="right">10.244</td>
<td align="right">7.991</td>
<td align="right">12.666</td>
</tr>
<tr class="even">
<td align="left">M_poly_3_beta</td>
<td align="left">b_Intercept</td>
<td align="right">-0.209</td>
<td align="right">-0.513</td>
<td align="right">0.077</td>
</tr>
<tr class="odd">
<td align="left">M_poly_3_beta</td>
<td align="left">sd_Part__huMech1</td>
<td align="right">3.496</td>
<td align="right">2.421</td>
<td align="right">4.889</td>
</tr>
<tr class="even">
<td align="left">M_poly_3_beta</td>
<td align="left">sd_Part__huMech2</td>
<td align="right">7.513</td>
<td align="right">5.099</td>
<td align="right">10.633</td>
</tr>
<tr class="odd">
<td align="left">M_poly_3_beta</td>
<td align="left">sd_Part__huMech3</td>
<td align="right">4.942</td>
<td align="right">3.404</td>
<td align="right">6.952</td>
</tr>
<tr class="even">
<td align="left">M_poly_3_beta</td>
<td align="left">sd_Part__Intercept</td>
<td align="right">0.692</td>
<td align="right">0.517</td>
<td align="right">0.931</td>
</tr>
<tr class="odd">
<td align="left">M_poly_3_beta_dist</td>
<td align="left">b_huMech1</td>
<td align="right">4.516</td>
<td align="right">2.975</td>
<td align="right">6.085</td>
</tr>
<tr class="even">
<td align="left">M_poly_3_beta_dist</td>
<td align="left">b_huMech2</td>
<td align="right">-13.723</td>
<td align="right">-17.080</td>
<td align="right">-10.462</td>
</tr>
<tr class="odd">
<td align="left">M_poly_3_beta_dist</td>
<td align="left">b_huMech3</td>
<td align="right">9.750</td>
<td align="right">7.587</td>
<td align="right">11.982</td>
</tr>
<tr class="even">
<td align="left">M_poly_3_beta_dist</td>
<td align="left">b_Intercept</td>
<td align="right">-0.171</td>
<td align="right">-0.439</td>
<td align="right">0.114</td>
</tr>
<tr class="odd">
<td align="left">M_poly_3_beta_dist</td>
<td align="left">b_phi_Intercept</td>
<td align="right">1.388</td>
<td align="right">1.084</td>
<td align="right">1.693</td>
</tr>
<tr class="even">
<td align="left">M_poly_3_beta_dist</td>
<td align="left">sd_Part__huMech1</td>
<td align="right">3.373</td>
<td align="right">2.393</td>
<td align="right">4.784</td>
</tr>
<tr class="odd">
<td align="left">M_poly_3_beta_dist</td>
<td align="left">sd_Part__huMech2</td>
<td align="right">7.372</td>
<td align="right">5.312</td>
<td align="right">10.308</td>
</tr>
<tr class="even">
<td align="left">M_poly_3_beta_dist</td>
<td align="left">sd_Part__huMech3</td>
<td align="right">4.966</td>
<td align="right">3.579</td>
<td align="right">6.857</td>
</tr>
<tr class="odd">
<td align="left">M_poly_3_beta_dist</td>
<td align="left">sd_Part__Intercept</td>
<td align="right">0.668</td>
<td align="right">0.497</td>
<td align="right">0.910</td>
</tr>
<tr class="even">
<td align="left">M_poly_3_beta_dist</td>
<td align="left">sd_Part__phi_Intercept</td>
<td align="right">0.752</td>
<td align="right">0.580</td>
<td align="right">1.030</td>
</tr>
</tbody>
</table>
<p><em>Note</em> that the Bayr package does not fully support distributional models, yet, and we have to make without some of the high-level commands, such as grpef.</p>
<p>The standard deviation of <code>rho</code> is positive and even half of the population average. While a regular Beta model adjusts the variance according to the usual mean-variance relationship, the distributional model also accounts for overall broader and narrower distributions. We would probably see some subtle adjustments in predictive plots, but this model is also very complex, as every participant get their own <code>rho</code>. Therefore, we defer a deeper inspection to the end of the next chapter, where we demonstrate the superiority of the distributional model using information criteria.</p>
<!-- , for example on mental *workload*. Imagine you are evaluating a simulator-based driver training, designed with the  idea that movement tasks are better learned if you begin fast, not accurate. This training uses *speed episodes*, where the trainee is asked to do 20% faster, at the expense of accuracy. In such an experiment, *workload* can play a moderating factor, because if you work very hard, you can be fast and accurate at the same time. In such a case, we can  -->
<div class="sourceCode" id="cb660"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb660-1"><a href="glm.html#cb660-1"></a><span class="kw">detach</span>(Uncanny)</span></code></pre></div>
</div>
<div id="part-skew" class="section level3">
<h3><span class="header-section-number">6.5.3</span> Participant-level skew in reaction times</h3>
<p>In <a href="glm.html#exgauss-reg">6.3.2</a>, the Exgaussian model seemed to sit well with reaction times, accomodating their left skew better than Gaussian or Gamma distributions. But if we review the participant-level plots carefully, we see that the shape of randomness differ between participants. There is visible differences in variance, as well as in skew.</p>
<p>The following distributional model estimates the same location effects (PrimeGeek), but grant all participants their own variance and skew. Two distributional parameters, <span class="math inline">\(\sigma\)</span> for variance and <span class="math inline">\(\beta\)</span> for skew are added to the multi-line formula interface. The maximum distributional model would estimate fully separate distributions per every participant and condition. But, since the two response distributions (Geek/nogeek) appear similar in the plots, a reduced predictor term is used, assuming that variance and shape are not affected by the experimental condition, but only the person doing the task. That is further justified by the fact, that the effect of priming categories were meager, at best.</p>
<div class="sourceCode" id="cb661"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb661-1"><a href="glm.html#cb661-1"></a><span class="kw">attach</span>(Hugme)</span></code></pre></div>
<div class="sourceCode" id="cb662"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb662-1"><a href="glm.html#cb662-1"></a>M_<span class="dv">1</span>_exg_dist  &lt;-</span>
<span id="cb662-2"><a href="glm.html#cb662-2"></a><span class="st">  </span><span class="kw">brm</span>(<span class="dt">formula =</span> <span class="kw">bf</span>(RT    <span class="op">~</span><span class="st"> </span><span class="dv">1</span> <span class="op">+</span><span class="st"> </span>PrimeGeek <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span> <span class="op">+</span><span class="st"> </span>PrimeGeek<span class="op">|</span>Part),</span>
<span id="cb662-3"><a href="glm.html#cb662-3"></a>                   sigma <span class="op">~</span><span class="st"> </span><span class="dv">1</span> <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>Part),</span>
<span id="cb662-4"><a href="glm.html#cb662-4"></a>                   beta  <span class="op">~</span><span class="st"> </span><span class="dv">1</span> <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>Part)),</span>
<span id="cb662-5"><a href="glm.html#cb662-5"></a>      <span class="dt">family =</span> <span class="kw">exgaussian</span>(),</span>
<span id="cb662-6"><a href="glm.html#cb662-6"></a>      <span class="dt">data =</span> D_hugme,</span>
<span id="cb662-7"><a href="glm.html#cb662-7"></a>      <span class="dt">inits =</span> <span class="dv">0</span>)</span></code></pre></div>
<p>Next, we extract the fixed efefcts, as well as the group-level standard deviation from the posterior. If participants show different patterns of randomness, we should see that in participant-level variation of <span class="math inline">\(\sigma\)</span> and <span class="math inline">\(\beta\)</span>.</p>
<div class="sourceCode" id="cb663"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb663-1"><a href="glm.html#cb663-1"></a>P_<span class="dv">1</span>_exg_dist &lt;-<span class="st"> </span><span class="kw">posterior</span>(M_<span class="dv">1</span>_exg_dist)</span>
<span id="cb663-2"><a href="glm.html#cb663-2"></a></span>
<span id="cb663-3"><a href="glm.html#cb663-3"></a>P_<span class="dv">1</span>_exg_dist <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb663-4"><a href="glm.html#cb663-4"></a><span class="st">  </span><span class="kw">filter</span>(type <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;fixef&quot;</span>,<span class="st">&quot;grpef&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb663-5"><a href="glm.html#cb663-5"></a><span class="st">  </span><span class="kw">group_by</span>(parameter) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb663-6"><a href="glm.html#cb663-6"></a><span class="st">  </span><span class="kw">summarize</span>(<span class="dt">center =</span> <span class="kw">median</span>(value),</span>
<span id="cb663-7"><a href="glm.html#cb663-7"></a>            <span class="dt">lower =</span> <span class="kw">quantile</span>(value, <span class="fl">.025</span>),</span>
<span id="cb663-8"><a href="glm.html#cb663-8"></a>            <span class="dt">upper =</span> <span class="kw">quantile</span>(value, <span class="fl">.975</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb663-9"><a href="glm.html#cb663-9"></a><span class="st">  </span><span class="kw">ungroup</span>()</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">parameter</th>
<th align="right">center</th>
<th align="right">lower</th>
<th align="right">upper</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">b_beta_Intercept</td>
<td align="right">-2.012</td>
<td align="right">-2.133</td>
<td align="right">-1.877</td>
</tr>
<tr class="even">
<td align="left">b_Intercept</td>
<td align="right">0.620</td>
<td align="right">0.595</td>
<td align="right">0.648</td>
</tr>
<tr class="odd">
<td align="left">b_PrimeGeekTRUE</td>
<td align="right">0.010</td>
<td align="right">0.002</td>
<td align="right">0.018</td>
</tr>
<tr class="even">
<td align="left">b_sigma_Intercept</td>
<td align="right">-2.918</td>
<td align="right">-3.039</td>
<td align="right">-2.797</td>
</tr>
<tr class="odd">
<td align="left">sd_Part__beta_Intercept</td>
<td align="right">0.474</td>
<td align="right">0.394</td>
<td align="right">0.589</td>
</tr>
<tr class="even">
<td align="left">sd_Part__Intercept</td>
<td align="right">0.113</td>
<td align="right">0.095</td>
<td align="right">0.137</td>
</tr>
<tr class="odd">
<td align="left">sd_Part__PrimeGeekTRUE</td>
<td align="right">0.027</td>
<td align="right">0.020</td>
<td align="right">0.035</td>
</tr>
<tr class="even">
<td align="left">sd_Part__sigma_Intercept</td>
<td align="right">0.445</td>
<td align="right">0.352</td>
<td align="right">0.564</td>
</tr>
</tbody>
</table>
<p>The estimates confirm, that participants vary significantly in how there RTs are dispersed, as well as how skewed they are. Finally, we can also examine whether there are any associations between location, variance and skew. Recall that parameter <code>beta</code> gives the model the flexibility to add extra skew for when the location is far from the left boundary. If that were the case, we would see a correlation between participant-level random effects betwen location and skew. The following extracts the participant-level random effects for <code>beta</code>, <code>mu</code> and <code>sigma</code>:</p>
<div class="sourceCode" id="cb664"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb664-1"><a href="glm.html#cb664-1"></a>P_<span class="dv">1</span>_exg_dist <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb664-2"><a href="glm.html#cb664-2"></a><span class="st">  </span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;ranef&quot;</span> <span class="op">&amp;</span><span class="st"> </span>fixef <span class="op">==</span><span class="st"> &quot;Intercept&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb664-3"><a href="glm.html#cb664-3"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">dist_par =</span> <span class="kw">str_match</span>(parameter, <span class="st">&quot;beta|sigma&quot;</span>),</span>
<span id="cb664-4"><a href="glm.html#cb664-4"></a>         <span class="dt">dist_par =</span> <span class="kw">if_else</span>(<span class="kw">is.na</span>(dist_par), <span class="st">&quot;mu&quot;</span>, dist_par)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb664-5"><a href="glm.html#cb664-5"></a><span class="st">  </span><span class="kw">group_by</span>(dist_par, re_entity) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb664-6"><a href="glm.html#cb664-6"></a><span class="st">  </span><span class="kw">summarize</span>(<span class="dt">center =</span> <span class="kw">median</span>(value)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb664-7"><a href="glm.html#cb664-7"></a><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb664-8"><a href="glm.html#cb664-8"></a><span class="st">  </span><span class="kw">spread</span>(<span class="dt">key =</span> dist_par, <span class="dt">value =</span> center) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb664-9"><a href="glm.html#cb664-9"></a><span class="st">  </span><span class="kw">select</span>(<span class="op">-</span>re_entity) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb664-10"><a href="glm.html#cb664-10"></a><span class="st">  </span>corrr<span class="op">::</span><span class="kw">correlate</span>()</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">rowname</th>
<th align="right">beta</th>
<th align="right">mu</th>
<th align="right">sigma</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">beta</td>
<td align="right">NA</td>
<td align="right">0.771</td>
<td align="right">0.040</td>
</tr>
<tr class="even">
<td align="left">mu</td>
<td align="right">0.771</td>
<td align="right">NA</td>
<td align="right">0.353</td>
</tr>
<tr class="odd">
<td align="left">sigma</td>
<td align="right">0.040</td>
<td align="right">0.353</td>
<td align="right">NA</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb665"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb665-1"><a href="glm.html#cb665-1"></a>D_hugme <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb665-2"><a href="glm.html#cb665-2"></a><span class="st">  </span><span class="kw">filter</span>(<span class="dv">60</span> <span class="op">&lt;</span><span class="st"> </span><span class="kw">as.numeric</span>(Part) <span class="op">&amp;</span><span class="st"> </span><span class="kw">as.numeric</span>(Part) <span class="op">&lt;</span><span class="st"> </span><span class="dv">90</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb665-3"><a href="glm.html#cb665-3"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> RT)) <span class="op">+</span></span>
<span id="cb665-4"><a href="glm.html#cb665-4"></a><span class="st">  </span><span class="kw">geom_density</span>() <span class="op">+</span></span>
<span id="cb665-5"><a href="glm.html#cb665-5"></a><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>Part)</span></code></pre></div>
<p><img src="Generalized_Linear_Models_files/figure-html/unnamed-chunk-163-1.png" width="90%" /></p>
<p>One could get excited about the strong correlation between <code>mu</code> and <code>beta</code>, but recall that the amount of skew introduced by <code>beta</code> is proportional to the location. At the same time we see a mild correlation between <code>mu</code> and <code>sigma</code>, which is just the usual mean-variance relationship, which other distributions have factored in. So, from a theoretical point of view, there is not much to gain.</p>
<p>That does not mean the distributional model is a fruitless endeavour. In the following chapter, we will learn how to compare models by predictive accuracy. As it will turn out in <a href="wwm.html#choose-dist">7.2.4</a>, the distributional Beta model in the Hugme case has a better predictive accuracy than the location-only model. The same is true for the Exgaussian distributional model and left to the reader as an exercise.</p>
<p>Distributional models also can do more than just better accomodate data. Applied researchers and experimentalists traditionally embrace on-average effects, but have been mostly blind to effects on variance. In safety relevant systems, strong variance can be a real problem, because this implies that extremely poor performance is becoming more likely, increasing the risk of hazard. For such systems, variance is a crucial parameter and the dispersion effects of design changes should be tracked, carefully. For the readers interested in exercising this technique, several data sets from this book are candidates for such an analysis:</p>
<ul>
<li>CUE8: Do teams differ in the variance of ToT variance?</li>
<li>Egan: What is the most robust web design (the one that reduces variance)?</li>
<li>IPump: Does training (or the Novel design) reduce variance?</li>
</ul>
<p>Experimentalists should scratch their heads, too, and take a fresh look at their theories. Systematic variation in variance canm give further clues about what is going on in the minds of their participants. For example, it has been suggested that the Uncanny Valley effect is due to category confusion, which suggests that the longest RTs are to be found in the trough, which [%M&amp;R] have shown to be the case. In [%Keeris], we have fleshed out this theory a little more, basically saying that at every trial the confusion only occurs if the mind picks up enough detail to reject the initial category (Human Face). So, if we reduce the presentation time, we would see more responses where the initial category is kept and the emotional response is positive, rather than extrenmely negative. This hypothesis implies that variance should be largest in the range of the valley, and that with shorter presentation times variance goes up, as responses are a mix of category confusion and category retainers. If we go even shorter, retaining the category will become prominent and variance decreases again.</p>
<!-- Strong variance can be a real problem, because this implies  that extremely poor performance is possible, increasing the risk of hazards. In hazardous environments, variance  is a crucial parameter should be track carefully. Especially,  a design improvement *on average* is not necessarily accompanied by less variance (other than the mean-variance relationship prescribed by the error distribution). -->
<!-- Generalized Linear Models gave us control over the average shape of randomness, average variance and average skew. When we use these models, we gain control over the data generating process on an individual level. Distributional models have enormous potential for research and applications. *Rating scales* are vulnerable to different response patterns when the anchoring is ambivalent. But that is not the only possible response pattern. For example, in our own studies on the Uncanny Valley effect, we used analog rating scales and found some participants to reduce the task to a binary answer (left or right end). A Beta distribution can handle such a pattern well, but only if it is consistent. -->
<!--  Another example is mental *workload*. Imagine you are evaluating a simulator-based driver training, designed with the  idea that movement tasks are better learned if you begin fast, not accurate. This training uses *speed episodes*, where the trainee is asked to do 20% faster, at the expense of accuracy. In such an experiment, *workload* can play a moderating factor, because if you work very hard, you can be fast and accurate at the same time. And a few things you only learn, when you try really hard. Try you must!
-->
<div class="sourceCode" id="cb666"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb666-1"><a href="glm.html#cb666-1"></a><span class="kw">detach</span>(Hugme)</span></code></pre></div>
<hr />
<p>In the last three chapters we have met the family of models called <em>Generalized Multi-level Linear Models</em> (GMLM). Linear linear models mainly rests on the linear term, or I would rather say: the linear term is the power of linear models. The predictor side of the formula can combine multiple linear terms, at the same time, qualitative data, as captured by factor variables, can be added, too. These simple mechanism unfold into a combinatoric expansion of possibilities.</p>
<p>Technically, multi-level models don’t expand the linear term. From a utilitarian perspective, random effects are just useful, because fixef-effects factors do not really shine when factors have an abundance of levels, like participants in an experimental study. But, multi-level models also bring a new perspective, which is painfully missing in many design studies. When we start thinking about users as a population, a population with a good deal of diversity, this makes us think about individuals, too.</p>
<p>Multi-level models also redeem variance as a first-class citizen among model parameters, and we are getting used to think of distributions. Generalized Linear Models expand on that perspective and deal exclusively with what happens on the response side of a model and we learned that for practically all standard measures a Gaussian model is wrong and an (almost) right distribution can just be picked from a whole zoo, using just a few rules. To me, the framework of Generalized Linear Models feel like the sudden relief from pain, when you untie a pair of bad fitting shoes after a full day hike. And sometimes even literally, some real pain, like ordered factors for a learning curve, just goes away.</p>
<p>In the recent past have seen many researchers beginning to see the beauty of the New Statistics approach. Suddenly, people start thinking of models, rather than arcane procedures, and start tinkering. Then their favorite regression engine gets a new feature, say distributional models, and that could be worth a try, could it not?</p>
<p>This book could stop right here, where I leave my readers with an ethusiastic speech, hoping that everybody has reached this page and that some people will read on and write more about interesting models in design research. But, the abundance of possible models leaves the question of when to stop. How do we decise whether introducing, say distributional random effects, really improves the model? As we will see in the final chapter, there are two opposite forces at work, model complexity, which is to be avoided, and model fit. Let’s tie our shoes to get the best grip!</p>
<!-- Both random effects variance parameters are certaintly positive  -->
<!-- The IPump case we compared two infusion pump interfaces in three sessions and also collected measures on workload, using a self-report ratings. The results suggest that participants responded with quite different ranges. The following Beta regression  -->
<!-- ```{r} -->
<!-- detach(IPump) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- D_pumps %>%  -->
<!--   group_by(Part) %>%  -->
<!--   mutate(Part_mean = mean(workload, na.rm = T)) %>%  -->
<!--   ungroup() %>%  -->
<!--   mutate(Part_ordered = rank(Part_mean)) %>%  -->
<!--   ggplot(aes(x = as.factor(Part_ordered), y = workload)) + -->
<!--   geom_point() -->
<!-- ``` -->
<!-- ```{r} -->
<!-- M_distr <-  -->
<!--   brms(workload ~ ) -->
<!-- ``` -->
<!-- The following distributional Beta regression  -->
<!-- Back to the CUE8 study. Do teams differ in how much variance arises, as the violin plot suggests? Before we test this by a distributional model, recall once again the principle of the mean-variance relation. Beta distributions have variance tied to the mean in much the same way as binomial distributions. When moving along the range between 0 and 1, variance is largest in the center and decreases towards the boundaries. That does not seem to be the case in CUE8. Compare the distributions of Teams B and K to H and L. They all are close to the upper boundary, but show inflated variance. It could be the case, that somehow some teams trigger more extreme responses, inflating variance. The following model tests the effect of testing condition and teams on the location and dispersion of responses simultaneously. -->
<!-- ```{r opts.label = "mcmc"} -->
<!-- F_unequal_var <- bf(SUS ~ 1 + Condition + (1 | Team), -->
<!--                     phi ~ 1 + Condition + (1 | Team)) -->
<!-- M_6_bet <- -->
<!--   D_cue8_mod %>%  -->
<!--   brm(F_unequal_var, -->
<!--       family = Beta(), iter = 1, chains = 1, -->
<!--       data = .) -->
<!-- M_6_bet <- -->
<!--   D_cue8_SUS %>%  -->
<!--   brm(fit = M_6_bet, data = ., -->
<!--       chains = 6, -->
<!--       iter = 4000, -->
<!--       warmup = 2000) -->
<!-- sync_CE(M_6_bet, Env = CUE8) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- T_fixef <-  -->
<!--   brms::fixef(M_6_bet)[,c(1,3,4)] %>%  -->
<!--   as_tibble(rownames = "fixef") %>%  -->
<!--   dplyr::rename(center = Estimate) -->
<!-- T_fixef -->
<!-- ``` -->
<!-- The default behaviour of brm is that the scale parameter $\phi$ is on a log scale. As usual, lifting it to original scale by exponantiation makes it a multiplier. By $\exp(0.773) = 2.167$, we see that there is just a small difference between the two conditions. But, are there other differences between teams? The following table shows the scale multipliers for individual teams. All multipliers are closely arranged around 1, which means there is little differences. The SUS scale can safely be adminstsred in remote and moderated usability testing. -->
<!-- ```{r} -->
<!-- T_ranef <- -->
<!--   ranef(M_6_bet) %>% -->
<!--   filter(nonlin == "phi") %>%  -->
<!--   mutate_if(is.numeric, exp) %>%  -->
<!--   discard_redundant() -->
<!-- T_ranef -->
<!-- ``` -->
<!-- ```{r} -->
<!-- detach(CUE8) -->
<!-- ``` -->
<!-- ### Exercises -->
<!-- 1. The IPump study took repeated measure of workload on a one-item quasi-continuous rating scale. Examine the effect of training and Design on workload. Then take a closer look at how participants used the scale. Build a distributional model. -->
<!-- ```{r opts.label = "future"} -->
<!-- attach(IPump) -->
<!-- max(IPump$D_pumps$workload) -->
<!-- D_pumps <- -->
<!--   D_pumps %>%  -->
<!--   mutate(workload = (workload + 1)/(150 + 2)) %>%  -->
<!--   group_by(Part) %>%  -->
<!--   mutate(mean_workload = mean(workload)) %>%  -->
<!--   ungroup() %>%  -->
<!--   mutate(Part_ord = as.factor(dense_rank(mean_workload))) -->
<!-- D_pumps %>%  -->
<!--   ggplot(aes(x = Part_ord, y = workload)) + -->
<!--   geom_boxplot() + -->
<!--   geom_point() -->
<!-- ``` -->
<!-- ```{r opts.label = "future"} -->
<!-- M_bet <- -->
<!--   D_pumps %>%  -->
<!--   brm(bf(workload ~ 1 + session * Design + (session * Design|Part_ord), -->
<!--          phi ~ 1|Part), -->
<!--       family = Beta(), -->
<!--       data = ., iter = 0, chains = 0) -->
<!-- M_bet <- -->
<!--   D_pumps %>%  -->
<!--   brm(fit = M_bet, -->
<!--       data = ., iter = 3000, chains = 6, warmup = 2000) -->
<!-- T_ranef <- -->
<!--   ranef(M_bet) %>%  -->
<!--   filter(nonlin == "phi") -->
<!-- T_ranef -->
<!-- ``` -->
<!-- ## Gaussian regression revisited

The classic linear model has mercilessly been debunked at the beginning of the chapter. Time and count measures are truly not Gaussian distributed. In \@ref(), the Gaussian situation was illustrated as a bunch of SMURFS that bump into each other.

However, , is a member of the GLM framework, with some rather unusual properties. It assumes normally distributed randomness, which implies a constant variance. In contrast to all other members so far, there is no relation between mean and variance of randomness. The link function simply is *identity* and is consequential for the two problems of Gaussian regression: linearity and bounded ranges. The general take is that, whenever there is a theoretically justified and reasonably pragmatic choice, one better leaves the Gaussian alone. Here it makes its re-entrance for a type of outcome measures that are so problematic that a pragmatic choice is indicated.

### Outlier removal

Outliers are observations that were hit by one strong force, that is of no further interest. Outliers can be upper or lower. One often sees the advice to remove outliers that exceed x times the standard deviation of the outcome variable. That is clearly mistaken as we have pointed out in \@ref(residual_analysis). Outliers must be identified by their residuals, first of all. Another problem arises because the removal intervals are symmetric. Boxplots pull up the Tukey fence at $1.35 \sigma_\epsilon$ beyond or above the first and third interquartile range. No matter, whether you can easily imagine that rule, it assumes symmetry. A boxplot of the two conditions X and Y in the CUE8 study looks like this:
-->
<!--
Boxplots routinely fail at non-symmetric pattens of randomness. Violin plots render a more accurate shape of the residual distribution. The distribution in the moderated condition almost resemble a ginko leaf, full of grace and substantial all the way up. The remote condition looks like a moth that fell prey to a spider (or an obese sword-billed hummingbird). The upper part is long but of thin substance. Seeing the condition on its own, one would perhaps pull up the outlier fence at 800 seconds. However, the tail of the moderated condition is substantial and there is no firm cue there being outliers at all. Since both conditions used comparable instructions and equal tasks, the maximum value of the moderated condition is a conservative upper fence.
-->
<!--

Again, the moderated condition is smooth as a leaf. There is no reason to believe that even the shortest observation is an outlier  at all and we pull this fence up in the remote condition. Here we observe a characteristic pattern, the lower socket of the distribution supported by a clear disruption in the cloud. One possible story for this irregularity is that some remote-testing teams have not checked on task completion and the data set contains observations of cheaters (only if they were payed). Cheating is something that is untypical for the real use of the car rental website. When people enter such a site, they usually want to rent a car. Cheaters in remote studies tell us nothing about performance of real users. They are interuptive, but irrelevant. 

Now, consider the famous *Stroop task*: participants are shown a word that is written to the screen in one of three ink colours (red, blue, green). They are asked to name the ink color as quickly as possible, usually by pressing one of four assigned keys (e.g., Y = red, X = blue, N = green). The task has a twist: the presented words are themselves colour words. It can happen that the word "RED" appears in red ink, which we call the *congruent condition*, but it can also happen that the word "BLUE" appears in red letters, which is *incongruent*. Hundreds of experiments have shown, that in the incongruent condition people respond noticably slower. This observation is called the *Stroop effect* and dozens of theories have spawned from this observation. Discussing them is beyond the scope, but to not let you completely dissatisfied: people seem to always read the words, although they were not asked to do so and we can conclude:

1. Reading is overlearned. It is easier to catch a thrown stone than to resist a word.
1. When people do not follow instructions, this is often a sign of *bottom-up processing*.

What strong forces of no interest can disturb a trial in the Stroop task? Here are some examples:

1. a door could slam
2. participants motivation dropped
3. participants got really exhausted
4. participants went into a zen state of mind, absorbed in deep self observation

What about a possible number 5 on the list: some participant saw the word "gun" in red and the following association raced through their minds:

gun -> huntsman -> man who shot the wolf in little red riding hood




What would be a legitimate way to catch outliers? The Stroop task has been replicated dozens and dozens of times. For rough approximation, one takes a sample of ten papers reporting on variants of the Stroop task and derives a broadly typical range from it. As a rule of thumb, for an upper threshold adding 3 standard deviations to the highest 
, say, between 500ms and 1500ms. 


-->
<!-- ## Robust Models -->
<!-- ```{r} -->
<!-- attach(Egan) -->
<!-- M_1 -->
<!-- insight::find_formula(M_1) -->
<!-- M_1_t <-  -->
<!--   D_egan %>%  -->
<!--   mutate(logToT = log(ToT)) %>%  -->
<!--   brm(formula(M_1), family = "student", data = .) -->
<!-- ls(Egan) -->
<!-- ``` -->

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mlm.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="wwm.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
